                                                                                                                                                                                                                                           Bioinformatics,         37,         2021,         i418–i425
                                                                                                                                                                                                                                         doi:        10.1093/bioinformatics/btab271
                                                                                                                                                                                                                                                                                     ISMB/ECCB        2021
KG4SL:               knowledge               graph               neural               network                for               synthetic
lethality                prediction                in               human               cancers
Shike          Wang1,†,          Fan          Xu1,†,          Yunyang          Li2,          Jie          Wang1,          Ke          Zhang1,3,          Yong          Liu4,
Min          Wu                               5,*          and          Jie          Zheng1,6,*
1School                  of                  Information                  Science                  and                  Technology,                  ShanghaiTech                  University,                  Shanghai                  201210,                  China,                  2School                  of                  Life                  Science                  and
Technology,                    ShanghaiTech                    University,                    Shanghai                    201210,                    China,                    3Shanghai                    Institute                    of                    Microsystem                    and                   Information                    Technology,
Chinese                 Academy                 of                 Sciences,                 Shanghai                 200050,                 China,                 4Joint                 NTU-UBC                 Research                 Centre                 of                 Excellence                 in                 Active                 Living                 for                 the
Elderly,                  Nanyang                  Technological                  University,                  Singapore                  639798,                   Singapore,                  5Institute                  for                   Infocomm                  Research,                  Agency                  for                   Science,
Technology              and              Research              (A*STAR),              Singapore              138632,              Singapore              and              6Shanghai              Engineering              Research              Center              of              Intelligent              Vision
and          Imaging,          Shanghai,          201210,          China
*To         whom         correspondence         should         be         addressed.
†The         authors         wish         it         to         be         known         that,         in         their         opinion,         the         ﬁrst         two         authors         should         be         regarded         as         Joint         Authors.
Abstract
Motivation:          Synthetic          lethality           (SL)           is           a          promising          gold           mine          for          the          discovery          of           anti-cancer          drug          targets.          Wet-lab
screening          of          SL          pairs          is          afﬂicted          with          high          cost,          batch-effect,          and          off-target          problems.          Current          computational          meth-
ods        for        SL        prediction        include        gene        knock-out        simulation,        knowledge-based        data        mining        and        machine        learning        meth-
ods.          Most          of          the          existing          methods          tend          to          assume          that          SL          pairs          are          independent          of          each          other,          without          taking          into
account            the            shared            biological            mechanisms            underlying            the            SL            pairs.            Although            several            methods            have            incorporated
genomic           and           proteomic           data           to           aid           SL           prediction,           these           methods           involve           manual           feature           engineering           that           heavily
relies        on        domain        knowledge.
Results:               Here,               we               propose              a               novel               graph               neural               network               (GNN)-based               model,               named               KG4SL,               by               incorporating
knowledge               graph               (KG)               message-passing               into               SL               prediction.               The               KG               was               constructed               using               11               kinds               of               entities
including         genes,         compounds,         diseases,         biological         processes         and         24         kinds         of         relationships         that         could         be         pertinent         to
SL.          The          integration          of          KG          can          help          harness          the          independence         issue          and         circumvent          manual          feature          engineering          by
conducting        message-passing        on        the         KG.         Our        model         outperformed        all         the        state-of-the-art         baselines         in         area        under        the
curve,        area        under        precision-recall        curve        and        F1.        Extensive        experiments,        including        the        comparison        of        our        model         with
an         unsupervised         TransE         model,         a         vanilla         graph         convolutional         network         model,         and         their         combination,         demonstrated
the        signiﬁcant        impact        of        incorporating        KG        into        GNN        for        SL        prediction.
Availability       and        implementation:        :        KG4SL        is        freely        available        at        https://github.com/JieZheng-ShanghaiTech/KG4SL.
Contact:        wumin@i2r.a-star.edu.sg;        zhengjie@shanghaitech.edu.cn.
Supplementary       information:        Supplementary        data        are        available        at        Bioinformatics        online.
1        Introduction                                                                                                                                                                             identify        SL        gene        pairs.        High-throughput        wet-lab        screening        methods,
                                                                                                                                                                                                  including                chemical                liraries                (Simons                et                al.,                2001),                pooled                RNAi
Complex        biological        systems        cannot        be        composed        by        a        large        number                                                                    screening       (Luo       et       al.,       2009)       and       CRISPR-based       genome       editing       tech-
of         genes         acting         independently,         but         rely         on         the         interactions         between                                                       nology           (Du           et           al.,           2017)           have           been           used           to           find           SLs,           but           they           are
genes       which       can       be       further       classified       into       enhancing       and       suppressive                                                                        thwarted        by        varius        barriers        such        as        high        cost,        off-target        effects,        and
effects              (Dhabhar,              2009).              The              suppressive              effects              characterize              the                                      batch       effects       (Liu       et       al.,       2020).       Hence,       it       is       compelling       to       devise       effi-
situation            that,            when             mutations             occur            simultaneously             in             a             pair             of                         cient          computational           methods          to           complement           the           downsides           of           the
genes,       some       important       functions       will        be        deactivated       which       seriously                                                                             wet-lab      screening      techniques.
decrease         cell         viability,         whereas         the         mutation         in         a         single         gene         might                                                       A        spectrum        of       computational        methods       has       been        proposed        for        SL
not         affect         the         cell         viability.         A         common         type         of         suppressive         effect         is                                     prediction.       These       methods       can       be       categorized       into       three       classes.       The
synthetic        lethality       (SL;        Dobzhansky,        1946),       which        has       been        a       prom-                                                                     first            class            involves            simulating            in            silico            knockouts            using            metabolic
ising         strategy         for         cancer         medicine         (Ashworth         et         al.,         2011).         If         a         spe-                                     network          models.          Folger          et          al.          (2011)          proposed          to          characterize          SLs
cific               gene               is               found               to               be               inactivated               in               tumor               cells,               drugs               thatby           modeling           effects           of           the           single-           and           double-knockouts           of           candi-
suppress          its         SL          partner         gene         can         cause          tumor         cells          to         die         but          spare                          date       genes       in       those       networks.       The       second       class,       referred       to       as       know-
normal           cells           (Hartwell           et            al.,            1997).           Hence,            SL           is           a           gold           mine           of      ledge-oriented         methods,         is         mostly         conducted         by         feature         engineering
anti-cancer          drug         targets,         and          intensive         efforts         have         been         exerted          to                                                   with       domain-specific       knowledge.       To       predict       SL       pairs,       these       methods
VC             The        Author(s)        2021.        Published        by        Oxford        University        Press.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        i418
This         is         an         Open         Access         article         distributed         under         the         terms         of         the         Creative         Commons         Attribution         License         (http://creativecommons.org/licenses/by/4.0/),         which         permits         unre-
stricted        reuse,       distribution,        and        reproduction        in        any        medium,       provided        the        original        work        is        properly        cited.

KG      for       SL       production                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          i419
employ       network        topology        features        such        as        graph        centrality        (Kranthi                                                                                                    improve          the         performance          of          SL          prediction,         we          need          a         new          method
et         al.,         2013),         network         flow        (Zhang         et         al.,         2015),         connectivity         hom-                                                                           capable      of      automatic      knowledge      integration       and      feature      extraction.
ology      (Jacunski      et       al.,       2015)       and       features       derived      from       genomic      data                                                                                                           Knowledge            graphs            (KGs)            are            a            type            of            multi-relational            graph,
including                  somatic                  copy                  number                  alteration                  (Jerby-Arnon                  et                  al.,                                         where           nodes            and            edges            have            different            types.            A            KG            is            denoted            by
2014),          short          hairpin          RNA          profiles          (Jerby-Arnon          et          al.,          2014),          and                                                                           G ¼ðV  ;    EÞ,         where         edges         in         set         E         are         defined         as         triplets         e ¼ðh;    s;    tÞ
gene           expression           profiles           (Jerby-Arnon            et            al.,            2014).            However,            the                                                                       indicating                    a                    particular                    relationship                    s  2       T                      between                    two                    nodes
two        classes        mentioned        above        rely        heavily        on        the        metabolic        network                                                                                             (Hamilton,      2020).      By      incorporating      a      KG      into      a      GNN,      one      can      miti-
models,      domain      knowledge       and      genomic      data,      without      fully      exploit-                                                                                                                   gate        the         aforementioned         independency         issue         by        directly        introducing
ing       the      valuable      information      of       known      SL       pairs.                                                                                                                                        those       latent       factors       as       nodes       in       the       graph.       Lin       et       al.       (2020)       proposed
          To         exploit         the         exiting         SL         data,         the         third         class         of         methods         apply                                                           an            end-to-end            knowledge            GNN            (KGNN)           and            achieved            good            per-
machine           learning           algorithms,           where           features           are           engineered           based                                                                                       formance      in       drug–drug       interaction       prediction.
on         both         domain        knowledge         and         heuristic         functions.         Paladugu         et         al.                                                                                               Here,          we          propose          a          novel          KGNN-based          method          for          SL          predic-
(2008)           proposed           to           train           a           support           vector           machine           (SVM)           for           SL                                                           tion,         named         KG4SL,         which         utilizes         KG         MP         as         a         back-end.         We         ap-
prediction       where       the        features       were       extracted       from       a       protein–protein                                                                                                         proach       the       independency       issue       by       injecting       various       factors       including
interaction          (PPI)          network.          MetaSL          (Wu          et          al.,          2014)          integrated          17                                                                           biological         processes,         diseases,         compounds         etc.         that         could         be         pertin-
features        and        weighted        outputs        of        10        classifiers        to        predict        SLs.        Aside                                                                                  ent       to       SL,       into       our       KG.       Our       model       comprises       three       parts.       In       the       first
from            the            traditional            machine           learning            methods,            graph            representa-                                                                                 part,           we           derive          a           gene-specific           subgraph           from          the           original          KG           for
tion          learning          approaches          have          been          proposed,          which          mostly          adopt                                                                                      each         gene.         In         the         second         part,         we         conduct         MP         on         the         gene-specific
an        encoder–decoder        paradigm.        In        this        paradigm,        an        encoder        tries        to                                                                                            subgraph,           to           automatically           associate          genes           with          factors          that           could
map            the            nodes            into            a            low-dimensional            embedding,            whereas            a            de-                                                             be          decisive          in         identifying         an         SL          pair.          In          the          third         part,          we          define         a
coder            takes            the            embedding            and            utilizes            it            to            reconstruct            the            node                                              decoder        to        reconstruct        gene–gene        similarity        in        a        supervised         fashion.
similarities        in        the        original        graph        (Hamilton,       2020),        thereby        recover-                                                                                                 To       the       best       of       our       knowledge,       this       is       the       first       framework      to       integrate
ing      missing      links.      This      paradigm      can      be       generalized       to      matrix      factor-                                                                                                    KG        with        GNN        for        SL        prediction.        We        compared        our        model        with        10
ization               (MF)-based               methods               and               graph               neural               network               (GNN)-                                                                 state-of-the-art           methods           for           SL           prediction,           and           our           model           outper-
based        methods.        Those        methods        use        distinct        designs        of        encoders,        but                                                                                            formed      all       the       baselines      in       area       under       ROC       curve       (AUC),       area       under
resemble           each           other           in           the           choices           of           decoders           (mostly           taking           the                                                        precision-recall         curve         (AUPR)         and         F1.         Another         contribution         of         our
form             of             inner             product             predictor             or              its             normalized             variants).             MF                                                 work       is       that       we       studied       the       impact       of       KG,       which       suggests       that       intro-
methods        adopt        a        MF        encoder.        SL2MF        (Liu        et        al.,        2020)         proposed         a                                                                               ducing         a         KG          combined          with         MP          process         in          GNN         can         significantly
MF         encoder         which         decomposes         the         SL         matrix,         gene         ontology         (GO)                                                                                        improve       the       performance       of      SL       prediction.
similarity       matrix,       and        PPI       matrix       to        a        low-dimensional       latent       space.
GRSMF           (Huang           et           al.,           2019)           introduced           a           self-representative           MF                                                                               2        Materials        and        methods
encoder             which             focuses              on             learning             a              representation              matrix             from
known             SL              pairs             and             further              integrates             the             functional             similarities                                                          In           this           section,           we           first           introduce          the           data           and           the           problem           of           SL
among                   genes                   derived                   from                   GO.                   Liany                   et                   al.                   (2020)                   adopted   prediction.             Then,             we             present             the             details             of             the             proposed             KG4SL
Collective          Matrix          Factorization          (CMF)          based          methods          to          integrate                                                                                              model.
data      from       heterogeneous       sources      to       predict      SLs.
          The            MF-based            encoders            are            just            shallow            embedding             methods,                                                                            2.1        Data        Collection
which           simply           optimize           a           unique           embedding           vector           for           each           node,                                                                     SynLethDB                             (http://synlethdb.sist.shanghaitech.edu.cn/v2/#/;                             Guo
without           sharing           any           parameter           between           nodes           or           leveraging           node                                                                               et            al.,            2016)            is            a            comprehensive           database           of            synthetic            lethal           gene
features          (Hamilton,          2020).          GNN,          a          state-of-the-art          framework          for                                                                                              pairs.         Its         latest         version         includes         a         set         of         36,402         human         SL         pairs,         as
deep          learning         on          graphs,         enhances          the          aforementioned          methods         by                                                                                         well        as        a        KG        with        11        kinds        of        entities        and        24        kinds        of        relationships
adopting           a           different           embedding           strategy.           GNN           defines           a           message-                                                                              as       shown       in       Table       1.       SynLethDB       also       includes       negative       SL       pairs,       i.e.
passing           (MP)           process           on           the           original           graph,           i.e.           at           each           iteration,                                                      Non-SL           and            synthetic           rescue            pairs.            However,            there            are           much            less
each            node            aggregates            all            the            embeddings            from            its            local            neighbor-                                                          known       negative       SL       pairs       than       known       positive       ones.       To       obtain       a       bal-
hood         as         a         message         which         is         combined         with         its         previous         embedding                                                                              ance         between         the         positive         and         negative         samples,         we         randomly         pick
to            generate            a            new            embedding.            Based            on            GNN,            Cai            et            al.            (2020)                                        up        unknown        pairs        as        negative        pairs        so        that        there        are        equal        numbers
adopted        a        novel        regularization        technique        called        dual        dropout        to        ad-                                                                                           of          positive          and          negative          SL          pairs.          Hence,          the          final          SL          dataset          con-
dress      the       sparsity      of       SL       networks.                                                                                                                                                               tains      72      804       gene      pairs      between      10       004      genes.
          However,      the       existing      GNN-based       methods      often       regard      each      SL                                                                                                                      The          KG,          denoted          as          SynLethKG,          includes          24          kinds          of          relation-
pair          as          an          independent          sample,          and          make          no          attempt          to          take          their                                                          ships           between          11           entities.           Among          24           kinds          of           relationships,           16          of
underlying                biological                mechanisms                into                account.                However,                some                                                                       them       are       related       to        genes       directly,       e.g.       (gene,        regulates,       gene),       (gene,
shared       factors       (such       as       biological       processes,       pathways,       cellular       com-                                                                                                        interacts,       gene)       and       (gene,       co-varies,       gene).       And       the       other       8       relation-
ponents         etc.)         might         latently         invalidate         the         assumption         of         independ-                                                                                          ships        are        associated        with        drug        and        compounds.        Besides,        7        out        of        11
ency.         For         instance,        poly        (adenosine        diphosphate-ribose)         polymerase                                                                                                              kinds         of          entities         are          directly          related          to          genes,         i.e.          pathway,         cellular
1        (PARP1)       and        breast        cancer        1        (BRCA1)        are        a        famous       SL        gene        pair,                                                                           component,             biological             process,             molecular             function,             disease,             com-
leading        to         the         first         clinically        approved         SL-based        cancer         drug,        PARP                                                                                      pound         and         anatomy.         They         are         in         the         format         of         (gene,         relationship,
inhibitor        (Lord        and        Ashworth,        2017).        PARP1        and        BRCA1        are        both                                                                                                 entity).       These        entities        can       be        reached       from        genes        in       one        hop,        where-
key       players       in       DNA        repair       process.       Meanwhile,        ATM       and        TP53       are                                                                                                as         the         other         three         kinds         of         entities         (pharmacologic         class,         side         effect
another      widely       known      SL       pair       (Kwok       et       al.,       2016),       and       ATM      is       also
a       key       instrument       in       DNA       repair       process       (Sanders      et       al.,       2020).       Here,                                                                                        Table      1.      Details       about      the       SL       data       and      KG       SynLethKG
the         DNA         repair         process        might         be         the         common        mechanism        underly-
ing       the      two       SL       pairs.                                                                                                                                                                                 SL      data                                                                                                                                                                                                    No.      of      genes                                                                                                                                                                                                                                                                                                 10      004
          A          subset          of          existing          methods          [e.g.          SVM,          random          forests          (RFs),
SL2MF            and            GRSMF]            have            injected            some            genomic            and            proteomic                                                                                                                                                 No.      of      interactions                                                                                                                                                                                                                        72      804
data          to          facilitate          the          SL          prediction,          and          the          results          of          these          studies                                                                                                                         Density                                                                                                                                                                                                                                                                                                                                                         0.14%
have        underscored        the        significance       of        integrating        additional        informa-                                                                                                         SynLethKG                                                                                                                                                     No.      of      entity      types                                                                                                                                                                                                                            11
tion.         Meanwhile,         GNN-based         methods         can         also         encode         such         infor-                                                                                                                                                                    No.      of      relationship      types                                                                                                                                                     24
mation      as       input      features.       However,       these       methods      extracted      features                                                                                                                                                                                   No.      of      nodes                                                                                                                                                                                                                                                                                             54      012
manually         based         on         domain         knowledge          and         some         features         might         be                                                                                                                                                            No.      of      edges                                                                                                                                                                                                                                                                                                 2      231      921
left       out.       Therefore,       to       attain       a       more       comprehensive       set       of       features       to

i420                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      S.Wang      et       al.
Table      2.      Details       about      the       entities      in       SynLethKG                                                                                                                                                                                                                                                                           Table      3.      Details       about      the       relationships       in      SynLethKG
Type                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Sample      sizeType                                                                                                                                                                                                                                                                                                                                                                                                                                                                               No.      ofNo.      ofNo.      of
Cellular      component                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             1670                       edges                                           source                                     target
Gene                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           67      062nodesnodes
Biological      process                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         12      703(Anatomy,      downregulates,      gene)                                                                                                                                                    31                                                                                             4                                                                                             24
Side      effect                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            5726(Anatomy,      expresses,      gene)                                                                                                                                                                                                                 617      175                            241                                                                23      881
Molecular      function                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   3203(Anatomy,      upregulates,      gene)                                                                                                                                                                                        26                                                                                             5                                                                                             22
Pathway                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              2069(Compound,      binds,      gene)                                                                                                                                                                                                                                      16      323                                          1922                                                 2306
Disease                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  137(Compound,      causes,      side      effect)                                                                                                                                                              139      428                            1079                                                 5702
Compound                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               2595(Compound,      downregulates,      gene)                                                                                                                             21      526                                          747                                                                2847
Pharmacologic      class                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  377(Compound,      palliates,      disease)                                                                                                                                                                         384                                                                              215                                                                50
Anatomy                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              402(Compound,      resembles,      compound)                                                                                                              6266                                                                1034                                                 1055
Symptom                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           453(Compound,      treats,      disease)                                                                                                                                                                                                         752                                                                              385                                                                77
                                                                                                                                                                                                                                                                                                                                                                 (Compound,      upregulates,      gene)                                                                                                                                                                 19      200                                          721                                                                3205
                                                                                                                                                                                                                                                                                                                                                                 (Disease,      associates,      gene)                                                                                                                                                                                                                               24      328                                          135                                                                6572
and       symptom)       can       be       reached       from       genes       in       two       hops.       After       remov-                                                                                                                                                                                                                               (Disease,      downregulates,      gene)                                                                                                                                                                         7616                                                                44                                                                              5730
ing          isolated          nodes,          the          final          graph          of           SynLethKG           contains           54          012                                                                                                                                                                                                    (Disease,      localizes,      anatomy)                                                                                                                                                                                               3373                                                                123                                                                398
nodes           and           2           231           921           edges           as           shown           in           Table           1.           Tables           2           and           3                                                                                                                                                        (Disease,      presents,      symptom)                                                                                                                                                                                            3401                                                                122                                                                427
show        the        details        about        the        entities        and        relationships         in        SynLethKG.                                                                                                                                                                                                                              (Disease,      resembles,      disease)                                                                                                                                                                                                     404                                                                              100                                                                98
Users       can       access       the        SynLethKG        through       searching       the       names       of        the                                                                                                                                                                                                                                 (Disease,      upregulates,      gene)                                                                                                                                                                                                            7730                                                                44                                                                              5614
genes      that      they      want       to      study      in       SynLethDB.                                                                                                                                                                                                                                                                                 (Gene,      covaries,      gene)                                                                                                                                                                                                                                                                             62      987                                          9174                                                 9706
                The           24           relationships           in           Table           3           describe           the           features           of           genes,                                                                                                                                                                              (Gene,      interacts,      gene)                                                                                                                                                                                                                                                                         148      379                            9633                                                 14      275
drugs                   and                   compounds.                   These                   relationships                   are                   collected                   from                                                                                                                                                                        (Gene,      participates,      biological      process)                                                                             619      712                            16      608                            12      703
Genbank,                   GO,                   Drugbank,                   DrugCental,                   PubMed,                   Bgee,                   String,                                                                                                                                                                                             (Gene,      participates,      cellular      component)                                                             97      652                                          11      916                            1670
LINCS             L1000,              SIDER4,             STARGEO,              Uberon              and              BioGRID.              The                                                                                                                                                                                                                   (Gene,      participates,      molecular      function)                                                                110      042                            14      404                            3203
specific       number       of       each       type       of       relationship       and       the       number       of       asso-                                                                                                                                                                                                                           (Gene,      participates,      pathway)                                                                                                                                                                                      57      441                                          11      519                            2069
ciated      nodes      are      also      shown      in      Table      3.      Besides,       the      types      of      the      enti-                                                                                                                                                                                                                        (Gene,      regulates,      gene)                                                                                                                                                                                                                                                                    267      791                            4649                                                 7105
ties       in       SynLethKG       and       the       number      of       each       entity       are       shown       in       Table                                                                                                                                                                                                                        (Pharmacologic      class,      includes,      compound)                           1205                                                                377                                                                837
2.
2.2        Problem        statement                                                                                                                                                                                                                                                                                                                                               Assume           that           e           is           the           central           node/entity           and           N(e)           is           the           set           of
Formally,             the             SL             data             can             be             modeled             as             a             matrix             S 2ð0;    1Þn/C2  n ,                                                                                                                                                                   neighbors         of         e         (i.e.         entities         directly         connected         to         e).         In         SynLethKG,
where         n         is         the         number         of         genes         involved         in         the         SL         pairs.         In         this         SL                                                                                                                                                                              the        size         of        N(e)         varies        greatly        among        the         entities.         For        example,         net-
matrix        S,        an        entry        si;j          is        1        if        there        is        an        SL        interaction        between        gene                                                                                                                                                                                      work           hubs           may           have           thousands           of           relations,           whereas           some           nodes
ei      and      gene      ej,      and      0       otherwise.      Note      that      gene      pairs      with      entry      value                                                                                                                                                                                                                         are       less       studied       and       thus       have       a       limited       number       of       neighbors.       In       this
0       are       unknown       pairs,       some       of       which       could       be       potential       SL       pairs       not                                                                                                                                                                                                                       work,          we          sample          a          fixed          number          of          k          neighbors          for          each          entity          to
yet              discovered.              The              KG              SynLethKG              is              denoted              by              G ¼ðVe  ;    VrÞ,                                                                                                                                                                                         characterize       its       local       structure       and       we       repeat       this       process       for       H       hops
which       contains       a       set       of       entities       Ve       and       a       set       of       relationships       Vr.       Each                                                                                                                                                                                                            (H  /C21           1).      In      particular,      if      a      node      has      less      than      k      neighbors,      we      sample
edge       in       the       KG       is       defined       as       a       triplet       T  ¼ðh;    r;    tÞ,        which       shows       a       re-                                                                                                                                                                                                     duplicates,        i.e.        a        neighbor        may        be        sampled        more        than        once.        The        set
lationship          of           type          r           between          head           entity          h          and           tail          entity          t,          where                                                                                                                                                                              of       sampled      k      neighbors      is       denoted      as      P(e).      An       example      of       2-hop      sub-
h;    t  2       Ve        and      r  2       Vr .                                                                                                                                                                                                                                                                                                              graph        with        neighbor         sampling        size         k ¼    4         in        each        hop        can        be        seen         in
                Given           the           SL           matrix           S           and           the           KG           G,           the           problem           we           aim           to                                                                                                                                                      Figure      1.
solve          is          to          predict          the          SL          relationship          between          gene          ei          and          gene          ej.                                                                                                                                                                                                  In          a         gene-specific         subgraph,         we          can         assign         different         weights         for
To           achieve           this           goal,           we            propose           a           GNN-based           model           to            learn            a                                                                                                                                                                                   edges        to        describe        the        importance        of        the        relations.        For        an        SL        pair        (ei,
scoring          function            ^s i;j  ¼       Fði;    jjW;    A;    bÞ          that          estimates          how          likely          gene                                                                                                                                                                                                        ej),              the              weight              for              an              edge              re;e0                 in              ei’s              subgraph              is              computed             by
ei        and        gene        ej        is        an        SL        pair,        where        W,        A        and        b        denote        the        learnable                                                                                                                                                                                     xeie;e0  ¼       gðej  ;    re;e0Þ,        where        e        is        one        of        the        entities        in        the        subgraph        of        ei,
parameters       in       function      F.                                                                                                                                                                                                                                                                                                                       and       e0  2       PðeÞ.       Besides,       ej         and       re;e0           are       the       feature       embeddings       of       gene
                                                                                                                                                                                                                                                                                                                                                                 ej              and              relation              re;e0   ,              respectively.              g              is              an              inner             product              function.
2.3        Overview        of        KG4SL                                                                                                                                                                                                                                                                                                                       Here,      xeie;e0        characterizes      the       importance       of       relation      re;e0          to       gene      ej.
The       overall       framework        of       KG4SL       is       laid       out       in        Table       1.        KG4SL       uti-
lizes        a        GNN        to        encode        the        gene        features        from        KG        for        SL        prediction                                                                                                                                                                                                            2.3.2                                 Aggregation       of      node       representations
in       three       steps.       First,       we       derive       a       gene-specific       weighted       subgraph       for                                                                                                                                                                                                                               For        any       central       entitye       in       the       subgraph       of       geneei,       we       aggregate       the       rep-
each         SL-related         gene         from         the         KG.         Specifically,         the         weight         of         every                                                                                                                                                                                                              resentations        of       all        its        picked        neighbors        to        update        its       own        representation.
edge        is        defined        by        a        gene-specific        relation        scoring        function        to        depict                                                                                                                                                                                                                     To         show         the         topological         neighborhood         structure         of         entitye         in         the         KG,
the        importance        of        the        relation        for        its        target        gene.        Second,        we        design                                                                                                                                                                                                               we        compute        the        weighted average       combination       ofe’s       neighborhood:
an      aggregation       layer       to       update      the       representation      for       a       given      gene      by
aggregating         the         representations         of         its         neighbors         in         the         gene-specific                                                                                                                                                                                                                                                                                                                                                 ePðeÞ  ¼           X                                           x ei~   e;e0 e0  ;                                                                                                                                                                                                                                                                       (1)
weighted      subgraph.       Third,       we       assign       a       score       for       each       gene       pair       com-                                                                                                                                                                                                                                                                                                                                                                                     e02PðeÞ
puted       by       the        normalized       inner       product       based       on       their       learned       repre-                                                                                                                                                                                                                                 where          e          is          the          representation           of          entity          e,          gene          ei           and          gene          ej          are          a
sentations.       Next,      we       introduce      these      three      steps      in       details.                                                                                                                                                                                                                                                          pair       in       the       SL       matrix,       and            ~x         is       the       normalized      gene-relation       score       by
                                                                                                                                                                                                                                                                                                                                                                 applying      a       softmax      function:
2.3.1                                 Gene-specific       weighted      subgraph
Given           an            SL-related           gene,           we           first           construct           a           weighted           subgraph                                                                                                                                                                                                                                                                                                            x ei~                            P                     exp ðxeie;e/C18 Þ
from          the          KG.          Identifying          relevant          nodes          and          determining          the          edge                                                                                                                                                                                                                                                                                                                               e;e/C18    ¼                        e       /C182PðeÞ           exp ðxei^                       e;^e/C18 Þ      :                                                                                                                                                                                                                               (2)
weights                  are                  two                  key                  operations                   to                  construct                  the                  gene-specific
weighted      subgraph.

KG      for       SL       production                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          i421
Fig.       1.       The       framework       of       KG4SL.       The       workﬂow       of       KG4SL       can       be       divided       into       three       modules,       including       gene-speciﬁc       weighted       subgraph       module,       aggregation       module       and
score       computation       module.       (1)       Gene-speciﬁc       weighted       subgraph:       First,       we       construct       a       weighted       subgraph       from       the       KG.       (2)       Aggregation:       Second,       for       each       SL       pair,       we       select
the      entities      and      relationships      that      are      directly      related      to      the      nodes.      Besides,      we      believe      the      biological      information      can      ﬂow      between      nodes      through      edges.      Thus,      we      also      aggre-
gate     the     information     of     indirectly     connected     entities     and     relationships.     Considering     the     problem     of     computing     power,     only      two     layers     of     entities     and     relationships     are     included.
(3)      Score      computation:      Third,      the      results      of      aggregation      for      two      genes      are      used      to      compute      their      SL      score      through      inner      product.      The      loss      function      of      KG4SL      is      composed      of
two       kinds       of       losses,       i.e.       the       base       loss       computed       based       on       the       truth       label       and       the       gene–gene       score,       and       the       L2       loss       computed       using       the       entity       embedding,       relation       embedding
and     aggregation     weights
           After            obtaining            the            picked            neighbors’            representation            ePðeÞ              of            a                                                                                                            J        ¼       maxð^s i;j  ;    0Þ/C0                        ^s i;j  /C2         si;j  þ      logð1 þ      expð/C0j                ^s i;jjÞÞ;                                                                       (4)
central        entity        in        one        hop,        similar        to        (Wang        et        al.,        2019a),        it        integra-                                                                                        where        ^s i;j         is       the       predicted       label       and       si;j         is       the       truth       label       for       the       edge.
tes      the       entity       representation       e       into      a       single      vector       to       update      e:                                                                                                                    We       also      add      an      L2-regularizer      defined      as:
                                                                                               /C16        /C16                                    /C17              /C17
                                                      e½h þ      1/C138¼               /           W            e½h/C138þ             ePðeÞ              þ      b         ;                                                                                                                                                                     (3)jjCjj  ¼jjejj þ jjrjj þ jjWjj
where               W               and               b               are               the               linear               transformation               weight               and               bias,                                                                                                                                                                    2                                                                                                                                       ;                                                                                                                                                                                                                                 (5)
respectively,                    and                   /                    is                    an                    activation                    function                    such                    as                    ReLU.              where   jj /C1 jj                   represents           the           L2           norm           for           entity           embedding,           relation
After         aggregating         neighbors’        information         through        H         hops,         the         final                                                                                                                   embedding       and      aggregation       weights.
feature         representation         of         gene           ^e i           is         e½H/C138          :                ^e j           is         obtained         in         the         same                                                           The       final       loss       combines       the       two       kinds       of       loss       functions       described
way.                                                                                                                                                                                                                                               above      as       follows:
                                                                                                                                                                                                                                                                                                                                                                                        X
                                                                                                                                                                                                                                                                                                    minW;        A;        b        ‘     ¼       minW;        A;        b                          J þ      ajjCjj;                                                                                                                                                  (6)
                                                                                                                                                                                                                                                                                                                                                                                           i;j
2.3.3                                 SL       prediction      score                                                                                                                                                                               where       A       is       the       trainable       weighted       matrix       in       which       each       element       rep-
Finally,         by         passing         information         from         two         subgraphs         of          KG,         we          at-                                                                                                 resents       the       gene-relation       score       and       L2       weight       a       is       a       balancing       hyper-
tain          the          final          representations            ^e i            and            ^e j            for          the          two          genes          in          the          SL                                              parameter.         Here        a         was         set         to        0.0039.        The         first         term         corresponds         to
matrix.           The           predicted           interaction           probability           between           gene           ei           and/C16/C17                                                                                          the           part           of           GNN           that           learns           the           linear           transformation           weight           W,
gene        ej        is        calculated        by          ^s i;j  ¼       /                            f ð^e i  ;      ^e jÞ        ,        where        f        is        the        inner        prod-                                     gene-relation        score        weight        A        and        bias        b        simultaneously.        The        second
uct        function        and        /         is         a        sigmoid        function,        squashing        the         output        to         a                                                                                        term        added        the        L2-regularizer.        Adam        algorithm        is        used        to        minimize
range            between            0            and            1.             Furthermore,             this            link            prediction            can             be                                                                   the       final       loss       and       the       learning       rate       is       set       to       0.002.       The       framework       of
viewed       as       a       binary       classification       problem,       by       setting       the       threshold       to                                                                                                                 KG4SL      is       outlined      in       Algorithim       1.
0.5.           ^s i;j           is         1         or         0,         which         indicates         whether         an         SL         relation         exists         be-
tween      a       candidate      pair       of      genes.
                                                                                                                                                                                                                                                   3        Results
                                                                                                                                                                                                                                                   In        this        section,        we        first         introduce        the        state-of-the-art        baseline        meth-
2.4        Overall        loss        and        optimization                                                                                                                                                                                      ods             and              their             implementation              details,             and             then              we              compare              our
Two        kinds        of        losses        are        designed        for        our        model,        including        base        loss                                                                                                   model       with       the       baselines,       followed       by       an       analysis       of       the       influence       of
and          L2          loss.          The          base          loss          J          is          computed          through         cross-entropy         of                                                                                 the         KG.         The         KG4SL         model         was         implemented         with         Python         3.6         and
the         truth         label         and         the         predicted         label         for         the         edges,         represented         as                                                                                      Tensorflow        1.15.0.       We       adopt       AUC,       AUPR       and        F1       as        the       evaluation
follows.                                                                                                                                                                                                                                           metrics.

i422                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      S.Wang      et       al.
                                                                                                                                                                                                                                          •                                  DDGCN        (Cai        et        al.,        2020)        is        built        to        adapt        to        the        sparsity        of        SL
        Algorithm         1         KG4SL                                                                                                                                                                                                            network,      which      includes      a      dual       form      drop      out.
        Input:         SL         matrix         S;KG         GðVe  ;    VrÞ;         neighborhood         ﬁeld                                                                                                                                       These          baselines          can          be          divided          into          three          categories,          MF-based
               P(e);         hyper-parameters         a,d,         k,         h         and         epoch;                                                                                                                                baselines                      (SL2MF,                      GRSMF                      and                      HOPE),                      RW-based                      baselines
        Output:           ^s i;j                                                                                                                                                                                                          (DeepWalk,                     Node2Vec                     and                     LINE)                     and                     GNN-based                     baselines
                                                                                                                                                                                                                                          (GraphSAGE,        GAT,        GCN        and        DDGCN).       We        evaluated        DeepWalk,
        1:      Initialization:                                                                                                                                                                                                           Node2Vec,       LINE       using       CogDL       (THUDM,       2020),       GraphSage,       GCN
        2:      entity       embedding       matrix      We;                                                                                                                                                                              and       GAT       via        graph       convolution       layer        in       DGL       (Wang       et        al.,       2019b).
        3:      relation      embedding      matrix      Wr;                                                                                                                                                                              Note         that         all         the          baselines         do          not         utilize         KG          as         inputs,          so          they         are
        4:      step         0;                                                                                                                                                                                                           only             trained             on             the             SL             interaction             matrix.             In             addition             to             the             SL
                                                                                                                                                                                                                                          interaction            matrix,            SL2MF            and            GRSMF            also            utilize            GO            semantic
        5:      while      step         <       epoch      do                                                                                                                                                                             similarity      matrices      for       SL       prediction.
        6:                                                               for ðei  ;    ejÞ2       S       do
        7:                                                                                            GSi½h/C138       Hh¼0         Gene-speciﬁc      Subgraph(ei);                                                                       3.1.2                                 Implementation      details
        8:                                                                                            GSj½h/C138       Hh¼0         Gene-speciﬁc      Subgraph(ej);                                                                       All            the             baselines            were             evaluated            at             5-fold             cross            validation            which
        9:                                                                                            for       m 2½i;    j/C138              do                                                                                          makes       the        best       use        of       the        available       data.       We        viewed       the       input        graph
        10:                                                                                                                        e½0/C138                GSm½0/C138       ;                                                             as          an           unweighted          and          undirected          graph.          For          SL2MF          and          GRSMF,
        11:                                                                                                                        for       h ¼    1,2,...,H       do                                                                    we               utilized               all               the               default               parameters               in               the               origin               papers.               For
        12:                                                                                                                                                     for       e  2       GSm½h/C138              do                           HOPE,       the       parameter       beta       was       set       to       0.02.       For       RW-based       methods,
        13:                                                                                                                                                                                 ePðeÞ½h /C0                    1/C138          Px emthe       number       of       walks       to       start       at       each       node       was       set       to       5,       the       length       of
                                                                                                                       e02PðeÞ           ~e;e0 e0 ;                                                                                       the       RW       start       at       each       node       was       set       to       10,       the       window       size       was       set       to
        14:                                                                                                                                                                                 e½h/C138                /ðWðe½h /C0                    1/C138þ              ePðeÞ½h /C0                    1/C138Þ þ              bÞ;3.       For       node2vec,       p       and       q       were       both       set       to       1,       which       respectively       con-
        15:                                                                                                                                                     end       for                                                             trol           how           fast           the           walk           explores           and           how           fast           the           walk           leaves           the
        16:                                                                                                                        end       for                                                                                          neighborhood          of          starting.          For          LINE,          the          alpha          was          set           to          0.1          and
        17:                                                                                                                                                  ^e m          e½H/C138       ;                                               other        was        set        to        2.        For        DDGCN,        we        use        the        same        setting        of        the        ori-
        18:                                                                                            end      for                                                                                                                       gin      paper.      For       GCN,      GAT      and      GraphSage,      we       all       used      two       convolu-
        19:                                                                                            Compute      the       predicted      probability                                                                                  tion           layers,          and          the           dimensionality           of          the           latent          spaces           in          the          first
                                          ^s i;j          /ðf ð^e i  ;     ^e jÞÞ;                                                                                                                                                        and       second       layers       were       chosen       to       be       5       and       16.       The       number       of       train-
        20:                                                                                            Compute      the       loss         ‘           ;                                                                                  ing        epochs        was        decided        via        the        early        stopping        strategy.        KG4SL        was
                                                                                                                                                                                                                                          evaluated         using         the         ratio         of         training,         validation         and         testing         data         as
        21:                                                               end       for                                                                                                                                                   8:1:1.        In        order        to        improve        the        stability        of        results,        we        randomly        split
        22:                                                               step         step þ       1;                                                                                                                                    dataset        into        10        pieces        and        took        one        of        them        as        the        testing        set.        Early
        23:                                   end       while                                                                                                                                                                             stopping      strategy      is       used       to       control      the       number      of       training      epochs.
                                                                                                                                                                                                                                          3.1.3                                 Comparison      with       baselines
3.1        Performance        evaluation                                                                                                                                                                                                  On       SynLethKG,       KG4SL      outperforms       all       baselines       as       shown       in       Table
3.1.1                                 Baselines                                                                                                                                                                                           4.            When            compared            with            the            second            best            model            GRSMF,            KG4SL
We       compare      KG4SL      with      the       following       baselines:                                                                                                                                                           improves                 the                performance                on                AUC,                 AUPR                and                F1                by                3.11%,
                                                                                                                                                                                                                                          2.16%      and      6.4%,       respectively.
•                                  SL2MF            (Liu            et            al.,            2020)            integrates            gene            similarities            based            on                                                  In         general,        GNN-based        models         achieve         better         performance         than
                                                                                                                                                                                                                                          shallow         embedding         methods         like         MF-based         and         RW-based         models.
          GO       biological      pathway      annotations      with       SL       matrix       to       predict       SL                                                                                                               This        may        because        GNN-based        models        can        learn        from        the        similarity
          pairs.                                                                                                                                                                                                                          between          SLs          and          enrich          the          embedding          of          genes          for          SL          prediction.
•                                  GRSMF      (Huang      et       al.,       2019)       is       a       graph       regularized      self-represen-                                                                                    DDGCN         represents         the         state-of-art         model         for         SL         prediction,         and         it
          tative        MF        model        which        also        uses        known        SL        pairs        and        GO-based                                                                                               achieves              the              best              performance              among              the              GNN-based              baselines.
          gene      similarities      to      predict       SL       pairs.                                                                                                                                                               MF-based         method         GRSMF         is         second         only         to         KG4SL,         which         shows
•                                  HOPE         (Ou         et         al.,         2016)         is         scalable         to         preserve         high-order        prox-                                                         that           the           combination           of           GO           gene           similarity           information            and           self-
                                                                                                                                                                                                                                          representation        matrix        decomposition        is        very        effective        for        SL        predic-
          imity                    of                    graphs                    and                    capable                    of                    capturing                    the                    asymmetric                 tion.         The          performance          of         KG4SL         is          even          higher,         which         shows         that
          transitivity.                                                                                                                                                                                                                   learning          gene          representations          from          the          KG          including          GO          informa-
•                                  DeepWalk           (Perozzi           et           al.,           2014)           is           a           graph           embedding           method                                                  tion      and      other      gene       features      can      further      improve      SL       prediction.
          which          uses          short          random          walks          (RWs)          to          learn          representations
          for       nodes       in      graphs.                                                                                                                                                                                           3.2        Model        Analysis
•                                  Node2Vec         (Grover         and         Leskovec,         2016)         also         learns        feature         rep-                                                                           3.2.1                                 Parameter      sensitivity
          resentations          for          node         in          graphs         but          adds          ﬂexibility         in          exploring                                                                                  We        present        the        sensitivity        analysis        for        some        key        hyperparameters        in
          neighborhoods.                                                                                                                                                                                                                  our         KG4SL,         including         the         neighbor         sampling         size         k         and         the         dimen-
•                                  LINE       (Tang       et       al.,       2015)       uses       an       effective       edge-sampling       method                                                                                  sion      of       entity       embedding       d,       as       shown      in       Figure      2.
          for             model             inference             and             preserves             both             the             ﬁrst-             and             second-                                                                    First,         we         change         the         number         of         samples         for         neighbor         k         and         ob-
                                                                                                                                                                                                                                          serve      the       model       performance.       KG4SL      achieves       the       best       AUC,      F1       and
          order      proximities      by      a      ﬁne-grained      objective       function.                                                                                                                                           AUPR              when              the              neighbor               sampling              size              k ¼    64.              When              sampling
•                                  Convolutional          network          (GCN)          (Kipf          and          Welling,          2016)          is           the                                                                   more             neighbors             with             higher             value             of             k,             the             information            sampled
          most         popular         GNN         architecture,         which         employs         the         symmetric-                                                                                                             may            become            redundant,            and            thus            the            model            performance            slightly
          normalized       aggregation      as       well      as       the       self-loop      update      approach.                                                                                                                    decreases         when         k         is         128.         Next,         we         also         investigate         the         influence         of
•                                  GraphSAGE      (Hamilton       et      al.,      2017)      introduces      the      idea      of      gener-                                                                                          the            dimension            of            embedding            d.            The            KG4SL            model            already            has            a
          alized      neighborhood       aggregation.                                                                                                                                                                                     good           performance           when           the           dimension           of           embedding           is           256.           Too
•                                  GAT           (Veli/C20ckovi/C19c           et           al.,           2017)           introduces           the           attention           mecha-                                                  large       dimension       of       embedding       is       a       burden       on       memory       and       computa-
                                                                                                                                                                                                                                          tion.       Eventually,        we        set       the       neighbor       sampling       size       as        64       and       the        di-
          nisms      to       GNN.                                                                                                                                                                                                        mension      of       embedding       as       256      for       our       KG4SL      model.

KG      for       SL       production                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          i423
Table      4.      Metrics      of       KG4SL       against       baselines       in       AUC,       AUPR      and      F1
Categories                                                                                                                                                                                                                                                 Methods                                                                                                                                                                                                                                                                                                                     AUC                                                                                                                                                                                                                                                                                                                                                             AUPR                                                                                                                                                                                                                                                                                                                                                                           F1
MF                                                                                                                                                                                                                                                                                                            SL2MF                                                                                                                                                                                                                                                                                          0.7811   6    0.0035                                                                                                                                                                                                                            0.8635   6    0.0021                                                                                                                                                                                                                            0.7446   6    0.0074
                                                                                                                                        GRSMF                                                                                                                                                                                                                                                                             0.9184   6    0.0039                                                                                                                                                                                                                            0.9362   6    0.0023                                                                                                                                                                                                                            0.8339   6    0.0049
                                                                                                                                        HOPE                                                                                                                                                                                                                                                                                                0.7776   6    0.0005                                                                                                                                                                                                                            0.7410   6    0.0006                                                                                                                                                                                                                            0.7089   6    0.0010
RW                                                                                                                                                                                                                                                                                                        DeepWalk                                                                                                                                                                                                                                                  0.8451   6    0.0024                                                                                                                                                                                                                            0.8600   6    0.0013                                                                                                                                                                                                                            0.7562   6    0.0027
                                                                                                                                        node2vec                                                                                                                                                                                                                                                                0.8362   6    0.0010                                                                                                                                                                                                                            0.8523   6    0.0014                                                                                                                                                                                                                            0.7503   6    0.0031
                                                                                                                                        LINE                                                                                                                                                                                                                                                                                                            0.8233   6    0.0028                                                                                                                                                                                                                            0.8327   6    0.0023                                                                                                                                                                                                                            0.7380   6    0.0056
GNN                                                                                                                                                                                                                                                                                     GCN                                                                                                                                                                                                                                                                                                              0.8329   6    0.0172                                                                                                                                                                                                                            0.8727   6    0.0110                                                                                                                                                                                                                            0.8508   6    0.0136
                                                                                                                                        GraphSAGE                                                                                                                                                                                                                            0.8398   6    0.0291                                                                                                                                                                                                                            0.8775   6    0.0188                                                                                                                                                                                                                            0.8569   6    0.0236
                                                                                                                                        GAT                                                                                                                                                                                                                                                                                                                  0.7914   6    0.0182                                                                                                                                                                                                                            0.8462   6    0.0103                                                                                                                                                                                                                            0.8152   6    0.0129
                                                                                                                                        DDGCN                                                                                                                                                                                                                                                                    0.8491   6    0.0106                                                                                                                                                                                                                            0.8998   6    0.0056                                                                                                                                                                                                                            0.8154   6    0.0074
                                                                                                                                        KG4SL                                                                                                                                                                                                                                                                                         0.9470   6    0.0003                                                                                                                                                                                                                             0.9564   6    0.0005                                                                                                                                                                                                                             0.8877   6    0.0017
Note:      The      best      results      for      each      index      are      in      bold.
Fig.       2.       Parameter       sensitivity       analysis       with        varying       k        and       d        for        KG4SL.       Left:        AUC,
AUPR,      and      F1      at      different      neighbor      sampling      sizes      k,      ranging      from      2      to      128.      Right:
AUC,       AUPR       and       F1       at       different       node       embedding       dimensions       d,       ranging       from       8       to
512
3.2.2                                 Convergence       analysis
With          the          above          parameters          set,          we          observe          the          convergence          of          the
model.          Figure          3          displays          the          change          of          loss          and          three          metrics          with
the        increase        of        epochs.       The        blue-dotted,       red-dashed       and        green        dash-
dot         lines         represent         the         metrics         of         training         data,         validation         data         and
testing        data,        respectively.        The        orange        line        shows        the        change        of        loss.
We        find        that        loss         falls        rapidly        within        the         first        10        epochs        and         begins                                                                                                                                                                                                         Fig.      3.      Convergence      analysis      of      KG4SL.      In      the      ﬁrst      subgraph,      the      orange      line      repre-
to       converge       gradually       at       the       20th       epoch.       Under       the       constraint       of       the                                                                                                                                                                                                                               sents       the      change       of      training       loss      during       the      training       process.      In       the      other       three       sub-
L2-regularizer,        loss        converges        to        0.3111        and        the        results        of        the        three                                                                                                                                                                                                                          graphs,      the      blue-dotted      line      represents      the      variation      trend      of      three      metrics      (AUC,      F1
metrics       in       the       training       set,       validation       set       and       test       set       have       the       similar                                                                                                                                                                                                                    and       AUPR)       during       the       training        process.       The        red-dashed       line        displays       the        values       of
variation       trend       which       shows       that       the        proposed        method       can       alleviate                                                                                                                                                                                                                                           the        above        metrics        for        validation        set        as        the        increase        of        epochs.        The        green        dash-dot
the       problem      of       overfitting.                                                                                                                                                                                                                                                                                                                         line     denotes     the     results     for     test     set.     At     the     ﬁrst     10     epochs,     the     training     loss     drops     rap-
                                                                                                                                                                                                                                                                                                                                                                     idly      and      the      three      metrics      for      training      set,      validation      set      and      test      set      increase      with      a
                                                                                                                                                                                                                                                                                                                                                                     uniform     trend.      Although      at      the     20th      epoch,      the      training      loss      and      three      metrics      begin
3.3        Impact        of        KG                                                                                                                                                                                                                                                                                                                                to     converge
While          automatically          integrating          the          KG          into          the          node          feature          con-
struction       is       crucial       in       our       work,       we       wonder       whether       the       KG       is       really
important       for       the       task       of       SL       prediction.       To       investigate       this       problem,                                                                                                                                                                                                                                    Table      5.      Impact      of      KG       analysis      on      AUC       and      AUPR
we                   test                  the                   SL                  prediction                  performance                   with                   and                  without                  the                                                                                                                                                                                                                                                   Method                                                                                                                                             AUC                                                                                                                                                              AUPR
SynLethKG       separately.
                Table            5            shows            the            prediction            performance            of            several            machine                                                                                                                                                                                                  KG-based                                                                                             TransE                                                                                                                             0.5870   6    0.0086                             0.6100   6    0.0109
learning          models           on           SynLethKG,           SL           graph          and           the           combination           of                                                                                                                                                                                                                SL-based                                                                                                       GCN                                                                                                                                                  0.8329   6    0.0172                             0.8727   6    0.0110
both.            KG-based            methods           intend            to            learn            low-dimensional            embed-                                                                                                                                                                                                                            KG þ      SL-based                             TransE þ      GCN                             0.9063   6    0.0071                             0.9138   6    0.0056
dings       of       entities       and       relations       in       SynLethKG       automatically.       We       take                                                                                                                                                                                                                                                                                                                               RF                                                                                                                                                                              0.8882   6    0.0130                             0.9218   6    0.0098
TransE       (Bordes       et       al.,       2013),       a       popular       unsupervised       KG       embedding                                                                                                                                                                                                                                                                                                                                 KG4SL                                                                                                                            0.9470   6    0.0003                             0.9564   6    0.0005
learning            method            as            an            example,            which            is            trained            based            solely            on
SynLethKG.                We                also                compare               the                contributions                of                a                single               SL                                                                                                                                                                       Note:      The      best      results      for      each      index      are      in      bold.
graph,               which               is               called               SL-based               method,               whose               performance               is
exactly          that           of           GCN           in           Table           4.           Then,          a           combination           of           the           KG-
and            SL-based            method,            named            ‘TransE   þ            GCN’,            is            further            tested.                                                                                                                                                                                                              (Glorot        and        Bengio,       2010)        as        the        initial        node        features,        obtains        AUC
From       the        first        three        rows       in        Table        5,        we        can        observe        that       additional                                                                                                                                                                                                                score       of       0.8329       and       AUPR       score       of       0.8727.       The       models       that       inspect
information            from           the           KG           guides           the           model           to           achieve           better           per-                                                                                                                                                                                                 the          KG          and          SL          graph          together          outperform          either          of          the          above.          All
formance        than         using         KG        or         SL         graph        alone.        TransE,         trained         on        the                                                                                                                                                                                                                  the            evidence            supports            that            the            KG            information            can            help            with            SL
KG        only,        reports        an        AUC        score        of        0.5870        and        an        AUPR        of        0.6100,                                                                                                                                                                                                                   prediction.
which        are        the        lowest        values        among        the        three        models.        GCN,        operat-                                                                                                                                                                                                                                                After            that,            an            ensemble            learning           method            named           RF            (Breiman,
ing       on       the       SL       graph       directly,       leverages       Xavier’s       uniform       distribution                                                                                                                                                                                                                                          2001),         which         also         uses         the         information         extracted         from         KG         and         SL,

i424                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      S.Wang      et       al.
                                                                                                                                                                                                                  GNN        model.        Many        existing        methods        view        each        SL        pair        as         an        inde-
                                                                                                                                                                                                                  pendent         sample        from        a        latent        representation,        but        this        assumption        is
                                                                                                                                                                                                                  an           oversimplification           in           the           biological          context           of           cancer           cells.           KG
                                                                                                                                                                                                                  can        bring        additional        information        such        as        biological        processes        which
                                                                                                                                                                                                                  could         be         crucial         for         discovering         new         SL         pairs.         By         injecting         the         KG,
                                                                                                                                                                                                                  our               model               was               capable               of               harnessing               the               aforementioned               issue,
                                                                                                                                                                                                                  without              manual              feature              engineering.              Extensive              experiments               have
                                                                                                                                                                                                                  been       conducted       to       examine       the       impact       of       KG.       The       results       show       that
                                                                                                                                                                                                                  both        KG        and        MP        in        GNN        are        essential        for        boosting        the        model        per-
                                                                                                                                                                                                                  formance.        Therefore,        KG4SL        represents        a        breakthrough        in        applying
                                                                                                                                                                                                                  supervised       machine      learning      to       SL       prediction.
                                                                                                                                                                                                                            Our       future       work       will       focus       on       the       following       directions.       First,       we
                                                                                                                                                                                                                  plan        to        further        improve        our        model,        using        the        strategy        of        contrastive
                                                                                                                                                                                                                  learning,          which          is          a          self-supervised          approach          to          learn          graph          repre-
                                                                                                                                                                                                                  sentation.           Secondly,           the           degrees           of           some           nodes           in           a           KG           might           be
                                                                                                                                                                                                                  very       large,       while       sampling       a       fixed-sized       neighborhood       may       not       fully
                                                                                                                                                                                                                  capture            the            neighborhood            topological            structures            of            the            nodes.            To
                                                                                                                                                                                                                  address          this          issue,          it          will          be          a          highly          desirable         future          work          to          con-
                                                                                                                                                                                                                  trast         multiple         receptive          fields         for         a         given         node,         to         learn         a         more         ro-
                                                                                                                                                                                                                  bust            and            enriched            embedding.            Moreover,            inspired            by            the            recently
                                                                                                                                                                                                                  proposed        KG        attention        network        (Wang        et        al.,        2019c),        we         are         inter-
                                                                                                                                                                                                                  ested            in            incorporating            the            attention            mechanism            into            the            KG            MP.
                                                                                                                                                                                                                  Considering            there            might            be            some            promiscuous           and            uninformative
                                                                                                                                                                                                                  neighbors               for               MP,               the               attention               could               play               the               role               of               a               filter
Fig.       4.       Visualization       of       SL       interaction       network.       Link       embeddings       are       generated       by                                                               (Hamilton,                   2020).                   By                   inspecting                   and                   visualizing                   the                   attention
different        models        of        gene–gene        pairs        in        the        SL        matrix.        Dot        colors        indicate        whether                                             weights,        interpretability        analysis        could        be        done        to        account        for        deci-
there        is        an        SL        relationship        between        a        pair        of        genes.        TransE        or        GCN        alone        cannot                                 sions         made         by         GNN         models.         Thirdly,        since         our         model         outperforms
distinguish     the     two     types     of     links     well     as     they     only     capture     information     from     the     KG                                                                      most              of              the              state-of-the-art              models,              it              is              desirable              to              utilize              our
or     SL      interaction,      whereas      KG4SL      can      distinguish     the      two      link      embeddings,      demon-
strating     its     ability     to     learn     from     both     the     SL     network     and     the     KG                                                                                                 model         to         discover         novel         SL         pairs         and         collaborate         with         biologists         to
                                                                                                                                                                                                                  validate          candidate          SL          pairs          as          drug          targets.          Last          but          not          least,          the
is       selected       to       be       compared       with       ‘TransE  þ       GCN’.       The       difference       be-                                                                                   Cancer       Dependency       Map       (DepMap)       project       (Tsherniak       et       al.,       2017)
tween             RF             and             ‘TransE   þ             GCN’             is             that             whether             the             features             are                            aims           to           examine           how           perturbing           a           given           target           gene           in           a           specific
extracted               automatically.               The               features               of               RF               should               be               carefully                                   cancer       type       might       affect       the       tumor       growth,       including       cases       when       the
selected          manually,          whereas          the          gene          embeddings          are          automatically                                                                                   best       targets       are       SL       partners       of       an       altered       gene.       Das       et       al.       (2019)       pre-
extracted        from        TransE        and        fed        into        GCN        to        generate        SL        prediction                                                                            dicted      the       SL       pairs       in      the       different      cancer       types       which      considered       the
results.                          Here,                          RF                          uses                          six                          features,                          namely,                          minTriangles,tissue             context.             Wan             et             al.             (2020)             introduced             the             cell-line-specific
maxTriangles,        minCoefficient,         maxCoeffiecient,        sp        and         sl.        For        each                                                                                             gene             expression             information             to             help             predict             SL             interaction,             since
gene         pair,         minTriangles         and         maxTriangles         reflect         the         max         and         min                                                                          most         SL         pairs         remain         cell-line         specific.         It         is         believed         that         such         con-
numbers       of        triangles        that       each       gene       forms.        The       minCoeffiecient        and                                                                                      text-specific         information        can        provide        useful        features        for        the         SL        pre-
maxCoeffiecient         reflect         the         min         and         max         likelihood         that         the         neigh-                                                                        diction               problem.              Integrating               the               DepMap               data               into               our               KG4SL
bors      of      these      two      genes      are      connected.      sp      is      a      Boolean      value      that      rep-                                                                           model,           we           can           develop           an           AI           system           to           facilitate           the           discovery           of
resents         whether         the         two         genes         are         in         the         same         community         detected                                                                  SL-based             anticancer             therapeutics.             As             many             gene             mutations             cause
by        the        label        propagation       algorithm.       sl        means        whether       two        genes        are                                                                             cancer      cells      to      inactivate,      it       is      possible      to      kill      the      cancer      cells      by      iden-
in              the              same              community              detected              by              Louvain              algorithm              (Blondel                                              tifying          the           SL           partners          of           these           genes           (O’Neil          et           al.,           2017).          In           the
et       al.,       2008).       The       results       show       that       ‘TransE  þ       GCN’       achieves       slight-                                                                                 area            of            drug            discovery            or            drug            repurposing,            using            AI            methods           to
ly       higher      AUC      and      slightly       lower      AUPR      than      RF.                                                                                                                          narrow      candidate      drug       targets      set      can      speed       up       the       research      process.
         Comparing       all        these       models       with       KG4SL       which       is        an       end-to-end
model         using         the         information         extracted        from         KG         and         SL         automatic-                                                                            Acknowledgements
ally,           KG4SL           yields           the           top           AUC           of           0.9470           and           AUPR           of           0.9564.
This      signifies       the      benefit      of       adding      a      suitable      KG       in      SL       prediction.                                                                                   The        authors        would        like         to        thank         Xin         Liu        from        ShanghaiTech        University         for
         Furthermore,         to         qualitatively         interpret         the         above         models’         learn-                                                                                 his      help      with      a      few      case      studies      of      predicted      SLs.
ing        abilities,        we        draw        the        link        features        extracted        from        these        models.
First,          we          fix          the          dimension         of          node          features          to          256.          Next,          the          fea-                                    Conﬂict      of      Interest:      none      declared.
tures          for          each          node          of          an          SL          pair          in          test          data          are          concatenated          to-
gether,            representing            the            link            embedding            between            them.            Then,            the                                                           References
high-dimensional            feature            vectors             are            mapped            into             a             2D            space            by
using          visualization          technique          t-SNE          (Van          der          Maaten          and          Hinton,                                                                           Ashworth,A.     et     al.     (2011)     Genetic     interactions     in     cancer     progression     and     treat-
2008).       As       Figure       4       shows,       orange       dots       denote       there       is       an       SL       relation                                                                            ment.     Cell,     145,     30–38.
between        a         pair        of         test         genes        and         blue        dots         are         the         opposite.        We         are                                            Blondel,V.D.      et      al.      (2008)      Fast      unfolding      of      communities      in      large      networks.      J.
curious        whether        the        models        can        tell        the        differences       between        the        two                                                                                Stat.     Mech.     Theory     Exp.,     2008,     P10008.
kinds          of          link          types.          Clearly,          on          one          hand,          TransE          has          the          weakest                                              Bordes,A.       et       al.       (2013)       Translating       embeddings       for       modeling       multi-relational
distinguish       capacity,       as       the       SL       label       information      is       not       taken      into       ac-                                                                                 data.          In:          Advances          Neural          Information          Processing          Systems,          Vol.          26,          pp.
count.              On              the              other              hand,             although             both              ‘TransE   þ              GCN’             and                                          2787–2795.     Lake     Tahoe,     Nevada,     USA,     December     5–12,     2013.
KG4SL        are        integrated        with        the        KG,        KG4SL        makes        better        use        of        this                                                                     Breiman,L.     (2001)     Random     forests.     Mach.     Learn.,     45,     5–32.
information,       separating      the       two       types      of       links      more      thoroughly.                                                                                                       Cai,R.      et      al.      (2020)      Dual-dropout      graph      convolutional      network      for      predicting
                                                                                                                                                                                                                        synthetic     lethality     in     human     cancers.     Bioinformatics,     36,     4458–4465.
                                                                                                                                                                                                                  Das,S.     et     al.     (2019)     Discoversl:     an     r     package     for     multi-omic     data     driven     predic-
4        Conclusion        and        discussions                                                                                                                                                                       tion     of     synthetic     lethality     in     cancers.     Bioinformatics,     35,     701–702.
                                                                                                                                                                                                                  Dhabhar,F.S.      (2009)      Enhancing      versus      suppressive      effects      of     stress      on     immune
In          this          article,          we          proposed          a          novel          framework          named          KG4SL          for                                                                function:                    implications                    for                    immunoprotection                    and                    immunopathology.
predicting        SL,        which        incorporates        a        knowledge        graph        (KG)        into        the                                                                                        Neuroimmunomodulation,     16,     300–317.

KG      for       SL       production                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          i425
Dobzhansky,T.          (1946)          Genetics          of          natural          populations.          xiii.          recombination                                                                   Luo,J.      et      al.      (2009)      A      genome-wide      rnai      screen      identiﬁes      multiple      synthetic      le-
     and         variability        in         populations         of         drosophila         pseudoobscura.        Genetics,        31,                                                                      thal     interactions     with     the     ras     oncogene.     Cell,     137,     835–848.
     269–290.                                                                                                                                                                                              O’Neil,N.J.       et       al.       (2017)       Synthetic       lethality       and       cancer.       Nat.       Rev.       Genet.,       18,
Du,D.          et          al.          (2017)          Genetic          interaction          mapping          in          mammalian          cells          using                                               613–623.
     crispr     interference.     Nat.     Methods,     14,     577.                                                                                                                                       Ou,M.       et      al.      (2016).      Asymmetric      transitivity       preserving       graph      embedding.      In:
Folger,O.           et           al.           (2011)           Predicting           selective           drug           targets           in           cancer           through                                  Proceedings              of              the              22nd              ACM              SIGKDD              International              Conference              on
     metabolic     networks.     Mol.     Syst.     Biol.,     7,     501.                                                                                                                                       Knowledge          Discovery           and           Data           Mining,          pp.           1105–1114.          San           Francisco,
Glorot,X.      and      Bengio,Y.      (2010).      Understanding      the      difﬁculty      of      training      deep                                                                                        CA,     USA,     August     13–17,     2016.
     feedforward                         neural                         networks.                        In:                         Proceedings                         of                        the                         ThirteenthPaladugu,S.R.           et           al.           (2008)           Mining           protein           networks           for           synthetic           genetic
     International                 Conference                 on                 Artiﬁcial                 Intelligence                 and                 Statistics,                 pp.                      interactions.     Bmc     Bioinformatics,     9,     1–14.
     249–256.     Sardinia,     Italy,     May     13–15,     2010.                                                                                                                                        Perozzi,B.         et        al.        (2014).        Deepwalk:        online        learning        of        social        representations.
Grover,A.      and      Leskovec,J.      (2016).      node2vec:      scalable      feature      learning      for      net-                                                                                      In:          Proceedings          of          the          20th          ACM          SIGKDD          International          Conference          on
     works.                    In:                    Proceedings                    of                    the                    22nd                    ACM                    SIGKDD                    InternationalKnowledge           Discovery           and           Data           Mining,           pp.           701–710.           New           York,           NY,
     Conference        on        Knowledge        Discovery        and        Data        Mining,        pp.        855–864.        San                                                                          USA,     August     24–27,     2014.
     Francisco,     CA,     USA,     August     13–17,     2016.                                                                                                                                           Sanders,J.T.        et        al.        (2020)        Radiation-induced        DNA        damage        and        repair        effects
Guo,J.      et      al.      (2016)      SynLethDB:      synthetic      lethality      database      toward      discovery                                                                                       on     3d     genome     organization.     Nat.     Commun.,     11,     1–14.
     of         selective         and         sensitive         anticancer         drug         targets.         Nucleic         Acids         Res.,         44,                                           Simons,A.       et       al.       (2001)       Establishment       of       a       chemical       synthetic       lethality       screen
     D1011–D1017.                                                                                                                                                                                                in     cultured     human     cells.     Genome     Res.,     11,     266–273.
Hamilton,W.       et        al.        (2017)        Inductive        representation        learning        on        large        graphs.                                                                 Tang,J.          et          al.         (2015).         Line:          large-scale          information          network          embedding.         In:
     In:              Advances              in              Neural              Information              Processing              Systems,              Vol.              30,              pp.                    Proceedings      of       the      24th      International       Conference      on      World      Wide      Web,      pp.
     1024–1034.     Long     Beach,     CA,     USA,     December     4–9,     2017.                                                                                                                             1067–1077.     Florence,     Italy,     May     18–22,     2015.
Hamilton,W.L.        (2020)        Graph        representation        learning.        In:        Synthesis        Lectures                                                                                THUDM.       (2020).       Cogdl       project       webpage.       https://github.com/THUDM/cogdl
     on     Artiﬁcial     Intelligence     and     Machine     Learning,     Vol.     14,     pp.     1–159.                                                                                                     (15     January     2021,     date     last     accessed).
Hartwell,L.H.       et       al.       (1997)       Integrating       genetic        approaches       into       the       discovery                                                                       Tsherniak,A.            et            al.            (2017)            Deﬁning            a            cancer            dependency            map.            Cell,            170,
     of     anticancer     drugs.     Science,     278,     1064–1068.                                                                                                                                           564–576.
Huang,J.     et     al.     (2019)     Predicting     synthetic     lethal     interactions     in     human     cancers                                                                                   Van          der          Maaten,L.          and          Hinton,G.          (2008)          Visualizing          data          using          t-SNE.          J.
     using                 graph                regularized                self-representative                 matrix                factorization.                BMC                                           Mach.     Learn.     Res.,     9(11),     2579–2605.
     Bioinformatics,     20,1–8.                                                                                                                                                                           Veli/C20ckovi/C19c,P.         et         al.         (2017)         Graph         attention         networks.         arXiv         Preprint         arXiv,
Jacunski,A.     et     al.     (2015)     Connectivity     homology     enables     inter-species     network                                                                                                    1710,     10903.
     models     of     synthetic     lethality.     PLoS     Comput.     Biol.,     11,     e1004506.                                                                                                      Wan,F.        et        al.        (2020)        Exp2sl:        a        machine        learning        framework        for       cell-line-spe-
Jerby-Arnon,L.       et       al.       (2014)       Predicting       cancer-speciﬁc       vulnerability       via       data--                                                                                  ciﬁc     synthetic     lethality     prediction.     Front.     Pharmacol.,     11,     112.
     driven     detection     of     synthetic     lethality.     Cell,     158,     1199–1209.                                                                                                            Wang,H.      et     al.      (2019a).     Knowledge-aware     graph     neural     networks     with     label
Kipf,T.N.          and          Welling,M.          (2016)          Semi-supervised          classiﬁcation          with          graph                                                                          smoothness        regularization        for        recommender        systems.        In:                    Proceedings        of
     convolutional     networks.     arXiv     Preprint     arXiv,     1609,     02907.                                                                                                                          the                  25th                 ACM                  SIGKDD                  International                  Conference                  on                  Knowledge
Kranthi,T.     et     al.     (2013)     Identiﬁcation     of     synthetic     lethal     pairs     in     biological     sys-                                                                                  Discovery        &        Data        Mining    ,        pp.        968–977.        Anchorage,        AK,        USA,        August
     tems     through     network     information     centrality.     Mol.     bioSyst.,     9,     2163–2167.                                                                                                   4–8,     2019.
Kwok,M.     et     al.     (2016)     Atr     inhibition     induces     synthetic     lethality     and     overcomes                                                                                     Wang,M.           et          al.           (2019b)          Deep          graph          library:          a          graph-centric,          highly-per-
     chemoresistance             in            tp53-or            atm-defective             chronic            lymphocytic            leukemia
     cells.     Blood,     127,     582–595.                                                                                                                                                                     formant      package      for      graph      neural      networks.              arXiv      Preprint      arXiv   ,       1909 ,
Liany,H.        et        al.        (2020)        Predicting        synthetic        lethal        interactions        using        heteroge-                                                                   01315.
     neous     data     sources.     Bioinformatics,     36,     2209–2216.                                                                                                                                Wang,X.        et        al.        (2019c).        KGAT:        knowledge        graph        attention        network        for        rec-
Lin,X.       et       al.       (2020).       KGNN:       knowledge       graph       neural       network       for       drug-drug                                                                             ommendation.          In:          Proceedings          of          the          25th          ACM          SIGKDD          International
     interaction         prediction.          In:         Proceedings         of         the         Twenty-Ninth         International                                                                          Conference                on                Knowledge                Discovery                &                Data                Mining,                pp.                950–958.
     Joint               Conference               on               Artiﬁcial               Intelligence,               IJCAI-20,               pp.               2739–2745.                                      Anchorage,     AK,     USA,     August     4–8,     2019.
     Yokohama,     Japan.                                                                                                                                                                                  Wu,M.      et      al.      (2014)      In      silico      prediction      of      synthetic      lethality      by      meta-analysis
Liu,Y.       et       al.       (2020)       SL2MF:       predicting       Synthetic       Lethality       in       Human       Cancers                                                                          of      genetic      interactions,      functions,      and      pathways      in      yeast      and      human      cancer.
     via                 Logistic                 Matrix                 Factorization.                  IEEE/ACM                  Trans.                 Comput.                 Biol.                          Cancer     Inform.,     13     (Suppl.     3),     71–80.
     Bioinformatics,     17,     748–757.                                                                                                                                                                  Zhang,F.       et      al.      (2015)      Predicting       essential      genes       and       synthetic       lethality       via       in-
Lord,C.J.         and         Ashworth,A.         (2017)         Parp         inhibitors:         synthetic         lethality         in         the                                                             ﬂuence      propagation      in      signaling      pathways      of      cancer      cell      fates.      J.      Bioinform.
     clinic.     Science,     355,     1152–1158.                                                                                                                                                                Comput.     Biol.,     13,     1541002.

