Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                             BMC Bioinformatics
https://doi.org/10.1186/s12859‑024‑05989‑                                                                                                                                                                                                                                                                                                                                                                                                                                                                                y
   RESEARCH                                                                                                                                                                                                      Open Access
MoAGL‑SA: a multi‑  omics adaptive
integration method with graph learning and self
attention for cancer subt ype classification
LeiCheng                                             1, QianHuang                                                                                                                            1, ZhengqunZhu                                                                                                                                                                                                                                                    2, Yanan                                                                                                                                     Li1, ShuguangGe                                                                                                                                                                                                                                              3, LongzhenZhang                                                                                                                                                                                              4 and PingGong                                                                                      1*
*Correspondence:                                                  Abstract
gongping@xzhmu.edu.cn
1 School ofMedical Imaging,                                       Background:                                                                                                                               The integration ofmulti-omics data throughdeep learning hasgreatly
Xuzhou Medical University,                                        improved cancer subtype classification, par ticularly infeature learning andmulti-omics
Xuzhou221004, Jiangsu, China                                      data integration. However, key challenges remain inembedding sample structure
2 The First Clinical School
ofMedicine, Xuzhou Medical                                        information intothefeature space anddesigning flexible integration strategies.
University, Xuzhou221004,                                         Results:                                                                                                                                          We propose MoAGL-SA, anadaptive multi-omics integration method based
Jiangsu, China
3 School ofMedical                                                ongraph learning andself-attention, toaddress these challenges. First, patient rela-
Information andEngineering,                                       tionship graphs are generated fromeach omics dataset using graph learning. Next,
Xuzhou Medical University,                                        three -layer graph convolutional networks are employed toextract omic-specific graph
Xuzhou221004, Jiangsu, China
4 Department ofRadiation                                          embeddings. Self-attention isthen used tofocus onthemost relevant omics, adap                                                                                                                                                                                                                                                                                                                                                                                                                                         -
Oncology, Affiliated Hospital                                     tively assigning weights todifferent graph embeddings formulti-omics integration.
ofXuzhou Medical University,                                      Finally, cancer subtypes are classified using asoftmax classifier.
Xuzhou221004, Jiangsu, China
                                                                  Conclusions:                                                                                                                                                                                                                                            Experimental results show thatMoAGL-SA outper forms several popu-
                                                                  lar algorithms ondatasets forbreast invasive carcinoma, k idney renal papillar y cell
                                                                  carcinoma, andk idney renal clear cell carcinoma. Additionally, MoAGL-SA successfully
                                                                  identifies key biomarkers forbreast invasive carcinoma.
                                                                  Keywords:                                                                      Adaptive multi-omics integration, Graph learning, Graph convolution
                                                                  network, Self-attention, Cancer subtype classification
                                                              Introduction
                                                              Cancer is a highly heterogeneous disease that seriously endangers human health. Accu-
                                                              rate cancer subtyping is crucial for precise diagnosis, treatment guidance, prognostic
                                                              stratification, and drug development. The advent and refinement of next-generation
                                                              sequencing technologies have paved the way for the accumulation of vast biological
                                                              datasets in public repositories, readily available for cancer subtyping research [1]. For
                                                              example, The Cancer Genome Atlas (TCGA), an influential cancer genomics project that
                                                              aggregates an array of data , including mRNA , DNA methylation, miRNA expression, and
                                                              mutation information, spanning over 30 cancer types and a multitude of patient cases
                                                              [2]. Unlike single-omics datasets, multi-omics data offer a holistic view of the molecu                                 -
                                                              lar dynamics driving cancer progression, providing a powerful resource for advancing
                                                              © The Author(s) 2024. Open Access  This article is licensed under a Creative Commons Attribution-NonCommercial-NoDerivatives 4.0
                                                              International License, which permits any non-commercial use, sharing, distribution and reproduction in any medium or format, as long
                                                              as you give appropriate credit to the original author(s) and the source, provide a link to the Creative Commons licence, and indicate if you
                                                              modified the licensed material. You do not have permission under this licence to share adapted material derived from this ar ticle or par ts of
                                                              it. The images or other third party material in this article are included in the article’s Creative Commons licence, unless indicated other wise
                                                              in a credit line to the material. If material is not included in the article’s Creative Commons licence and your intended use is not permitted
                                                              by statutor y regulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy
                                                              of this licence, visit http:// creat iveco mmons. org/ licen ses/ by- nc- nd/4. 0/.

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                    Page 2 of 19
                   precision medicine. Studies have shown that integrating multi-omics data significantly
                   enhances the efficac y of clinical outcomes [3–6].
                     However, the high-dimensional nature of multi-omics data and relatively small sample
                   sizes present significant challenges for feature extraction and cancer subtype classifica-
                   tion. Traditional machine learning methods often rely on feature selection for dimen-
                   sionality reduction [7], but such techniques struggle to fully capture complex, nonlinear
                   relationships within the data. In contrast, deep learning (DL) offers new approaches for
                   multi-omics integration, with its powerful data processing capabilities and adaptability
                   to complex data structures. DL not only efficiently handles high-dimensional omics data
                   but also learns intricate patterns and nonlinear relationships , making it highly promising
                   for cancer subtype classification tasks.
                     Most DL-based multi-omics integration methods are unsupervised and do not fully
                   leverage available sample label information. These methods typically integrate omics
                   data at the input level, then use DL models to transform them into lower-dimensional
                   features. For example, SALMON [8] links mRNA , miRNA , and clinical data for breast
                   cancer prognosis, while Subtype-GAN [9] uses variational autoencoders and GANs for
                   subtype classification. However, these approaches often neglect relationships between
                   patients during feature learning and treat all omics data equally, which can introduce
                   bias and reduce classification accuracy. More recently, the rise of supervised learning
                   and the increasing availability of annotated datasets have allowed DL models to leverage
                   sample labels for more accurate cancer subtype classification. For instance, MOSAE [1]
                   and DeepOmix [10] utilize autoencoders    (AEs) to produce omics-specific representa                                                                                          -
                   tions that are later fused for classification. Although these super vised methods improve
                   model accurac y and interpretability, they still lack consideration for patient relationships
                   and the var ying importance of different omics data in classification tasks .
                     To address these issues, graph convolutional networks (GCNs) [11,                                                                                                    12] have emerged
                   as a powerful tool for multi-omics integration.GCNs effectively model relationships
                   between samples and leverage graph-structured data, providing stronger data fitting
                   and generalization capabilities. For instance, Wang et    al. [13] introduced multi-omics
                   graph convolutional networks (MORONET), a multi-omics integration learning frame-
                   work that employs GCNs for omics-specific learning and incorporates a view correla-
                   tion discovery network (VCDN) to unearth intricate cross-omics correlations within
                   the label space. Li et    al. [14] crafted a multi-omics integration model based on graph
                   convolutional network (MoGCN), a model designed for cancer subtype classification.
                   This model initially utilizes AEs and the similarity network fusion (SNF) method for
                   dimensionality reduction of the original features and for constructing the patient simi-
                   larity network (PSN), respectively. Subsequently, both the vector features and the PSN
                   are fed into the GCN for further training and evaluation. Bo Yang et    al. [15]proposed
                   multi-reconstruction graph convolutional network (MRGCN) for the integrative repre-
                   sentation of multi-omics data. This method first generates graphs for each omics dataset
                   based on neighborhood relationships and then encodes each dataset to procure individ-
                   ual embeddings. MRGCN also formulates an indicator matrix to represent the scenario
                   of data absence and integrates each individual embedding into a unified representation.
                   Despite the advancements brought by these GCN-based models, they typically rely on
                   adjacency matrices that are constructed through manual calculations or based on prior

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                     Page 3 of 19
                   knowledge, which usually introduce subjective parameters that lead to time-consuming ,
                   and can’t accurately reflect the actual sample relationship. How to automatically learn
                   the sample relationship from complex and heterogeneous omics data is a challenge in
                   the classification of cancer subtypes .
                     Another critical challenge in multi-omics research is how to effectively integrate
                   diverse omics data. Most existing methods mentioned above generally connect differ-
                   ent omics features, ignoring the contribution of different omics data in the classifica-
                   tion task . The attention mechanism [16,                                                      17], a concept that has risen to prominence in
                   various fields, offers a promising alternative. It operates by discerning the significance
                   of different segments within the training data, enabling models to concentrate on the
                   most informative aspects. Great successes have been made on many problems, includ-
                   ing machine translation [18], recommendation [19], image classification [20], etc. In
                   particular, self-attention (SA), a variant of the attention mechanism, has demonstrated
                   exceptional prowess in modeling complex relationships by concurrently focusing on all
                   positions within the same sequence. Its ability to model arbitrar y dependencies, as evi-
                   denced in tasks such as machine translation and sentence embedding, underscores its
                   potential. [21]. Deploying SA for omics integration can enable more flexible and adap                                                                                                           -
                   tive learning of omics’ importance, leading to better classification results .
                     In this paper, a novel multi-omics integration method named MoAGL-SA was devel-
                   oped for cancer subtype classification to cope with the above-mentioned questions.
                   Firstly, MoAGL-SA addresses the limitations of previous methods by automatically
                   learning patient relationship graphs for each omic, using graph learning to capture
                   structural information. This eliminates the need for predefined graphs and allows the
                   model to reflect actual patient relationships more accurately. Secondly, GCNs were
                   used to aggregate original features and structural information into low-dimensional
                   graph embedding . Then, graph embeddings of various omics were adaptively integrated
                   through the SA. Finally, the integrated feature representation was fed into the classi-
                   fier to accomplish the cancer subtype classification task . Experiments on three datasets
                   demonstrate the superior performance of MoAGL-SA , as well as its ability to identif y
                   key biomarkers relevant to cancer subtypes .
                     The innovation of our work can be summarized as follows:
                     •                  We introduce graph learning into GCN-based models to automatically structure
                         sample relationship graphs from complex, heterogeneous multi-omics data, elimi-
                         nating the need for manually predefined relationships .
                     •                               We develop an SA-based multi-omics integration method that adaptively learns the
                         importance of each omic, offering a more flexible and effective solution for cancer
                         subtype classification.
                   Materials and methods
                   Dateset
                   Omics data from TCGA were utilized to evaluate the performance of MoAGL-SA for
                   cancer subtype classification. Three cancer datasets including breast invasive carci-
                   noma (BRCA), kidney renal papillar y cell carcinoma (KIRP), and kidney renal clear cell

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                                                                                                                                                                        Page 4 of 19
                                                  Table1                                                                                                                          Summar y of the datasets
                                                  Dataset                                                                                                                                                        CategoriesNumber of features
                                                                                                                                                                                                                mRNA                                                                                                                                                        DNA methylation                                                                                                                                                        miRNA
                                                  BRCA                                                                                                                                                                                                                             Normal: 22, Basal: 128, Her2: 65,3217                                                                                                                                                                            3140                                                                                                                                                                                                                                                                                                                                                         383
                                                                                                  LumA: 405, LumB: 182
                                                  KIRP                                                                                                                                                                                                           Early : 191, Late: 65                                                                                                                                                                                                                                                                                                                                                     16,175                                                                                                                                                    16,244                                                                                                                                                                                                                                                                                                                                 393
                                                  KIRC                                                                                                                                                                                                        Early : 184, Late: 129                                                                                                                                                                                                                                                                                                                                   16,406                                                                                                                                                    16,459                                                                                                                                                                                                                                                                                                                                 342
                                                       Fig.1                                                 The overall architecture of MoAGL-SA
                                                  carcinoma (KIRC) were used in this paper. Only subjects with matched mRNA expres-
                                                  sion, DNA methylation, and miRNA expression data were included in the study for each
                                                  cancer type. PA M50 breast cancer subtypes were adopted as the label of BRCA patients.
                                                  KIRP and KIRC were regarded as two binar y classification tasks , which were classified as
                                                  early-stage and late-stage according to the pathological stage following the study of [22].
                                                  The details of the datasets are listed in Table 1.
                                                  The architecture of MoAGL‑SA
                                                  The architecture of MoAGL-SA , shown in Fig . 1, consists of two critical modules: omic-
                                                  specific graph embedding and adaptive multi-omics integration with SA. The omic-
                                                  specific graph embedding module generates low-dimensional representations for each
                                                  omic, while the adaptive multi-omics integration module fuses these embeddings using
                                                  SA . The subsequent sections provide detailed descriptions of each module.
                                                  Notation
                                                  Let            X      ={                   X     1  ,     ...,    X  M   } denote a multi-omics dataset, where M is the number of the
                                                  omics. The feature matrix of omics m     ∈{                             1,     ...,   M  } is                                                                                                         X  m     =[              xm,...,     xm                                   RN×D     m,
                                                                                                                                                                                           m                                                                                         1                          N  ]∈
                                                  where N is the number of samples and D                                                                                                         is the dimension of omics m’s original feature.
                                                 Ym  =[                    ym,...,    ym                                   RNdenotes the corresponding labels of samples. The low dimen-
                                                                              1                          N  ]∈                                                                            m  =[                   m                         m
                                                  sional features learned on omics m is Z                                                                                                                     [1,...,     zN  ]∈                              RN×d      m where dm is the low
                                                  dimension of learned features. �·�                                 Frepresents the Frobenius norm.

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                            Page 5 of 19
                               Omic‑specific graph embedding
                               The initial step of omic-specific graph embedding is to construct sample relationship
                               graph of omic-specific. Sample relationship graph of the m                   -th omics is denoted with
                              G m        =                                   (V            m    ,    Am    ,    E m  )           ,                   where V   m     =[vm,     ...,     vmRN×D     m is a set of N                                               nodes, and each
                                                                                        m                 m1               N×N  ]∈N
                               node is a sample. Am     =[                          A11,     ...,    ANN  ]∈            R            ,  Amij   is the set of relations between two
                               samples in the m                                         -th omics data, and E m        ⊂                   V   m      ×                  Am      ×                  V   m is the edge set, and each edge
                               emij               =                                     (v                mi                                       ,      Amij                      ,      v mj                         )       ∈                    E  m represents the relation between sample                                                                                                                                                                                                                           i and j                                                     . Current sam-
                               ple relationship graph construction methods rely on prior knowledge or manual rules to
                               obtain adjacency matrices. This may result in the underutilization of graph nodes and fail-
                               ure to uncover potential distant connections. On the other hand, fully connected graphs
                               can introduce their own set of issues, such as the aggregation of irrelevant or redundant
                               node information, which can obscure meaningful insights and hinder the performance of
                               the model [23]. In addressing these concerns, the integration of graph learning into exist                                         -
                               ing graph architecture is a strategic move. By drawing inspiration from the work of[24] and
                               [25], this method involves learning a soft adjacency matrix rather than relying on a prede    -
                               fined, rigid graph structure.
                                   Given input V   m initialized to                             X  m , graph learning module generate a soft adjacent matrix
                               Am. First , dimensionality re duction is p erforme d to map the original dimension                                                                                      Dm to a
                               lower dimension                     dgraph  . ge soft adjacenc y matrix                                     Am is then generated through graph
                               learning to represent the relationships between samples, calculated as follows:
                                             Am                            e   m,                       i        =                                       1,    ...,       N      ,       j         =                                       1,    ...,       N,
                                                i                     =                               softmax (i                          )
                                        
                                                                    �            �   w m        ⊙                       v m�         ·���
                                                                                  �                i    �      w m        ⊙                       v mj�                                                          (1)
                                                            ,          1      −                                                           ,
                                                1              =                               Relu�w m⊙v m�     ��w m⊙    v m�
                                                                                                      i                        j
                               where         wm       ∈                Rdgraph refers to the shared learnable weight vector, and w m      ⊙                                                     vm denotes
                                                                                                                                                                                                   i
                               the Hadamard product, representing the element-by-element multiplication of two vec-
                               tors. Relu(·)             ac  tiv    ation f      unc  tion w    a     s employe      d to addre   ss the i  ssue of g     radient v    ani  shing
                               dur           ing the training pha                s        e. )e f                 unc        tion softmax (·) was applied to each row of matrix                                  Am
                               to ensure that the learned soft adjacency matrix                                                     Am met the following property:
                                         ∑N
                                                 Amij             =                                    1,     Amij             ≥0.                                                                                (2)
                                          j=  1
                               To optimi ze the le arnable weight ve ctor                                             wm  , we use a modimed loss function based on
                               [24, 26]:
                                                          M      (             N     (                   (                  m         m      ))                              )
                                                        ∑              ∑∑                                                   i                     ·      vj            2
                                         LGL          =              N     2             Amij               +                          η1      −                                                v�v m��v m�+                          γAmF(3)
                                                        m=    1                iMj                                         i          j
                              1   −         v mi·v          mj is the consine distance between sample i and j. The first term of Equ(3)
                                        v mi  v mj    
                               means that if sample i and j are farther apart in the feature space,                                                                            Amij will get smaller
                               controlled by the weight value. Similarly, samples that are proximate to each other within
                               the fe ature sp ace exhibit g re ater relation v alve s    . mis charac ter istic s er     ve s to inhibit the
                               agg   re g ation of infor  mation f  rom noi s  y no   de  s dur  ing g   raph convolution. Me  anwhile,                                                                          η is

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                  Page 6 of 19
                             a trade-off parameter that governs the significance of nodes in the graph. An average
                             operation was also performed on all nodes to reduce the impact of node count. Besides,
                             the second term of Eq.   (3) is used to control the sparsity of                                                             Am     ,       where  γ is a trade-o
                             p arame ter, and a l arger                      γ makes the soft adjacenc y matrix                                    Am more sparse. By optimiz                                 -
                             ing     LGL, the mo del learns the mapping relationship b etwe en input data and adjacenc  y
                             matr i x dur ing the training pha se. Gis enables the mo del to generate appropr iate adja                                                                                                                                                                                                                                                                                                                                                                                -
                             cency matrix                Am for the test data.
                                 To add s elf -conne   c tions     , the adjacenc      y matr i   x                          Am was modimed as follows, according to
                             [11]:
                                                            1               1              1  (               )    ˆ   1
                                        ˜                   2 ˆ      ˆ      2              2                           2                                                                                (4)
                                       Am      =                             ˆD−AmD−=                             ˆD−Am+ID−,
                             where         Dˆ denotes the diagonal node degree matrix of                                               Amˆ        ,     and I is the identity matrix .
                                  e se cond step of the omic-sp e ciIc g raph emb e dding involves feature extraction. We
                             u   s    e G               C            N to extrac    t the fe      ature    s w       ith input X  m and the corresponding soft adjacent matrix
                              Am . GCN was built by stacking multiple convolutional layers, and each layer is demned as:
                                           m                 (               m )             (  ˜m                       )
                                       H(l+      1)   =    f    H    m(l )  ,A˜      =    σ     A      m  m(l ) W      m(l ),                                                                           (5)
                             where         H   m(l                ) i s the input of the l-th layer and               H  m(0)        =Xm. W     m(l                ) i s the weig ht matr i   x of the
                             l-th layer.          σ(·)     denotes a non-linear activation function. (e σnal embedding                                                                     Z m is the
                             output of the last layer.
                             Adaptive multi‑omics integration with SA
                             To enhance multi-omics data integration, it is essential to fuse omic-specific graph embed-
                             dings in a flexible manner. Recent studies have revealed that SA can be viewed as a weighted
                             summation of values, with quer y and keys playing a crucial role in determining the weight
                             coefficients for the corresponding values. In this study, we utilized the quer y matrix Q and
                             ke    y matr ice s        K to compute attention scores for each omics dataset.  ese attention scores
                             were then use d to p erform a weighte d summation of the omic-sp e ciKc value matrices                                                                                      V      ,
                             enabling adaptive multi-omics integration.
                                 D ue to the relatively large sc ale of the network  s   , a time-e cient w ay like [27] w a s use d
                             to calculate attention score for each omic . Initially, two learnable linear transformations
                             were applie d to the g raph emb e dding of each omic , represente d by weight matrices                                                                                   W    Q
                             and     W    K re  sp    e    c  tively. Sub  s  e    quently, the quer           y matr  i    x        Q and the key matrix                       K were derived
                             as follows:
                                           m                                         [                         ]
                                       Q       =   Z m W    Q   ,    Q   =    avg      Q1  ,     ...,    QM      ,                                                                                      (6)
                                           m                                             [                          ]
                                       K        =   Z  m W     K         ,    K=stack       K        1   ,     ...,    K    M.                                                                          (7)
                             Here, the avg operation computes element-wise averages to represent consensus infor                                                                                                       -
                             mation across the omics, while the stack operation combines all key matrices, increasing
                             the rank by one. The parameters W    Q and                                             W    K are shared across all omics to maintain
                             sc alability for di erent numb ers of omic s   . Given that the output of the omic s attention

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                                                                                                                                    Page 7 of 19
                                             mechanism is a weighted average of the value matrices V   m  , it becomes paramount for
                                             these matrices to retain the original structural information to the f ullest extent . Con                                                                            -
                                             trasting with the matrices                                                          Q and                  K  , the computation of                                                   V involves utilizing a graph
                                             enco der, sp e ciVcally designe d to encapsulate higher-order str uctural information.  is
                                             graph encoder operates based on the function g and is characterized by the weight
                                             matrix W    V:
                                                           V    m        =      g(Am,    Zm  )      =                            mVm
                                                                                                                                                        V              ,                                                                                                                                          (8)
                                                                                           [                                       ]
                                                           V          =    stack               V        1  ,     ...,    V     M       ,                                                                                                                                                                          (9)
                                              e attention s core is c alc ulate  d b a  s e  d on the inner pro  duc t of the quer     y matr i  x                                                                                                                                                      Q and
                                             the ke y matri x                                K   m  . A higher inner product value indicates greater relevance of a spe                                                                                                                             -
                                             c   imc omic           , re    sulting in hig   her weig   ht b        eing a        ssig       ne         d to that omic in the inte    g       ration.  i   s
                                             approach reduces computational complexity compared to matrix multiplication.  e
                                             attention score                                wmas is computed as:
                                                            wm          =                                 soft         max (√Q                  ·    K     m  )                                                                                                                                               (10)
                                                                 as                                                                                                                                                                          ,
                                             where the                         softmax(·)                      function normalizes all choices, and √                                                                                     d denotes the scaling fac                                                                                                                                                           -
                                             tor. Follow      ing such intuition, the dnal re   pre    s   ent   ation of omic     s attention me         chani   sm w      a       s
                                             computed as:
                                                                            ∑M
                                                            Z      =                    wmas V    m.                                                                                                                                                                                                          (11)
                                                                           m=1
                                             To ensure that the structural integrity of the data is preser ved, we implemented an omics
                                             reconstruction task as a supplementar y task to the SA-based integration.  is unsu                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              -
                                             p er    v ise d ta sk is traine d concurrently and ensures that the g raph enco der retains a suf                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   -
                                             dc ient amount of or  ig   inal infor  mation by re    constr   uc ting the omic-  sp   e    c idc fe  ature  s      .  e
                                             decoder operates as follows:
                                                             ˜                      (  ˜m                   )                   m
                                                            Z m            =                         ˜gA,    Z  ′  =                             ˜Am′ W    mD,                                                                                                                                                (12)
                                             where                Z′            =              ∑m   =      1MZm,                                             and W    mD is the weight matrix for the decoder.
                                                   De obje ctive f unction for omic s re constr uction, g uide d by the me an square error,
                                             i   s demne         d a        s             :
                                                                                          1     ∑M
                                                            LD          =                                                     1|Z         m        −Z m |2˜F.                                                                                                                                                 (13)
                                                                                M       N       m=1
                                             Cancer subtype classification task
                                             To classify cancer subtypes, cross-entropy loss                                                                                                                        LCE    (·)          was employed to train
                                             MoAGL-SA:

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                Page 8 of 19
                                                   ∑Ntr          (                                           ),
                                       LC         =       1  ,         ∑              −                     SA(x                                                     n  ),                  yn      (14)
                                                   n=1
                             where Ntr is the number of samples in the training set.
                                 In summar y, the total loss function for MoAGL-SA is formulated as:
                                       L      =                   LC        +                     α LGL      +                     β  LD     ,                                                      (15)
                             where        α and       β are trade-o  parameters that control the importance of the graph learning
                             and omics reconstruction tasks, respectively.
                             Results
                             Per formance evaluation
                             Different metrics were utilized to evaluate the performance of the model for cancer sub-
                             type classification. For binar y classification tasks , accurac y (ACC), F1 score, and the area
                             under the receiver operating characteristic cur ve (AUC) were used. For multi-class clas-
                             sification tasks, accuracy (ACC), weighted F1 score (F1-w), and macro-average F1 score
                             (F1-m) were employed. ACC and F1 were defined as follows:
                                       ACC        =                                                                                            TP      +                  TNTP      +                  TN        +                  FP      +                  FN                    ,(16)
                                       F1     =  ×   ×                                Precsision×Recall                                                                                             (17)
                                                          Precsision     +                          Recall        ,
                                       Precsision      =                                                                    TPTP      +                        FP,      Recall=                                                                                    TPTP      +                        FN                    ,(18)
                             where TP, TN, FP, and FN were regarded as true positives, true negatives, false positives,
                             and false negatives, respectively.
                             Implementation details
                             Figure 2 illustrates the architecture of the MoAGL-SA network . Initially, each omic data                     -
                             set is input into a single linear layer, which maps the data to lower dimensions, followed
                             by a graph learning layer. The original features and the learned adjacency matrices are
                             then fed into a three-layer GCN to generate omic-specific graph embeddings, with the
                             activation function set to LeakyReLU. After ward, the learned features are input into the
                                Fig.2                                                 The overall architecture of MoAGL-SA

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                                                                       Page 9 of 19
                                     integration module, where three separate linear layers combined with corresponding
                                     average, stack , and graph encoder operations transform the features into the SA input
                                     matrices Q     ,               K ,          and    V. Following attention calculation, a fe e dfor  ward neural network  ,
                                     residual connections , and normalization are applied before feeding the features into
                                     the line ar layers for cla ssiVc ation and g raph de  co  ding    .  e pro  ce ss culminate s w ith the
                                     application of the softmax function for Vnal subtype classiVcation.
                                           e MoAGL-SA network was implemented using P yTorch 1.10.1 and P ython 3.9.7.
                                     Training was performed on a system equipped with an Intel(R) Core( TM) i9-13900k
                                     C PU and an N VIDI A G eForce RTX 4090 GPU. S e veral ke y hy p er parameters  , such a s
                                     the fe    ature dimensionality of the thre       e-l  ayer G            C         N, numb       er of e   p       o       chs             , le    ar    ning rate, and
                                     trade-o  parameters  , were caref ully conVg ure d.  e G  C  N fe ature dimensions were set
                                     to [400, 200, 100] for all data set s  , w ith the numb er of training ep o chs set to 2500.  e
                                     le     ar     ning rate w      a       s s   e   t to     1e                  −                       4  , and the AdamW optimizer was employed for the classi1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  -
                                     c ation t a  sk      . Co sine Anne aling w a  s us e  d to adjust the le ar ning rate throughout training     .
                                     In the g raph le ar ning mo dule, the trade-o  parameters                                                                                   γ and           η were set to 0.3 and 0.7,
                                     respectively, while                           α and           β in the βnal loss function were both set to                                                               1e                  −                       3.
                                          Figure                                              3 show s the v  ar  i ation in the lo ss v  alue ac ro ss e p   o   chs w  ith the ab   ove p arame ter
                                     s e tting  s on MoAGL-SA         . A   s s e   en in Fig       .                                                                                                                                                                                                                                                                                                                                             3, the mo   del re  ache s a low and st able lo ss v  alue
                                     under these current parameters.
                                     Comparison with previous methods
                                     MoAGL-SA was compared with five classification methods that utilized multi-omics
                                     data including three classical methods i.e., k-nearest neighbor classifier (KNN), support
                                     vector machine classifier (SVM), random forest (RF), as well as two DL-based methods
                                     of MoGCN and MORONET.
                                          •                                         KNN, the final classification was decided by voting of k-nearest neighbors;
                                          •                                     SVM identified an optimal hyperplane that maximally separated the different classes
                                                 of data set ;
                                        Fig.3                                                 The variation of the loss value with epoch on MoAGL-SA

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                  Page 10 of 19
                     •                                   RF constructed multiple decision trees during training and output the class that was
                         the mode of all the individual decision tree predictions;
                     •                            MoGCN [14] is a multi-omics integration method for cancer subtype classification
                         based on GCN, which used AE and SNF to reduce dimensionality and construct the
                         patient similarity network respectively ;
                     •                                          MORONET [13] is a method that jointly used GCN and VCDN for multi-omics data
                         classification.
                   All competitive methods were implemented using publicly available code. The scikit-
                   learn package was mainly used to implement these methods with default parameters.
                   The input data of these classical methods was a single matrix that concatenated the
                   multi-omics data. MoGCN, MORONET, and the proposed method were implemented
                   using the PyTorch framework . MoGCN utilized AEs trained with mean squared error,
                   followed by SNF for patient similarity network construction and a two-layer GCN.
                   MORONET trained a three-layer GCN with L2 parametric loss, while VCDN was
                   trained with cross-entropy loss. The critical threshold k, determining the average num                                                                                                         -
                   ber of neighbors per sample, was set between 2 and 10.
                     Following the approach in [13], 70% of the samples were used for training, and the
                   remaining 30% for testing . During testing , all samples were input to the model to main-
                   tain their relationships, and evaluation metrics were calculated only on the test set. The
                   average evaluation metrics were calculated from five random runs. Additionally, com-
                   putational time was reported to provide a comprehensive over view of performance effi-
                   cienc y. Table 2 displays the classification results on three datasets .
                     As shown in Table    2, MoAGL-SA outperformed the five compared methods in most
                   classification tasks. Although RF and SVM produced slightly higher F1 scores on the
                   KIRC dataset, MoAGL-SA achieved comparable ACC and superior AUC. Compared
                   to the other two GCN-based methods(MoGCN and MORONET), MoAGL-SA demon-
                   strated superior performance in all classification tasks, though at the cost of increased
                   computational time due to the need for optimizing the sample relationship graph.
                     The method integrated three types of omics data of mRNA, DNA methylation, and
                   miRNA . To demonstrate the necessity of integrating multiple omics data for classifica-
                   tion, we compared the classification results of the method using multi-omics data and
                   single-omics data. Fig . 4 gives the results .
                     Figure  4 shows that MoAGL-SA consistently outperformed the method based on sin                    -
                   gle-omics data, except for AUC in the KIRC classification task , where DNA and miRNA
                   individually achieved higher performance. The performance of different omics varied
                   across classification tasks. For example, mRNA performed best in BRCA classification,
                   whereas miRNA outperformed mRNA in KIRP classification. These results underscore
                   the importance of integrating multi-omics data for robust classification performance.
                   Ablation experiments
                   To further validate the robustness of the proposed method, ablation studies were sys-
                   tematically conducted, focusing on both key modules and key parameters.
                     To assess the effectiveness of the graph learning module, we replaced the graph learn-
                   ing module with a KNN and cosine distance-based graph construction method (denoted

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                             Page 11 of 19

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                                                                                                                                                                                                                                                                                                            Page 12 of 19
                                                                            Fig.4                                                 Classification results between multi-omics data and single-omics data via MoAGL-SA
                                                                     as Consine_n), where n (2,4,8,16) represents the number of k-nearest neighbors. Table                                                                                                                                                                                                                                                                           3
                                                                     displays the results for MoAGL-SA compared to Consine_n. From Table                                                                                                                                                                                                                           3, the following
                                                                     results are obser ved: firstly, for the fixed composition method Consine_n, the classifica                                                                                                                                                                                                                          -
                                                                     tion performance of the model is sensitive to the value n. For example, in BRCA classifi                                                                                                        -
                                                                     cation task , the maximum and minimum values of ACC are obtained when n is set to 2
                                                                     and 16 respectively. Secondly, compared with Consine_n methods, MoAGL-SA achieves
                                                                     superior performance across various classification metrics. This demonstrates the effec-
                                                                     tiveness of the graph learning module in constructing the sample similarly graph and
                                                                     yielding more accurate and robust classification results .
                                                                              In addition, to verif y the effectiveness of SA module in omics integration, we replaced
                                                                     SA with two alternative methods, simply adding various omics features (GCN_NA) and
                                                                     assigning equal weights (GCN_EQU) to different omics. From Table   4, we can see that
                                                                     GCN_NA and GCN_EQU methods have a relatively low classification performance in
                                                                     most evaluation metrics for three cancers, which illustrates that ordinary integration
                                                                     strategy that directly adds various omics features or assigns equal weights for differ-
                                                                     ent omics may be insufficient for cancer subtype classification. This demonstrates the
                                                                     superiority of the SA-based integration module in enhancing MoAGL-SA’s ability to
                                                                     Table3                                                                                                                          Ablation experiments of the graph learning module
                                                                     Method                                               BRCA                                                                                                                               KIRP                                                                                                                  KIRC
                                                                                                                          ACC (%)                                          F1‑                                            w (%)                                          F1‑m (%)                                                                                               ACC (%)                                          F1 (%)                                          AUC (%)                                          ACC (%)                                          F1 (%)                                          AUC (%)
                                                                     Consine_2                                                          85.48                                                                                       85.10                                                                                                   77.71                                                                                                       81.30                                                                                       88.12                                                             77.42                                                                                          71.28                                                                                       74.52                                                             74.92
                                                                     Consine_4                                                          82.99                                                                                       82.71                                                                                                   71.32                                                                                                       84.67                                                                                       90.52                                                             82.85                                                                                          70.43                                                                                       73.26                                                             73.28
                                                                     Consine_8                                                          81.91                                                                                       80.60                                                                                                   74.13                                                                                                       85.97                                                                                       91.31                                                             81.82                                                                                          70.85                                                                                       74.72                                                             74.35
                                                                     Consine_16                                        81.49                                                                                       81.14                                                                                                   76.23                                                                                                       81.56                                                                                       88.62                                                             83.26                                                                                          69.79                                                                                       73.77                                                             73.16
                                                                     MoAGL-SA                                                            88.22                                                                                  88.05                                                                                               83.83                                                                                                   90.13                                                                                  93.81                                                      92.03                                                                                      72.34                                                                                  77.02                                                      76.22
                                                                     Bold indicators the highest value of each evaluation metric
                                                                     Table4                                                                                                                          Ablation experiments of the omics integration module
                                                                     Method                                           BRCA                                                                                                                                 KIRP                                                                                                                   KIRC
                                                                                                                      ACC (%)                                            F1‑                                            w (%)                                            F1‑m (%)                                                                                                 ACC (%)                                            F1 (%)                                            AUC (%)                                            ACC (%)                                            F1 (%)                                            AUC (%)
                                                                     GCN_NA                                                                   77.10                                                                                         73.92                                                                                                     70.71                                                                                                         83.38                                                                                         88.42                                                               90.61                                                                                            70.00                                                                                         74.28                                                               76.14
                                                                     GCN_EQU                                                81.41                                                                                         79.98                                                                                                     68.67                                                                                                         80.78                                                                                         86.85                                                               90.05                                                                                            68.09                                                                                         73.60                                                               76.06
                                                                     MoAGL-SA                                            88.22                                                                                    88.05                                                                                                 83.83                                                                                                     90.13                                                                                    93.81                                                        92.03                                                                                        72.34                                                                                    77.02                                                        76.22
                                                                     Bold indicators the highest value of each evaluation metric

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                                         Page 13 of 19
                                 effectively distinguish cancer subtypes. Furthermore, the best classification performance
                                 is achieved by combining two key modules in the proposed method for three cancers,
                                 which demonstrates the superiority of graph learning for automatically constructing
                                 sample relationship graph and SA based adaptive integration modules for cancer sub-
                                 type classification.
                                     Further ablation experiments were conducted to evaluate the impact of the key param-
                                 eters(α       ,   β ,  γ , η ).    α and         β are the coe cients for graph learning loss and omics reconstruc                                                                                                                                                                                                                                                                                                                                                                                                              -
                                 tion loss , respectively, which determine the importance of graph learning and omics
                                 reconstruction task  . Ablation experiments were conducted where one parameter was
                                 βxed at          1e                  −                       3 while the other was set to [ 1e                  −                       1        , 1e                  −                       2 , 1e                  −                       3 , 1e                  −                       4 ,        0]. γ and η are key
                                 p      arame      ters for g           raph le         ar        ning lo       ss                          , where γ controls the sparsity of the adjacenc y matrix
                                 and      η gover  ns the imp   or   t ance of the sample s      . Simil arly, the abl ation exp   er  iment w  a   s s e t
                                 w                ith the           s         e two p         arame         ters of 0.3 and 0.7, re           sp                     e                      c         tively, w                ith one p         arame         ter unchange                      d in
                                 the solid state and the other set to [0.1, 0.3, 0.5, 0.7, 0.9]. Table                                                                                                                                                                                                                                                                                                                                                                                                                                          5 displays the re sult s of
                                 the ablation results for the key parameters.
                                     It i      s ob        s       er                                        ve                 d that the optimal p               er         for         mance ac      ro        ss all thre                 e d      at       a              s       e       t              s o               cc       urs when b               oth
                                 α and         β are set to 1e-3.  is parameter setting demonstrates an e ective balance, pro                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 -
                                 viding high accurac  y and F1 scores , indicating that it optimally balances p erformance
                                 across di erent datasets . Notably, variations in                                                            α have a more pronounced impact on
                                 model performance compared to changes in                                                                β  .  is suggests that the graph learning
                                 ta sk   , relative to the omics re constr uction ta sk   , plays a more cr ucial role in in uencing
                                 the βnal cla ssiβcation outcomes  . O ur mo del achie ves the b est p erformance under the
                                 setting         γ=0.3         η=0.7 compared with other settings. This further demonstrates the impor                      -
                                 tant role of graph learning tasks in the model. The above results show that appropriate
                                 parameter selection is the key to achieve robust and stable performance.
                                 Case studies
                                 To demonstrate the ability of MoAGL-SA in identif ying cancer subtypes and enhance
                                 the interpretability of MoAGL-SA, the Permutation importance method [28] was
                                 applied for important feature ranking and selection on BRC A dataset whose subtype was
                                 more complex compared to the other two datasets. Firstly, the input data is normalized
                                 to the range of [0,1]. Next, randomly set one of the feature values to 0 in the test dataset
                                 and use the trained model to re-predict the dataset. The attenuation of model perfor-
                                 mance represents the importance of the feature. In the BRCA classification task , ACC is
                                 used to evaluate the attenuation of MoAGL-SA model’s performance. By repeating this
                                 process for all features, the feature importance calculation is completed and the feature
                                 with the most significant decrease is identified as the most critical feature. Table  6 dis                     -
                                 plays the top 100 biomarkers for the BRCA dataset from mRNA expression data.
                                     Several biomarkers identified by MoAGL-SA have known associations with breast
                                 cancer. For example, E2F8 demonstrated robust elevation in both breast cancer cell lines
                                 and clinical tissue samples. Studies by Ye et  al. confirmed that upregulation of E2F8 sig-
                                 nificantly promoted breast cancer cell proliferation and tumorigenicity both in vitro and
                                 vivo [29]. KIF20A was more frequently expressed in human epidermal growth factor
                                 receptor 2 (HER2) -positive and triple-negative breast cancer than in the luminal type.
                                 Masako et     al. [30] indicated that KIF20A expression was an independent prognostic

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Page 14 of 19
                                                                                                         Table5                                                                                                                          Ablation experiments of the key parameters
                                                                                                         Method                                                                    BRCA                                                                                                                                                                                                      KIRP                                                                                                                                                                                  KIRC
                                                                                                                                                                                   ACC (%)                                             F1‑                                            w (%)                                             F1‑m (%)                                                                                                 ACC (%)                                             F1 (%)                                             AUC (%)                                             ACC (%)                                             F1 (%)                                             AUC (%)
                                                                                                         α=1e              −                    1,                                 81.78                                                                                         77.18                                                                                                      67.99                                                                                                          85.30                                                                                         85.90                                                               83.32                                                                                             57.45                                                                                         72.97                                                               71.48
                                                                                                         β=1e              −                    3
                                                                                                         α=1e              −                    2,                                 80.54                                                                                         80.20                                                                                                      75.25                                                                                                          87.90                                                                                         87.60                                                               83.83                                                                                             58.51                                                                                         73.47                                                               73.82
                                                                                                         β=1e              −                    3
                                                                                                         α=1e              −                    4,                                 80.12                                                                                         76.73                                                                                                      70.16                                                                                                          79.22                                                                                         88.41                                                               89.27                                                                                             62.77                                                                                         74.82                                                               73.36
                                                                                                         β=1e              −                    3
                                                                                                         α=0,                                                                      70.12                                                                                         65.91                                                                                                      56.41                                                                                                          79.22                                                                                         80.41                                                               76.66                                                                                             53.19                                                                                         66.15                                                               65.62
                                                                                                         β=1e              −                    3
                                                                                                         α=1e                           −             3,                           85.10                                                                                         83.25                                                                                                      80.32                                                                                                          86.62                                                                                         86.76                                                               79.94                                                                                             60.64                                                                                         73.47                                                               71.69
                                                                                                         β=1e                           −             1
                                                                                                         α=1e                           −             3,                           87.18                                                                                         85.36                                                                                                      82.39                                                                                                          89.22                                                                                         88.41                                                               80.30                                                                                             69.57                                                                                         74.48                                                               74.68
                                                                                                         β=1e                           −             2
                                                                                                         α=1e              −                    3,                                 85.93                                                                                         83.91                                                                                                      82.74                                                                                                          87.92                                                                                         87.41                                                               81.73                                                                                             62.77                                                                                         74.45                                                               71.78
                                                                                                         β=1e              −                    4
                                                                                                         α=1e              −                    3,                                 81.37                                                                                         76.27                                                                                                      67.18                                                                                                          80.52                                                                                         88.89                                                               79.68                                                                                             60.64                                                                                         72.97                                                               67.95
                                                                                                         β=0
                                                                                                         α=1e              −                    3,                                 88.22                                                                                     88.05                                                                                                  83.83                                                                                                      90.13                                                                                     93.81                                                         92.03                                                                                        72.34                                                                                     77.02                                                         76.22
                                                                                                         β=1e              −                    3
                                                                                                        γ=0.1,                                                                     85.78                                                                                         86.09                                                                                                      77.99                                                                                                          89.22                                                                                         88.41                                                               83.17                                                                                             64.89                                                                                         74.47                                                               68.08
                                                                                                        η=0.7
                                                                                                        γ=0.5,                                                                     86.35                                                                                         86.43                                                                                                      81.85                                                                                                          90.02                                                                                         88.89                                                               86.91                                                                                             62.77                                                                                         73.45                                                               68.38
                                                                                                        η=0.7
                                                                                                        γ=0.7,                                                                     84.61                                                                                         86.10                                                                                                      81.22                                                                                                          89.22                                                                                         88.41                                                               84.75                                                                                             60.64                                                                                         73.48                                                               66.62
                                                                                                        η=0.7
                                                                                                        γ=0.9,                                                                     82.32                                                                                         83.36                                                                                                      80.23                                                                                                          89.22                                                                                         88.41                                                               77.02                                                                                             59.57                                                                                         73.97                                                               67.99
                                                                                                        η=0.7
                                                                                                        γ=0.3,                                                                     85.54                                                                                         87.14                                                                                                      74.43                                                                                                          89.22                                                                                         88.24                                                               90.09                                                                                             70.64                                                                                         74.13                                                               71.57
                                                                                                        η=0.1
                                                                                                        γ=0.3,                                                                     85.95                                                                                         87.24                                                                                                      81.68                                                                                                          90.02                                                                                         90.05                                                               91.22                                                                                             70.64                                                                                         73.47                                                               74.03
                                                                                                        η=0.3
                                                                                                        γ=0.3,                                                                     86.80                                                                                         87.56                                                                                                      82.76                                                                                                          87.92                                                                                         87.41                                                               91.27                                                                                             70.64                                                                                         72.97                                                               70.79
                                                                                                        η=0.5
                                                                                                        γ=0.3,                                                                     83.78                                                                                         82.25                                                                                                      79.08                                                                                                          86.62                                                                                         86.76                                                               91.22                                                                                             71.70                                                                                         73.53                                                               71.23
                                                                                                        η=0.9
                                                                                                        γ=0.3,                                                                     88.22                                                                                     88.05                                                                                                  83.83                                                                                                      90.13                                                                                     93.81                                                         92.03                                                                                        72.34                                                                                     77.02                                                         76.22
                                                                                                        η=0.7
                                                                                                         Bold indicators the highest value of each evaluation metric
                                                                                                         factor for breast cancer. In addition, overexpression of tumor-related KIFs correlated
                                                                                                         with worse outcomes of breast cancer patients and could work as a potential prognostic
                                                                                                         biomarker [31]. Additionally, DEPDC1B was also demonstrated to be overexpressed in
                                                                                                         breast cancer [32].
                                                                                                                       Meanwhile, the biomarkers mentioned above also ranked high among differentially
                                                                                                         expressed genes in mRNA expression data, as shown in Table  7, which is sorted by false
                                                                                                         discover y rate (FDR). It validated the effectiveness of the proposed method in discover-
                                                                                                         ing these key biomarkers .
                                                                                                                       Furthermore, to demonstrate the relationship between identified biomarkers and
                                                                                                         breast cancer, KEG G pathway enrichment analyses and G O enrichment analyses , includ-
                                                                                                         ing biological process (BP) annotation, cellular component (CC) annotation, and molec-
                                                                                                         ular function (MF) annotation for the selected top 100 biomarkers, were conducted

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Page 15 of 19
                                                                                                                                                                                                                   Table6                                                                                                                         The top 100 biomarkers of BRCA dataset
                                                                                                                                                                                                                   Rank                                                                                                              Gene                                                                                                                                                                                 Rank                                                                                                              Gene                                                                                                                                                                                                                 Rank                                                                                                              Gene                                                                                                                                                                       Rank                                                                                                              Gene
                                                                                                                                                                                                                   1                                                                                                                                                                     E2F8                                                                                                                                                                                     26                                                                                                                                                   TCEAL7                                                                                                                                                                           51                                                                                                                                                   LRRFIP1                                                                                                                                                 76                                                                                                                                  CARKD
                                                                                                                                                                                                                   2                                                                                                                                                                     YBX2                                                                                                                                                                                27                                                                                                                                                   GRIN2A                                                                                                                                                                         52                                                                                                                                                   SLITRK5                                                                                                                                                  77                                                                                                                                  PLAC9
                                                                                                                                                                                                                   3                                                                                                                                                                     PRR11                                                                                                                                                               28                                                                                                                                                   EPCAM                                                                                                                                                                               53                                                                                                                                                   PRKCA                                                                                                                                                                  78                                                                                                                                  SLITRK6
                                                                                                                                                                                                                   4                                                                                                                                                                     KIF20A                                                                                                                                                      29                                                                                                                                                   MAL2                                                                                                                                                                                                     54                                                                                                                                                   PBX2                                                                                                                                                                                        79                                                                                                                                  SOX5
                                                                                                                                                                                                                   5                                                                                                                                                                     PRPH                                                                                                                                                                            30                                                                                                                                                   ZG16                                                                                                                                                                                                           55                                                                                                                                                   ESY  T1                                                                                                                                                                          80                                                                                                                                  SLIT1
                                                                                                                                                                                                                   6                                                                                                                                                                     DEPDC1B                                                                                                              31                                                                                                                                                   MMP28                                                                                                                                                                          56                                                                                                                                                   C10or f58                                                                                                                              81                                                                                                                                  PHLDA2
                                                                                                                                                                                                                   7                                                                                                                                                                     TFR2                                                                                                                                                                                    32                                                                                                                                                   ZSCAN10                                                                                                                                               57                                                                                                                                                   PYY2                                                                                                                                                                                         82                                                                                                                                  ESAM
                                                                                                                                                                                                                   8                                                                                                                                                                     MYBL1                                                                                                                                                       33                                                                                                                                                   IBSP                                                                                                                                                                                                                           58                                                                                                                                                   SLCO4C1                                                                                                                              83                                                                                                                                  CKMT1A
                                                                                                                                                                                                                   9                                                                                                                                                                     BTNL9                                                                                                                                                             34                                                                                                                                                   NDRG4                                                                                                                                                                              59                                                                                                                                                   SPINK4                                                                                                                                                             84                                                                                                                                  CLDN18
                                                                                                                                                                                                                   10                                                                                                                                                   KCNG3                                                                                                                                                   35                                                                                                                                                   FA2H                                                                                                                                                                                                             60                                                                                                                                                   DRP2                                                                                                                                                                                     85                                                                                                                                  CKMT1B
                                                                                                                                                                                                                   11                                                                                                                                                   BARD1                                                                                                                                                      36                                                                                                                                                   PLD4                                                                                                                                                                                                            61                                                                                                                                                   PKHD1                                                                                                                                                              86                                                                                                                                  LRRC55
                                                                                                                                                                                                                   12                                                                                                                                                   VSTM1                                                                                                                                                       37                                                                                                                                                   P2RY1                                                                                                                                                                                              62                                                                                                                                                   ARHGEF7                                                                                                                          87                                                                                                                                  GPRASP1
                                                                                                                                                                                                                   13                                                                                                                                                   TREH                                                                                                                                                                               38                                                                                                                                                   PA   X   2                                                                                                                                                                                                               63                                                                                                                                                   SLITRK3                                                                                                                                                  88                                                                                                                                  RDH16
                                                                                                                                                                                                                   14                                                                                                                                                   NEFL                                                                                                                                                                                 39                                                                                                                                                   SLC6A11                                                                                                                                                         64                                                                                                                                                   DQX1                                                                                                                                                                              89                                                                                                                                  CAMK1
                                                                                                                                                                                                                   15                                                                                                                                                   SOX17                                                                                                                                                          40                                                                                                                                                   RPL39L                                                                                                                                                                               65                                                                                                                                                   COX7A1                                                                                                                                          90                                                                                                                                  ERBB2
                                                                                                                                                                                                                   16                                                                                                                                                   CDH1                                                                                                                                                                     41                                                                                                                                                   IP6K3                                                                                                                                                                                                         66                                                                                                                                                   SGCA                                                                                                                                                                                                                                                            91                                                                                                                                  C9or f7
                                                                                                                                                                                                                   17                                                                                                                                                   psiTPTE22                                                                                                        42                                                                                                                                                   MYOM3                                                                                                                                                                    67                                                                                                                                                   TRIM29                                                                                                                                                        92                                                                                                                                  SPIN1
                                                                                                                                                                                                                   18                                                                                                                                                   TMEM59L                                                                                                           43                                                                                                                                                   PYCR1                                                                                                                                                                                            68                                                                                                                                                   CKMT2                                                                                                                                                             93                                                                                                                                  SSX4
                                                                                                                                                                                                                   19                                                                                                                                                   TRPM2                                                                                                                                                       44                                                                                                                                                   APOBEC3B                                                                                                                            69                                                                                                                                                   PTGDS                                                                                                                                                                 94                                                                                                                                  GPR128
                                                                                                                                                                                                                   20                                                                                                                                                   PALMD                                                                                                                                                  45                                                                                                                                                   ANKRD26P1                                                                                                        70                                                                                                                                                   TARP                                                                                                                                                                                           95                                                                                                                                  C8or f84
                                                                                                                                                                                                                   21                                                                                                                                                   LTBP2                                                                                                                                                                     46                                                                                                                                                   SLC5A11                                                                                                                                                         71                                                                                                                                                   ST8SIA6                                                                                                                                               96                                                                                                                                  UGT1A5
                                                                                                                                                                                                                   22                                                                                                                                                   FAM83A                                                                                                                                  47                                                                                                                                                   CES3                                                                                                                                                                                                                72                                                                                                                                                   NRN1                                                                                                                                                                                97                                                                                                                                  GDPD2
                                                                                                                                                                                                                   23                                                                                                                                                   RAB42                                                                                                                                                            48                                                                                                                                                   PPP1R9A                                                                                                                                                      73                                                                                                                                                   XKR8                                                                                                                                                                                         98                                                                                                                                  RNF17
                                                                                                                                                                                                                   24                                                                                                                                                   NMU                                                                                                                                                                                49                                                                                                                                                   SEPT5                                                                                                                                                                                                 74                                                                                                                                                   RABEPK                                                                                                                                                   99                                                                                                                                  ADH1A
                                                                                                                                                                                                                   25                                                                                                                                                   ME1                                                                                                                                                                                           50                                                                                                                                                   SEMA6C                                                                                                                                                               75                                                                                                                                                   C6or f145                                                                                                            100                                                                                                                                  APOA2
                                                                                                                                                                                                                   Table7                                                                                                                         The top 10 differentially expressed genes in mRNA expression data of BRCA dataset
                                                                                                                                                                                                                   Rank                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Gene                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 FDR
                                                                                                                                                                                                                   1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          DEPDC1B                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   2.70E-22
                                                                                                                                                                                                                   2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 NAT1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              3.81E-20
                                                                                                                                                                                                                   3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          E2F8                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              3.81E-20
                                                                                                                                                                                                                   4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 AGR3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          4.63E-20
                                                                                                                                                                                                                   5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          KIF20A                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           4.63E-20
                                                                                                                                                                                                                   6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 SCUBE2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         5.38E-20
                                                                                                                                                                                                                   7                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 ERBB4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                5.38E-20
                                                                                                                                                                                                                   8                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 TFF1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      7.64E-20
                                                                                                                                                                                                                   9                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 CA12                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             2.31E-19
                                                                                                                                                                                                                   10                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               BCL11A                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          2.74E-19
                                                                                                                                                                                                                   Bold indicates that this gene also appears among the top genes selected by MoAGL‑SA
                                                                                                                                                                                                                   using DAVID (https:// david. ncifc                                                                                                                                                    rf. gov/). The results of the enrichment analysis are
                                                                                                                                                                                                                   shown in Fig . 5. The left subgraph of Fig .                  5 shows the KEGG pathway enrichment analy                     -
                                                                                                                                                                                                                   sis results, where the gene ratio refers to the pathway of biomarkers associated with a
                                                                                                                                                                                                                   specific biological pathway or function related to the total number of biomarkers . A high
                                                                                                                                                                                                                   gene ratio indicates that a significant portion of biomarkers is related to the pathway,
                                                                                                                                                                                                                   suggesting a strong biological relevance. Conversely, a low gene ratio indicates a weaker

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                       Page 16 of 19
                        Fig.5                                                 The results of enrichment analysis on top 100 biomarkers
                     association with a specific biological pathway. The right subgraph of Fig . 5 shows the GO
                     enrichment analysis results. BP involves pathways and broader processes that the genes
                     are involved in; CC pertains to the parts of the cell or extracellular environment where
                     the genes are active; and MF describes the biochemical activities carried out by gene
                     products .
                        The KEGG pathway enrichment showed that these biomarkers were significantly
                     enriched in the Metabolic pathways, Cell adhesion molecules, Pathways of neurodegen-
                     eration-multiple diseases, Arginine and proline metabolism, etc. Metabolic pathways of
                     energy production and utilization were dysregulated in tumor cells, and this dysregula-
                     tion was a newly accepted hallmark of cancer. Intervention measures on breast cancer
                     pathways might benefit high-risk groups of breast cancer [33]. Cell adhesion molecules
                     was considered essential for transducing intracellular signals responsible for adhesion,
                     migration, invasion, angiogenesis, and organ-specific metastasis, which was related to
                     the metastasis of breast cancer [34]. The GO enrichment analysis of the top 100 bio                                                                                                     -
                     markers showed that their biological function focused on the phosphocreatine biosyn-
                     thetic process, axonogenesis, plasma membrane, basolateral plasma membrane, creatine
                     kinase activity, and kinase activity. Cancer-related axonogenesis and neurogenesis were
                     regarded as a novel biological phenomenon, and spatial and temporal associations
                     between increased ner ve density and preneoplastic and neoplastic lesions of the human
                     prostate were identified [35], indicating that these biological phenomena might also be
                     related to breast cancer, which deserves future research. In the aspect of the plasma
                     membrane, the fourier transform infra-red spectra of plasma membrane proteins
                     showed significant differences between normal and benign tissues compared to malig-
                     nant tissues of breast cancer at 1536 and 1645 cm−1                              [36]. Although there is no dire ct
                     literature to prove that the remaining biomarkers are relate  d to bre a  st c ancer subty  p  e s    ,
                     the enrichment analysis results show that some biomarkers are involve d in metabolic
                     pathways  , mole cular f unctions  , cellular comp onent s and biolog ical pro cesses that may
                     b                         e rel         ate                           d to the de                                     velopment and pro         g                     re             ssion of bre                 a                       st c               ancer. me             s           e cnding                 s prov                   ide a
                     promising way for future research.
                     Discussion
                     This paper proposes MoAGL-SA, a novel multi-omics integration method combining
                     graph learning , GCN, and SA in a unified framework . The graph learning of MoAGL-SA
                     could adaptively construct sample relationship graph from each omics data, overcoming
                     the limitations of traditional graph construction methods that required prior knowledge

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                     Page 17 of 19
                   or manual operations. The SA-based multi-omics integration module automatically
                   assigns weights to each omic, producing a final integrated feature representation.
                     Our experimental results on BRCA , KIRP, and KIRC demonstrate the superior classi-
                   fication performance of MoAGL-SA compared to classical methods and state-of-the-art
                   DL-based methods. One important obser vation is that DL-based methods like MoGCN
                   and MORONET did not consistently outperform classical methods on certain datasets.
                   This may be due to the fact that MoGCN relies on AEs for feature extraction, which can
                   be suboptimal for handling high-dimensional omics data, particularly for the KIRP and
                   KIRC datasets. Additionally, MORONET, while effective for multi-class classification,
                   appears less suited for binary classification tasks, which might explain its lower per-
                   formance. However, MoAGL-SA consistently achieved superior results across multiple
                   metrics. Its end-to-end learning approach, coupled with adaptive multi-omics integra-
                   tion, allowed it to handle complex data relationships more effectively, demonstrating its
                   advantage over both classical and DL-based methods.
                     Our case studies on BRCA demonstrate MoAGL-SA’s capacity to identif y biologically
                   relevant biomarkers associated with breast cancer subtypes, such as E2F8, KIF20A , and
                   DEPDC1B. The KEGG and GO enrichment analyses further validated the biological rel-
                   evance of these biomarkers, highlighting their roles in processes like cell adhesion and
                   metabolic pathways , which are known to be involved in cancer development.
                     Despite its promising results, MoAGL-SA has several limitations. First, the model
                   requires complete multi-omics data, limiting its use in cases with missing data. Future
                   work will explore imputation techniques or models that handle incomplete datasets.
                   Second, while MoAGL-SA performed well on BRCA , KIRP, and KIRC datasets, its gen-
                   eralizability to other cancers remains to be evaluated, requiring validation on independ-
                   ent datasets. Lastly, although the model identified potential biomarkers, further clinical
                   trials are needed to confirm their biological significance and relevance to cancer sub-
                   types . Addressing these limitations will be the focus of future research.
                   Conclusion
                   In this study, we propose MoAGL-SA , an advanced multi-omics integration framework
                   that combines graph learning, GCNs, and self-attention to enhance cancer subtype
                   classification. By automatically constructing sample relationship graphs and adaptively
                   weighting each omics, MoAGL-SA not only achieves outstanding classification accurac y
                   but also effectively identifies biologically relevant biomarkers. Its adaptive, end-to-end
                   design outperforms classical and DL-based methods, providing a promising tool for
                   advancing precision medicine through flexible multi-omics integration.
                   Author contributions
                   P.G. conceived the project. P.G., L.C., Q.H., and Y.L. designed and implemented the algorithms and models. L.C., S.G., Z.Z.
                   and L.Z. analyzed and interpreted the data. P.G. and L.C. drafted the manuscript. All authors approved the final article. All
                   authors have read and agreed to the published version of the manuscript.
                   Funding
                   This work was supported by the Natural Science Foundation of China (No. 82001987), the Postgraduate Research &
                   Practice Innovation Program of Jiangsu Province (No.KYCX23_2937), and the Innovation and Entrepreneurship Project of
                   Xuzhou Medical University Science Park (No. CXCYZX2022010).
                   Availability of data and materials
                   BRCA, KIRP and KIRC datasets were available in the TCGA ( https:// portal. gdc. cancer. gov/). MoAGL-SA is implemented in
                   Python and is available on GitHub (https:// github. com/ gpxzmu/ MoAGL- SA)

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                                 Page 18 of 19
                                Declarations
                                Ethics approval and consent to participate
                                Not applicable.
                                Consent for publication
                                Not applicable.
                                Competing interests
                                Not applicable.
                                Received: 18 July 2024   Accepted: 13 November 2024
                                References
                                 1.                                                         Tan K, Huang W, Hu J, Dong S. A multi-omics super vised autoencoder for pan-cancer clinical outcome endpoints
                                        prediction. BMC Med Inform Decis Mak. 2020;20:1–9.
                                 2.                                                         Tomczak K, Czer wińska P, Wiznerowicz M. Review the cancer genome atlas (  TCGA): an immeasurable source of
                                        knowledge. Contemp Oncol/Współcz Onkol. 2015;2015(1):68–77.
                                 3.                                                         Günther OP, Chen V, Freue GC, Balshaw RF, Tebbutt SJ, Hollander Z, Takhar M, McMaster WR, McManus BM, Keown
                                        PA, etal. A computational pipeline for the development of multi-marker bio-signature panels and ensemble classi-
                                        fiers. BMC Bioinform. 2012;13(1):1–18.
                                 4.                                                         Kim D, Li R, Dudek SM, Ritchie MD. Athena: identifying interactions between different levels of genomic data associ-
                                        ated with cancer clinical outcomes using grammatical evolution neural network. BioData Mining. 2013;6(1):1–14.
                                 5.                                                         Singh A, Shannon CP, Gautier B, Rohart F, Vacher M, Tebbutt SJ, Lê Cao K-A. Diablo: an integrative approach for
                                        identifying key molecular drivers from multi-omics assays. Bioinformatics. 2019;35(17):3055–62.
                                 6.                                                         Li J, Xie L, Xie Y, Wang F. Bregmannian consensus clustering for cancer subtypes analysis. Comput Methods Pro-
                                        grams Biomed. 2020;189: 105337.
                                 7.                                                         Menyhárt O, Győr ffy B. Multi-omics approaches in cancer research with applications in tumor subtyping, prognosis,
                                        and diagnosis. Comput Struct Biotechnol J. 2021;19:949–60.
                                 8.                                                         Huang Z, Zhan X, Xiang S, Johnson TS, Helm B, Yu CY, Zhang J, Salama P, Rizkalla M, Han Z, etal. Salmon: sur vival
                                        analysis learning with multi-omics neural networks on breast cancer. Front Genet. 2019;10:166.
                                 9.                                                         Yang H, Chen R, Li D, Wang Z. Subtype-GAN: a deep learning approach for integrative cancer subtyping of multi-
                                        omics data. Bioinformatics. 2021;37(16):2231–7.
                                 10.                                                         Zhao L, Dong Q, Luo C, Wu Y, Bu D, Qi X, Luo Y, Zhao Y. Deepomix: A scalable and interpretable multi-omics deep
                                        learning framework and application in cancer sur vival analysis. Comput Struct Biotechnol J. 2021;19:2719–25.
                                 11.                                                         Kipf TN, Welling M. Semi-super vised classification with graph convolutional networks. arXiv preprint arXiv : 1609.
                                        02907 2016.
                                 12.                                                         Sun F, Sun J, Zhao Q. A deep learning method for predicting metabolite-disease associations via graph neural
                                        network. Brief Bioinform. 2022;23(4):266.
                                 13.                                                         Wang T, Shao W, Huang Z, Tang H, Zhang J, Ding Z, Huang K. Moronet: multi-omics integration via graph convolu-
                                        tional networks for biomedical data classification. bioRxiv, 2020–07 2020.
                                 14.                                                         Li X, Ma J, Leng L, Han M, Li M, He F, Zhu Y. MOGCN: a multi-omics integration method based on graph convolu-
                                        tional network for cancer subtype analysis. Front Genet. 2022;13: 806842.
                                 15.                                                         Yang B, Yang Y, Wang M, Su X. MRGCN: cancer subtyping with multi-reconstruction graph convolutional network
                                        using full and partial multi-omics dataset. Bioinformatics. 2023;39(6):353.
                                 16.                                                         Vaswani A, Shazeer N, Parmar N, Uszkoreit J, Jones L, Gomez AN, Kaiser Ł, Polosukhin I. Attention is all you need. Adv
                                        Neural Inform Process Syst. 2017. https:// doi. org/ 10. 48550/ arXiv. 1706. 03762.
                                 17.                                                         Xu P, Zhu X, Clifton DA. Multimodal learning with transformers: A sur vey. IEEE Trans Pattern Anal Mach Intell.
                                        2023;45(10):12113–32.
                                 18.                                                         Luong M-T, Pham H, Manning CD. Effective approaches to attention-based neural machine translation. arXiv pre -
                                        print arXiv : 1508. 04025 2015.
                                 19.                                                         Khattar D, Kumar V, Varma V, Gupta M. Hram: A hybrid recurrent attention machine for news recommendation.
                                        In: Proceedings of the 27th ACM international conference on information and knowledge management; 2018. p.
                                        1619–22.
                                 20.                                                         Mnih V, Heess N, Graves A, etal. Recurrent models of visual attention. Adv Neural Inform Process Syst. 2014;27.
                                 21.                                                         Srivastava P, Bej S, Yordanova K, Wolkenhauer O. Self-attention-based models for the extraction of molecular interac-
                                        tions from biological texts. Biomolecules. 2021;11(11):1591.
                                 22.                                                         Ma B, Meng F, Yan G, Yan H, Chai B, Song F. Diagnostic classification of cancers using extreme gradient boosting
                                        algorithm and multi-omics data. Comput Biol Med. 2020;121: 103761.
                                 23.                                                         Liu X, Gao F, Zhang Q, Zhao H. Graph convolution for multimodal information extraction from visually rich docu-
                                        ments. arXiv preprint arXiv : 1903. 11279 .2019
                                 24.                                                         Yu W, Lu N, Qi X, Gong P, Xiao R. Pick: processing key information extraction from documents using improved graph
                                        learning-convolutional networks. In: 2020 25th International conference on pattern recognition (ICPR); 2021. p.
                                        4363–70. IEEE.
                                 25.                                                         Ouyang D, Liang Y, Li L, Ai N, Lu S, Yu M, Liu X, Xie S. Integration of multi-omics data using adaptive graph learning
                                        and attention mechanism for patient classification and biomarker identification. Comput Biol Med. 2023;164:
                                        107303.

Cheng et al. BMC Bioinformatics          (2024) 25:364                                                                                                                                                                                      Page 19 of 19
                                 26.                                                         Jiang B, Zhang Z, Lin D, Tang J, Luo B. Semi-super vised learning with graph learning-convolutional networks. In:
                                         Proceedings of the IEEE/CVF conference on computer vision and pattern recognition; 2019. p. 11313–20.
                                 27.                                                         Huang H, Song Y, Wu Y, Shi J, Xie X, Jin H. Multitask representation learning with multiview graph convolutional
                                         networks. IEEE Trans Neural Netw Learn Syst. 2020;33(3):983–95.
                                 28.                                                         Fisher A, Rudin C, Dominici F. All models are wrong, but many are useful: Learning a variable’s importance by study-
                                         ing an entire class of prediction models simultaneously. J Mach Learn Res. 2019;20(177):1–81.
                                 29.                                                         Ye L, Guo L, He Z, Wang X, Lin C, Zhang X, Wu S, Bao Y, Yang Q, Song L, etal. Upregulation of E2F8 promotes cell pro                                                                                                                              -
                                         liferation and tumorigenicity in breast cancer by modulating G1/S phase transition. Oncotarget. 2016;7(17):23757.
                                 30.                                                         Nakamura M, Takano A, Thang PM, Tsevegjav B, Zhu M, Yokose T, Yamashita T, Miyagi Y, Daigo Y. Characterization
                                         of KIF20A as a prognostic biomarker and therapeutic target for different subtypes of breast cancer. Int J Oncol.
                                         2020;57(1):277–88.
                                 31.                                                         Li T-F, Zeng H-J, Shan Z, Ye R-Y, Cheang T-Y, Zhang Y-J, Lu S-H, Zhang Q, Shao N, Lin Y. Overexpression of kinesin
                                         super family members as prognostic biomarkers of breast cancer. Cancer Cell Int. 2020;20:1–16.
                                 32.                                                         Boudreau HE, Broustas CG, Gokhale PC, Kumar D, Mewani RR, Rone JD, Haddad BR, Kasid U. Expression of BRCC3, a
                                         novel cell cycle regulated molecule, is associated with increased phospho-ERK and cell proliferation. Int J Mol Med.
                                         2007;19(1):29–39.
                                 33.                                                         Brown KA. Metabolic pathways in obesity-related breast cancer. Nat Rev Endocrinol. 2021;17(6):350–63.
                                 34.                                                         Li D-M, Feng Y-M. Signaling mechanism of cell adhesion molecules in breast cancer metastasis: potential therapeu-
                                         tic targets. Breast Cancer Res Treat. 2011;128:7–21.
                                 35.                                                         Ayala GE, Dai H, Powell M, Li R, Ding Y, Wheeler TM, Shine D, Kadmon D, Thompson T, Miles BJ, etal. Cancer-related
                                         axonogenesis and neurogenesis in prostate cancer. Clin Cancer Res. 2008;14(23):7593–603.
                                 36.                                                         Fahmy HM, Ismail AM, El-Feky AS, Serea ESA. Elshemey WM Plasma membrane proteins: A new probe for the charac-
                                         terization of breast cancer. Life Sci. 2019;234: 116777.
                                Publisher’s Note
                                Springer Nature remains neutral with regard to jurisdictional claims in published maps and institutional affiliations.

