                                                                                                                                                                                                                                  Brief ings        in        Bioinformatics,       2022,      23(4),       1–10
                                                                                                                                                                                                                                                           https://doi.org/10.1093/bib/bbac231
                                                                                                                                                                                                                                    Advance       access       publication       date:       14       June       2022
                                                                                                                                                                                                                                                                            Problem        Solving        Protocol
Multiphysical                   graph                   neural                   network                   (MP-GNN)                   for
COVID-19                   drug                   design
Xiao-Shuang        Li,                   Xiang        Liu,         Le        Lu,   Xian-Sheng        Hua,                      Yi n g            C  h iand       Kelin        Xia
Corresponding      author:      Kelin      Xia,     xiakelin@ntu.edu.sg
Abstract
Graph           neural          networks          (GNNs)          are           the          most          promising           deep          learning           models          that          can          revolutionize           non-Euclidean          data          analysis.
However,               their                full                potential                is                severely                curtailed                by                poorly                represented                molecular                graphs                and                features.               Here,               we                propose                a
multiphysical           graph           neural           network           (MP-GNN)           model           based           on           the           developed           multiphysical           molecular           graph           representation           and
featurization.     All      kinds      of      molecular      interactions,     between      different      atom      types      and      at      different      scales,     are      systematically      represented
by        a        series        of        scale-specif ic        and        element-specif ic        graphs        with        distance-related        node        features.       From        these        graphs,       graph        convolution
network           (GCN)           models           are           constructed           with           specially           designed           weight-sharing           architectures.          Base           learners           are           constructed           from
GCN           models           from           different           elements           at           different           scales,          and           further           consolidated           together           using           both           one-scale           and           multi-scale
ensemble        learning        schemes.       Our        MP-GNN        has        two        distinct        properties.       First,       our        MP-GNN        incorporates        multiscale        interactions        using
more            than            one            molecular            graph.           Atomic            interactions            from            various            different           scales            are            not            modeled            by            one            specif ic            graph            (as            in
traditional            GNNs),           instead            they            are            represented            by            a            series            of            graphs            at            different            scales.           Second,           it            is            free            from            the            complicated
feature       generation       process       as       in       conventional       GNN       methods.      In       our       MP-GNN,      various       atom       interactions       are       embedded       into       element-
specif ic          graph          representations          with          only          distance-related          node          features.         A          unique          GNN          architecture          is          designed          to          incorporate          all
the       information       into       a       consolidated       model.      Our       MP-GNN       has       been       extensively       validated       on       the       widely       used       benchmark       test       datasets
from       PDBbind,      including       PDBbind-v2007,      PDBbind-v2013       and       PDBbind-v2016.      Our       model       can       outperform       all       existing       models       as       far       as
we        know.       Further,       our        MP-GNN        is        used        in        coronavirus        disease        2019        drug        design.       Based        on        a        dataset        with        185        complexes        of        inhibitors
for         severe         acute         respiratory         syndrome         coronavirus         (SARS-CoV/SARS-CoV-2),        we         evaluate         their         binding         aff inities         using         our         MP-GNN.
It         has         been         found         that         our         MP-GNN         is         of         high         accuracy.        This         demonstrates         the         great         potential         of         our         MP-GNN         for         the         screening         of
potential      drugs      for      SARS-CoV-2.     Availability:      The      Multiphysical      graph      neural      network      (MP-GNN)      model      can      be      found      in      https://github.
com/Alibaba- DAMO- DrugAI/MGNN.      Additional        data        or        code        will        be        available        upon        reasonable        request.
Keywords:                   Graph        neural        network,      Graph        representation        and        featurization,       Protein–ligand        binding,      Drug        design,      Ensemble        learning
Introduction                                                                                                                                                                large-scale                network                data                with                the                main                focus                of                pre-
So              far,             more              than              262              million              infections              and              5              million  dicting           the           properties           of           new           nodes           or           edges           within           the
fatalities                      haves                      been                      succumbed                      to                      the                      new                      severenetwork.      Recently,      GNNs       have       been       used       in       biomolecular
acute                   respiratory                   syndrome                   coronavirus                   (SARS-CoV-2)                                                 data      analysis      and      achieved      great      performance      for      various
in                the                coronavirus                disease                2019                (COVID-19)                pandemic                               steps           in           drug           design           and           discovery           [2–10].          Among           these
which         has         swept         across         all         213         countries         and         territories.                                                   models,              AquaSol               [2]               uses               directed               acyclic               graph               based
The                 signif icance                 of                 designing                 eff icient                 antibodies                 and                    recursive       neural       networks       to       predict       molecular       solubility.
drugs        for        COVID-19        cannot        be        overemphasized.      Artif icial                                                                            In           DeepVS           [3],          an           effective           atom           context           representation
intelligence-based                           models                           have                           demonstrated                           great                   is                employed                that                can                take                into                consideration                protein–
power            in            various            steps            in            drug            design            [1].           Among            these                    ligand                         complex                         properties.                        An                         integrated                         model                         of
models            are            graph            neural            network            (GNN)            models,           which                                             the                  compound-structure-based                  GNN                  and                  the                  protein-
are         end-to-end         learning         models         that         take         in         a         molecular                                                     sequence-based                   convolution                   neural                   network                   (CNN)                   is
graph          representation          and          directly          output          the          prediction.                                                              developed               for               compound               protein               interactions               [7].              GAN
Originally,                    GNNs                     were                     developed                     for                     the                     analysis                     ofmodel                   is                   introduced                   for                   chemical                   stability                   prediction
Xiao-Shuang      Li      Xiao-Shuang      Li      is      a      PhD      student      in      Shanghai      Jiao      Tong      University,     and      also      a      research      intern      at      the      Alibaba      DAMO      Academy.
Xiang      Liu      Xiang      Liu      is      a      PhD      student      from      Nankai      University      in      China.     He      is      a      visiting      student      in      Nanyang      Technological      University      from      December      2019      to      June
2020.
Le      Lu      Le      Lu      is      IEEE      Fellow.     He      is      the      Head      of      Medical      AI      research      and      development      of      Alibaba      Group,     and      also      Senior      Director      of      DAMO      Academy      USA.
Xian-Sheng      Hua      Xian-Sheng      Hua      is      IEEE      Fellow.     He      is      the      head      of      CityBrain      Lab      and      leads      the      Artitificial      Intelligence      Center      of      DAMO      Academy      in      Alibaba      Group.
Ying      Chi      Ying      Chi      is      the      team      leader      of      Drug      Discovery      Intelligence      at      the      Alibaba      DAMO      Academy.     She      did      PhD      in      Imperial      College      London      and      Postdoctoral
research      in      Oxford      University      in      UK.     Her      current      research      and      development      interest      is      all      types      of      AI      methods      for      various      drug      discovery      problems,     e.g.virtual
screening,     protein      and      immunity      related.
Kelin      Xia      Kelin      Xia      is      an      assistant      professor      at      School      of      Physical      and      Mathematical      Sciences,     Nanyang      Technological      University,     Singapore.     His      research      interests
are      topological      data      analysis,     molecular-based      mathematical      biology      and      machine      learning.
Received:       March      7,     2022.     Revised:       April      24,     2022.     Accepted:       May      18,     2022
©      The      Author(s)      2022.     Published      by      Oxford      University      Press.     All      rights      reserved.     For      Permissions,     please      email:      journals.permissions@oup.com

2         |       Li        et        al.
in                      DeepChemStable                      [8].                     A                      convolution                      spatial                      graph                                  multiscale       effects,      the       scale       (or       resolution)       related       kernel
embedding                          layer                          (C-SGEL)                          based                          graph                          convolution                                    functions                  are                  used.                 Among                  them,                 the                  most                  common
network             (GCN)             model             is             developed             for             molecular             prop-                                                                         ones                 are                 the                 generalized                 exponential                 kernels                 and                 the
erty                prediction                [9].               A                structure-aware                interactive                GNN                                                                  generalized        Lorentz        kernels.      For        two        atoms                                                                          ri     and           rj,      their
is                  designed                  to                  learn                  essential                  long-range                  interactions                                                     atomic               interaction               can               be               modeled               by               the               generalized
among       atoms       and       to       fully       utilize       biomolecular       structural                                                                                                               exponential         kernel         as         follows:
information                      [11].                     GNN                      models                      have                      also                      been                      used
in                          drug–target                          aff inity                          prediction                          [11–15],                         antibiotic                                                                                /Phi1(||ri       −                     rj ||;    η)            =         e−        (||ri −          rj ||/η)   κ         ,                                                                                                                                                                                                    (1)
discovery                [16],               protein–protein                binding                aff inity                change
upon                 mutation                 [17]                 and                 various                 other                 drug                 discovery                                              or         by         generalized         Lorentz         kernel         as
and         development         [18].
        Even                    though                    GNNs                    have                    shown                    great                    promise                    for                                                               /Phi1(||ri       −                     rj ||;    η)            =                                                                                                                                      1
drug               design,              their               full               potential               has               been               hindered               by                                                                                                                                            1     +     (||ri       −                     rj ||/η)     κ            .                                                                                                                                                    (2)
the                    ineff icient                    graph                    topological                    representations                    and
featurization.                Currently,                most                 biomolecular                 GNNs                 use                                                                               Here,               η      is           scale           (or           resolution)           parameter,          and                                               κ      is           order
the          covalent-bond-based          graph          representation,         which          is                                                                                                               parameter,      which       is       usually       taken       as       2.      Based       on       rigidity–
to          model          a          molecule          as          a          graph          with          atoms          represented                                                                           f   lexibility                model,               we                can                def ine                the                node                importance
as              nodes              and              covalent              bonds              as              edges.            Node              and              edge                                           using         rigidity         index         as         follows:
features                     are                     then                     generated                     from                     different                     types                     of
physical,                chemical                 and                 biological                 properties.                However,                                                                                                                             μ(           ri ;    η)            =     ∑  wj /Phi1(||ri       −                     rj ||;    η)      ,                                                                                                                                                                                          (3)
these                         covalent-bond-based                         molecular                         topologies                         fail                                                                                                                                                   j
to                        eff iciently                        characterize                        non-covalent                        interactions,
which                      can                      be                      of                      great                      importance                      for                      biomolecular             where                   wj       is         an         atomic         type-dependent         weight.        Note         that
complexes,                   including                    protein–protein                    complexes,                   pro-                                                                                   kernel               functions               with               different               scale               values               will               focus
tein–ligand                           complexes,                          protein–DNA/RAN                           complexes                                                                                    on              atomic              interactions              at              different              scales.             If              a              small                                  η
and            DNA/RNA–ligand            complexes.           To            alleviate            the            prob-                                                                                            value          is          used,        the          kernels          characterize          only          strong          cova-
lem,                        a                         f ixed                         cutoff-distance-based                         molecular                         graph                                       lent             interactions             with             the             values             for             other             interactions
representation           has           been           developed.         However,         molecular                                                                                                              at                  longer                  distance                  as                  (nearly)                  0.                In                  contrast,                under                  a
interactions                 are                 usually                 of                 different                 scales.                The                 f ixed                                          larger                η     value,        relatively         weaker         interactions         will         also         be
cutoff-distance-based                   topology                   tends                   to                   miss                   a                   great                                                 included.       Node        importance        will        vary        with        scale        values        in
amount        of        information        and        it        is        nontrivial        to        identify        the                                                                                        a         similar         way.
‘best’              cutoff              distance.             Currently,             the              bottleneck              for              the                                                                       In         our         scale-specif ic         graph         representations,       molecules
design        of        eff icient        molecular        GNN        models        is        the        suitable                                                                                                are           modeled           by           a           series           of           graphs           systematically           gener-
topological           representations           and           featurization           that           char-                                                                                                       ated             from             different             scales.           Mathematically,           a             fully             con-
acterize         the         multiphysical         properties         of         biomolecules.                                                                                                                   nected            molecular            graph            is            generated            with            scale-related
        Here,                          we                           develop                           multiphysical                           molecular                           graph                          weight              value,             i.e.             the              atomic              interaction              from              Eq.(1)or
representations                and                featurization.               Based                on                them,               we                                                                     Eq.(2),            on              each              edge.            Based              on              the              scale-specif ic              graph
propose       a       multiphysical       graph       neural       network       (MP-GNN)                                                                                                                        representation,            the             normalized             adjacent             matrix             can             be
model.                    Our                      MP-GNN                     employs                     an                     ensemble                      learning                                          def ined         as
scheme             to             incorporate             both             scale-specif ic             GNN             models                                                                                                                                                             {/Phi1(
and                 element-specif ic                 GNN                 models.                It                 has                 been                 found                                                                                             A(i,     j)       =ˆ                     ||ri       −                     rj ||;    η)      ,                                 i       ̸=         j
that                     our                     MP-GNN                     model                     can                     deliver                     state-of-the-art                                                                                                                     0,                                                                                                                                                                                                                                  i       =         j,                                                                                                                                                                                  (4)
results                      for                      protein–ligand                      binding                      aff inity                      prediction
and               achieve               extremely               high               accuracy               in               SARS-CoV               BA                                                             and         the         normalized         degree         matrix         can         be         def ined         as
dataset,           which            contains            185                                          Mpro-ligand            complexes            and                                                                                                                                                  {μ(           ri ;    η)      ,                                 i       =         j
their         experimental         binding         aff inities.
                                                                                                                                                                                                                                                                          D(i,     j)       =ˆ
                                                                                                                                                                                                                                                                                                           0,                                                                                                                       i       ̸=         j.                                                                                                                                                                                                                                        (5)
Results
Physically,                          atomic                           interactions                           within                           and                           between                              In       this       way,      the       scale       effects       are       incorporated       into       molec-
molecules        are        of        various        types,       ranging        from        strong        ones                                                                                                  ular         graph         representation.
such                 as                 covalent                 bonds,                disulf ide                 bonds,                ionic                 bonds,                                                     Further,        we         propose         a         new         type         of         node         feature         vector
hydrogen            bonds,           to            relatively            weaker            ones,           such            as            van                                                                     that           is           solely           dependent           on           atomic           interaction           function
der          Waals          forces,        electrostatic          interactions,        hydrophobic                                                                                                               /Phi1.            For             the  i-th             node,            an            n-th             dimensional             node             feature
and                         hydrophilic                         effects.                        Mathematically,                        the                         atomic                                        vector                  vi (η)            =       (vi1 (η)      ,     vi2 (η)      ,     ...,     vin (η))               is         def ined         as         follows:
interaction        between        two        atoms        with        coordinates                                                                                        ri    and            rj
can               be               def ined               as               an               interaction               function                            /Phi1(||ri           −                          rj ||)       vik (η)            =     ∑            χ(        xk−          1         ≤                 /Phi1(||ri       −                     rj ||;    η)      <               xk ),     k       =         1,     2..,     n.                              (6)
with                ∥ri               −                              rj ∥the                     Euclidean                     distance.                    To                     model                     the                                  j=1

                                                                                                                                                                                                                                                   Multiphysical        graph        neural        network        (MP-GNN)                                                       |      3
Here,      we       assume       all       atomic       interactions                                                              /Phi1(||ri    −                 rj ||;    η)are                   Cl-graph          will          be          meaningless.         However,         the          Cl          atom          can
within        the        region                          [0,     xmax ],       which        is        equally        divided        into                                           n                be             important             for             ligand             properties.            So             we             can             consider
intervals                      {(xk−          1 ,     xk );     k              =                 1,     2,     ..,     n}with x0               =                 0and  xn               =           multiple-element                    graphs,                   such                    as                    CNCl-graph,                   COCl-
xmax.       The        indicator        function                                           χ     equals        to        1        if        the        following                                    graph,        etc.
condition              is              satisf ied              and              0              otherwise.             Mathematically,
the             node             vector             is             the             frequencies             (or             the             numbers)             of                                  Multiphysical          graph          neural          network
atomic         interactions         within         a         certain         range.                                                                                                                 In                  our                  MP-GNN                  model,                 a                  series                  of                  scale-specif ic                  and
       Finally,           molecules            are            usually            of            various            sizes,           which                                                            element-specif ic             graphs             are             generated             from             molecules.
will            result            in            different-sized            molecular            graphs.           To            facil-                                                              From          each          graph,         a          GNN          architecture          is           constructed.         To
itate            weight-sharing            among            different            molecular            graphs,                                                                                       signif icantly                   reduce                   the                   learning                   parameters,                  weight-
we             use             node             importance                           μ      as             in             Eq.            (3)             to             remove             less     sharing                     schemes                     and                     ensemble                     learning                     models                     are
important            extra            nodes,           so            that            a            same-sized            molecular                                                                   considered.          Molecular           structural           topologies            are           of           great
graph         is         obtained.                                                                                                                                                                  importance                       for                       their                       functions.                      Various                       quantitative
                                                                                                                                                                                                    structure–activity/property                                 relationship                                 (QSAR/QSPR)
Element-specif ic          graph          for          GNN                                                                                                                                          have                      been                      developed                      to                      establish                      relations                      between
Other                than                scale                effects,               element                types                are                the                other                        molecular                 groups,                motifs,                conserved                 regions,                domains
key                     factor                     for                     multiphysical                    atomic                     interactions.                   For                          and                     other                     molecular                     topologies                     with                     their                     functions
instance,                      carbon                       atoms                       are                       usually                       associated                       with               [31–33].                  In                   GNN                   models,                  weight-sharing                   schemes                   are
hydrophobic                     interactions,                   while                     nitrogen                     and                     oxygen                                               used                        to                        characterize                        common                        molecular                        topologies,
atoms            are            correlated            to            hydrophilic            interactions            and/or                                                                           as                   similar                   structure                   topologies,                  def ined                   by                   the                   same
hydrogen                      bonds.                    To                      enable                      a                      systematic                      description                      weights                       in                       GNNs,                     tend                       to                       induce                       similar                       functions.
of                      atomic                      interactions,                     we                      consider                      element-specif ic                                       Moreover,                          weight-sharing                           schemes                           can                           signif icantly
graph          representations          [19].         Recently,         the          combination          of                                                                                        reduce       parameters       and       network       complexities.      In       our       MP-
element-specif ic            representations            and            machine            learning                                                                                                  GNN,        we         use         the         same         weight         schemes         among         the         same
models             has             achieved             great             success             in             drug             design             [19–                                               scale-specif ic         and         element-specif ic         graph.       We         also         allow
30].                       More                        recently,                       an                        element-specif ic                        GNN                        model          to               use               same               weight               schemes               among               relatively               similar
has                        been                        developed                        and                        has                        achieved                        state-of-the-         element-specif ic               graphs,              to               reduce               computational               cost
art                  performance                  in                  quantitative                  toxicity                  analysis                  and                                         and        when        there        is        relatively        less        training        data.       Ensemble
solvation         prediction         [10].                                                                                                                                                          learning             models             use             multiple             base             learning             algorithms
       The                          essential                          idea                          for                          element-specif ic                          represen-              to                 boost                 the                 performance                 of                 the                 prediction.                Here,                we
tations                    is                    to                    decompose                    a                    molecule                    into                    a                    series                    ofconsider                    two                    types                    of                    ensemble                    learning,                   i.e.                   single-
atom-sets,           which            composed            of            certain            specif ic            types            of                                                                 scale             (one-scale)             stacking             and             multiscale             stacking.            The
elements.               In                general,               a                protein                molecule                is                composed                                         one-scale             stacking             ensemble             model             is             used             to             alleviate
of                 roughly                 f ive                 most                 important                 elements,                denoted                 as                                 the            impact            of            randomness            caused            by            initialization.          The
EP        =             [C,            N,            O,            S,            H].            A             DNA             or             RNA             also             have             f ive             mostmultiscale             stacking             is             for             boosting             the             performance             by
important                   elements,                 denoted                   as                               ED          =                   [C,                 N,                 O,                 P,                 H].the        consolidation        of        base        learners        that        focus        on        different
For                ligands                or                chemical                molecules,               they                tend                to                have                         scales         and         have         less         overlap.
more                 types                 of                 elements.                Here,                we                 consider                 only                 nine
types               of               most-commonly               used               ones,             and               denote               as                                EL                   MP-GNN          for          COVID-19          drug          design
=                  [C,                 N,                 O,                 S,                 H,                 F,                 Cl,                 Br,                 I].                 In                  general,                 an                  element-MP-GNN         for         protein–ligand         interactions
specif ic                      GNN                      model                      contains                      a                      series                      of                      molecularRecently,        a         series         of         topological         models         have         been         devel-
graphs           that           are           constructed           based           on           different           element                                                                        oped            for            the            characterization            of            protein–ligand            interac-
types.                  For                   instance,                  a                   protein                   can                   be                   represented                   by  tions                  and                  have                  achieved                  great                  successes                  [25,             29,          30].
a                    series                    of                    element-specif ic                    graphs,                  including                    single-                             The              essential              idea              of              these              models              is              to              def ine              special
element        graphs        (C-graph,       N-graph,       O-graph,       S-graph        and                                                                                                       matrices          that          focus          on          interactions          between          the          protein
H-graph),            double-element             graphs             (CN-graph,            CO-graph,                                                                                                  and           the           ligand,         instead           of           interactions           within           either           the
CS-graph,      CH-graph,      NO-graph,      NS-graph,      NH-graph,      OS-                                                                                                                      protein            or            the            ligand,           and            to            construct            molecular            topo-
graph,              OH-graph               and               SH-graph),              three-element               graphs                                                                             logical         models         based         on         these         matrices         [25].
and                 other                 graphs                 with                 more                 types                 of                 elements.                Each                           Mathematically,      we        can        set        the        protein–ligand        interac-
element-specif ic                       graph                       characterizes                       certain                       type                       of                                 tion         matrix                    M       as         follows:
atomic            interactions.           Note            that            the            all-atom            graph            as            in                                                                                                       ⎧
previous            GNN            models            is            just            a            special            case            of            element-                                                                                            ⎪
                                                                                                                                                                                                                                                     ⎪     /Phi1(∥ri       −                     rj ∥;    η) ,if        ri        ∈                  RP ,     rj         ∈                  RL
specif ic            graph.           Normally,           we            do            not            need            to            use            all            the                                                                                 ⎨
                                                                                                                                                                                                               M(mi ,     mj )       =               ⎪                                                       or        ri        ∈                  RL ,     rj         ∈                  RP(7)
combinations          [25,                                  29,        30].         To          balance          the          computational                                                                                                          ⎪
                                                                                                                                                                                                                                                     ⎩     ∞                       ,                                                                                                                                                                                                              otherwise.
cost                      and                      model                      accuracy,                     we                      usually                      only                      consider
the               element-specif ic               graphs               with               suff icient               amount               of
atoms.               For                instance,              ligand                molecules                may                contain                Cl                                          Here,              ri     and            rj     are           coordinates           for           the                         i-           and      j-th           atoms,
atom          but          they          usually          have          only          one          or          two          Cl          atoms.         A                                            and            mi        and            mj        are            their            indices            in            the            matrix.           Two            sets

4        |       Li        et        al.
RP        and            RL        are            atom            coordinate            sets            for            protein            and            lig-                                          training                 set                 is                 obtained                 by                 the                 ref ined                 set                 minus                 the
and,               respectively.               Note                that                only                interactions                between                                                         core         set.
protein               atoms               and               ligand               atoms               are               considered,             while                                                           To              test              the              performance              of              our              model              for              COVID-19
interactions                  between                  atoms                  within                  either                  the                  protein                                             drug          design,         we          consider          a          SARS-CoV          BA          dataset,         which
or               the               ligand               are               ignored               by               setting               their               distances               as                  contains      185Mpro-ligand      complexes      and      their      experimen-
∞      ,              i.e.              an               inf initely               large               value.              Other               than               the               gener-             tal       binding       aff inities.      Among       the       185       ligands,      there       are       44
alized                 kernel                 functions                 as                 in                 Eqs.                (1)                 and                 (2),                we                 canX-ray             crystal             structures             and             the             rest             are             in             2D             SMILES
also            def ine            Euclidean-distance            and            electrostatics-based                                                                                                   strings.                The                 software                 MathPose                 is                 used                 to                 predict                 3D
atomic         interaction         functions         as                                                                                                                                                structures          of          those          2D          ligands          and          generate          the          binding
                                                                                                                                                                                                       complexes           of           all           185           ligands           with                                Mpro.          To           carry           out           the
                                                      /Phi1(||ri       −                     rj ||)       =                                                                           1                validation,          we           randomly           split           the           SARS-CoV           BA           set           into
                                                                                                    ||ri       −                     rj ||κ            ,                                                                                                                                                                                                                                 (8)f ive            non-overlapped            folds.           In            each            task,           our            MP-GNN            is
                                                                                                                                                                                                       trained          on          the          part          of          SARS-CoV          BA          dataset          in          conjunc-
and                                                                                                                                                                                                    tion             with             the             PDBbind-v2019             set.            More             specif ically,            one
                                                                                                                                                                                                       fold       (or       division)       is       used       as       the       validation       set       in       each       task,
                                       /Phi1(||ri       −                     rj ||)       =                                                                                                                                                   1()    ,                                                                                                                                                        (9)and           the           rest           four           folds           are           combined           with           the           PDBbind-
                                                                                                                             cqi qj                                                                    v2019        general        set        to        form        the        training        set.       No        pre-train        is
                                                                                     1      +       exp             −     ∥ri −          rj ∥                                                          done         before         training.
where                 qi      and             qj      are            partial            charges            for            the                   i-th            and           j-th                     Benchmark         tests         for         MP-GNN
atoms,       and         parameter                                         c   is         a         constant         value.       All         the         three                                        More         than         40         different         scoring         functions         or         models         have
types             of             atomic             interactions             are             considered             in             our             MP-                                                 been             extensively             tested             on             the             three             PDBbind             datasets.
GNN.                                                                                                                                                                                                   Figure         2              shows                 the                 comparison                 between                 our                 MP-GNN
       Further,         element-specif ic          graphs          are          constructed          only                                                                                              and         the         other         models.       The         upper         part         depicts         the         overall
between                    protein                    atoms                    and                    ligand                    atoms.                   As                    stated                  performance,                   and                    our                    method                    is                    marked                    in                    red.                   All
above,      a       protein       molecule       is       usually       composed       of       roughly                                                                                                results            are            measured            by            Pearson            correlation            coeff icient
f ive            important            elements                                     EP         =            [C,           N,           O,           S,           H]            and            ligands   (denoted                as                  Rp).               Our                method                stays                ahead                of                all                other
composed          of          nine          types                               EL        =          [C,        N,        O,        S,        H,        F,        Cl,        Br,        I].        We  works            for             all            three             datasets,           except            second             to            TopBP            in
generate                  a                  series                  of                  element-specif ic                  bipartite                  graphs                                          PDBbind-v2016.         More          specif ically,         the          current          best                                                                           Rp       on
in                  our                  MP-GNN.                Each                  bipartite                  graph                  is                  composed                  of               PDBbind-v2007            is            0.831            achieved            by            FPRC            [37],          while            on
two                sets                of                same-typed                atoms                with                one                set                from                the              PDBbind-v2013             and             PDBbind-v2016             are             0.808             and             0.861
protein                  and                  the                  other                  from                  the                  ligand.                Edges                  can                  beboth         achieved         by         TopBP         [25].        Our         MP-GNN         surpasses         the
only                formed                between                the                two                sets                (thus                the                name                of              current       best       results       on       PDBbind-v2013       by       2%       and       stays       in
the            bipartite            graph),           and            are            determined            by            interaction                                                                    line         with         the         current         best         results         of         PDBbind-v2007         with
matrix                  as                  in                  Eq.                 (7).                 In                  general,                 when                  the                  multiscalea       slight       advantage.      On       PDBbind-v2016,      it       is       1%       lower       than
kernel           functions           are           used,          a           total           of                           36           =              4       ∗               9types           of     TopBP.         In          the          line          chart,         the          right          part          where                                   Rp        over          0.6
bipartite        graphs        are        generated        without        the        consideration                                                                                                     is             dense,            and             the             clear             ranking             is             displayed             below.            It             is
of                      H                      atoms.                     Moreover,                     four                      different                      types                      of                      scaleworth           mentioning           that           our           method           achieves           signif icant
(or                   resolution)                   parameters                   are                   used,                  i.e.              η                             =2,                  5,                  10improvement                   on                   the                   hardest                   dataset,                 PDBbind-v2013,
and                20                Å.Figure         1           illustrates                the                general                architecture                                                    which                      has                      a                      more                      unbalanced                      distribution                      between
of                    our                    MP-GNN                    model                    for                    protein–ligand                    interaction                                   training        and        test        set        (See                              Table         1).            Figure         3            demonstrates
analysis.        More         details         of         MP-GNN         model         can         be         found         in                                                                          the          performance          for          two          stacking          schemes          and          learning
Method.                                                                                                                                                                                                rate               of               our               model               on               PDBbind-v2007.              A               more               detailed
       Datasets                    We                    consider                    three                    most                    commonly                    used                                 illustration           of           our           detailed           results           for           all           three           datasets
benchmark             datasets             for             protein–ligand             binding             aff inity                                                                                    can         be         found         in         Tables         S1         to         S3.
prediction,     including      PDBbind-v2007,     PDBbind-v2013      and
PDBbind-v2016.            All             the             datasets             used             in             this             paper             are                                                  MP-GNN         for         COVID-19         drug         design
shown            in              Table         1.          There            are            pre-train            sets,          training            sets                                                The          COVID-19          pandemic,         started          in          late          December          2019
and         test         sets         for         the         separated         experiments         on         PDBbind-                                                                                and         caused         by         new         SARS-CoV-2,        has         infected         more         than
v2007,       v2013        and        v2016.       There        are        intersections        between                                                                                                 262                  million                  individuals                  and                  has                  caused                  more                  than                  5
the              datasets,             so              the              pre-train              set              is              randomly              selected                                         million               fatalities               in               all               of               the               continents               and               over               213
from               the               non-intersected               samples               of               one               dataset.              The                                                  countries       and       territories       by       11       November       2021.      Currently,
union        of        these        three        datasets        is        4413.       For        PDBbind        v2007,                                                                                different            drug            targets            of            SARS-CoV-2,           such            as            the            main
the                pre-train                set                contains                1000                items                from                3114                non-                           protease              (Mpro,             also              called                          3CLpro),             papain-Like              protease
intersected         samples.        For         PDBbind         v2013,        1000         from         1455                                                                                           (PLpro),                 RNA-dependent                  RNA                  polymerase                  (RdRp),                 5’-to-
non-intersected          samples          and          for          PDBbind          v2016,         all          357                                                                                   3’                        helicase                        protein                        (Nsp13),                       have                        been                        investigated.
non-intersected                     complexes                     are                     used                     for                     pre-training.                                               Among                    them                    is                    the                    main                    protease,                   which                    is                    one                    of
The                  core                  set                  acts                  as                  the                  test                  set                  for                  evaluation.                 Thethe             best-characterized             targets             for             coronaviruses.            It             has

                                                                                                                                                                                                       Multiphysical        graph        neural        network        (MP-GNN)                                 |     5
Figure      1.     The      framework      of      the      protein–ligand      complex      binding      aff inity      prediction      system      for      drug      design      task.     A      complex      of      SARS-CoV-2      main      protease
inhibitor        is        used        as        an        example        here.       This        process        consists        of        three        steps:        (1)        generating        the        scale-specif ic        graphs        for        the        protein–ligand        complex,       (2)
processing       a       group       of       element-specif ic       graphs       with       multiphysical       graph       neural       network       for       22       repeat       experiments,      and       performing       one-scale       stacking
on       the       repeat       experiments       to       give       a       prediction       for       one       resolution       and       (3)       giving       a       f inal       decision       by       combining       multi-scale       predictions       with       multi-scale
stacking.      Nodes       f illed       or       outlined       in       red       are       from       the       ligand.
Ta   b   l   e                1   .A        summary        of        our        selected        datasets. mean(B)            refers        to        the        mean        atom        number        for        binding        sites,       and             mean(G)            refers        to        the
mean        atom        number        for        the        un-cropped        element-specif ic        graph.       The        ratio        between                                             mean(B)             and       mean(G)             describes        the        average
complexity        of        the        dataset
Name                                              Size                   Pre-train       size                  Descriptions                                                                                                                         mean(B)                   mean(G)                    mean(G)
                                                                                                                                                                                                                                                                                                         mean(B)
PDBbind       v2007       [36]                    1300                   1000                                  Ref ined       set.      Core       set       size       195.                                                                        583                       151                        0.259
PDBbind       v2013       [36]                    2959                   1000                                  Ref ined       set.      Core       set       size       195.                                                                        195                       56                         0.287
PDBbind       v2016       [36]                    4057                   358                                   Ref ined       set.      Core       set       size       285.                                                                        441                       108                        0.245
PDBbind       v2019       [36]                    17       652           –                                     General       set.                                                                                                                   432                       114                        0.264
SARS-CoV       BA       [34]                      185                    –                                     Inhibitors       of       SARS-CoV/SARS-CoV-2       main       protease       having                                                 583                       149                        0.256
                                                                                                               experimental       binding       aff inity.

6        |     Li        et        al.
Figure             2.   Comparison            with            recent            works            on            three            datasets,           including            topology-based            methods,           image-based            methods            and            traditional            molecular
descriptor-based          methods.         The          performances          of          other          models          are          taken          from          [25,                                34,     35].         The          upper          part          is          an          overall          comparison.         The          lower          part
is       a       clear       performance       ranking       of       works       with       Rp         higher       than       0.6       on       three       datasets.      All       results       are       measured       with       Rp .
Figure      3.        The      left      and      middle      box      charts      depict      the      range      of      performance      for      two      stacking      schemes.     The      line      chart      on      the      right      shows      the      decay      rule      for
learning       rate       and       the       Rp         curves       for       training       and       test       on       the       f irst       repeat       experiment       for       PDBbind-v2007       with       the       exponential       kernel,    η            =       10.
been           found           that           although           the           overall           sequence           identity                                                           model                on                this                special                dataset.               In                order                to                benchmark
between         SARS-CoV         and         SARS-CoV-2         is         just         80%,        the                                                       Mpro                     our                   method                   against                   MathDL                   [34],                 which                   is                   a                   leading
of            SARS-CoV-2            shares            96.08%            sequence            identity            to            that                                                     approach        for        binding        aff inity        prediction        on        SARS-CoV        BA
of             SARS-CoV.            The             great             gene             conservation             provides             the                                               dataset,             we               use               the               same               dataset               partitioning               scheme
opportunity              for              drug              repurposing,            i.e.            use              of              SARS                     Mpro                     and         cross-validation         strategy         [34].        The         test         set         is         divided
inhibitor         for         potent         of         SARS-CoV-2                                       Mpro          inhibitor.                                                      into                  f ive                  partitions                  for                  5-fold                  cross                  validation,                so                  the
       Recently,     a      dataset      of      185      inhibitors      of      SARS-CoV/SARS-                                                                                       test         labels         are         used         alternately         for         validation.        It         is         worth
CoV-2               Mpro,           which            have            experimental            binding            aff inities,                                                           mentioning          that          although          MP-GNN          and          MathDL          use          the
has                     been                     collected.                    The                     eff icient                     software                     MathPose            same        dataset        and        random        dividing        scheme,       the        partition
has              been              employed              to              predict              their              3D              structures,             and                           can         be         different.        The         average                              Rp      and         Kendall’s         tau         (τ                 )         for
the            binding            complexes            between                                      Mpro          and            these            ligands,                             our              MP-GNN              model              is              0.855              and              0.654,            which              is              better
which       are       denoted       as       SARS-CoV       BA.      We       test       our       MP-GNN                                                                              than         the         results         of         MathDL,        which         are         0.729         and         0.540.

                                                                                                                                                                                                                                        Multiphysical        graph        neural        network        (MP-GNN)                                                 |      7
Discussion                                                                                                                                                                                 length          of          sub-graph          descriptors,         concatenation          operator
Factors          that          impact          the          improvement          extent          of                                                                                        can        completely         deliver         the        features         while         implicitly        con-
multi-scale          stacking                                                                                                                                                              tain         the         message         of         atom         types.        Notice         that         the         features
It        is        revealed        in        supplementary        that        multi-scale        stacking                                                                                 of       element-specif ic       graph       do       not       include       atom       types,      and
improves             the                   Rp        by             6–7%             for             PDBbind             ref ined             datasets,                                    that            is            why            symmetric            operators            are            irrational            for            sub-
but           improves           it           by           15–24%           for           the           assessment           on           SARS-                                            graph         feature         fusion.
CoV               BA               dataset,              which               is               a               huge               gap.              We               assume               theAblation         analysis         for         element-specif ic         graph         and         single
possible               reason               is               the               diversity               and               the               complexity               of                     aggregation
the                 latter                 training                 set.                For                 every                 dataset,                the                 average      Result-related                    f igures                    in                    this                    section                    are                    from                    some
atom                 number                 of                 the                 binding                 site,mean(B)                     and                 of                 the     ablation                      study                      designed                      earlier.                     At                      the                      beginning,                     a
sub-graph,                          mean(G)                    are               recorded               in         Table         1.              As               can               be     single            graph            is            used            to            describe            the            whole            binding            site.
seen        from        Tables                       S1,        S2       and          S3,      there        is        no        obvious        linear                                      It                    is                    a                    huge                    bipartite                    structure                    and                    we                    have                    to                    use
relationship       between       prediction       complexity       and                                                                               mean(B).                              a                   large                   cropping                   size                   such                   as                   130.                  The                   network                   was
Meanwhile,               the                ratio                between                           mean(B)                    and            mean(G)                     is                able                to                converge                but                the                result                is                more                chaotic.              Early
given                in                the                last                column                of                Table                5,               which                is                directlyexperiments             have             shown             that             converting             from             a             single
related           to           the           non-empty           ratio           in           all           sub-graphs.          Under                                                     graph                  to                  element-specif ic                  graphs                  makes                            Rp        increased
the        premise        that        most        binding        sites        include        C,      N,      O,      S        and                                                          from           0.69           to           0.749           without           any           stacking           ensemble.          This
H,           a             high             ratio             means             that             more             element             types             appear                             shows                that                seeing                things                from                a                single                scale                is                much
in        ligand.       In        another        word,       the        more        the        non-empty        sub-                                                                       clearer                 than                 looking                 at                 the                 whole                 graph                 directly.                The
graphs,           the             more             suff icient             is             the             information,           and             the                                       subsequent           ensembles           improve           the           prediction                                                                Rp       again.
more             the             network             learns             from             training.           As             a             result,           we                             Then,        we         realized         that         to         some         extent,        these         sub-graphs
presume                 that                 this                 ratio                 has                 negative                 correlation                 with                      can         be         viewed         as         complete         graph,        that         is,        any         two         nodes
the           task           complexity.          Meanwhile,          on           the           same           test           set,          a                                             in                     a                     sub-graph                     can                     obtain                     each                     other’s                     information
more              diversif ied              training              set              can              help              to              obtain              better                           through            one            aggregation.          This            means            the            superposition
results.              In               conclusion,             we               believe               that               the               f irst               reason                     of                multilayer                aggregation                may                lead                to                redundant                and
for                 such                 great                 progress                 on                 SARS-CoV                 BA                 is                 the                 mes-overlapping                             information.                            So                             we                             deconstructed                             the
sage                discrepancy                between                training                and                validation                set.                                            graph                convolution                layer                in                MP-GNN                and                removed                the
The               training               set               is               more               rich               in               information,              thus               the        aggregation          after          the          f irst          layer.         The          best          single          scale                                        Rp
model           handles           the           validation           set           with           great           facility           and                                                   increased               from               0.749               to               0.767               as               expected,              conf irming
the              stacking              improves              more              than              the              dataset              that              has                               the         effectiveness         of         single         aggregation.
consistent      training      and      test      set.     The      second      reason      is      that
SARS-CoV       BA       has       a       training       set       that       is       several       to       10       times
larger                  than                  the                  PDBbind                  ref ined                  datasets,                 meanwhile                                  Method
includes         not         only         most         data         from         previous         year,        but         also                                                            Multiphysical          GNN
four          divisions          of          SARS-CoV          BA          dataset          that          have          similar                                                            Graph        neural        network        The        GNN        in        our        MP-GNN        consists
complex         structure         to         the         validation         set.                                                                                                           of         two         parts,       i.e.       a         ‘head’         part         and         a         ‘tail’         part.       The         ‘head’
Schemes         for         feature         fusion                                                                                                                                         part                 converts                 the                 node                 vector                 information                 from                 each
Through             experiments,            we             f ind             that             channel-wise             sum-                                                                bipartite      graph      to      a      hidden      feature      vector.     The      ‘tail’      part      is
mation         for        node        feature         fusion        and        concatenate         for        sub-                                                                         a        fully        connected        neural        network        that        learns        the        binding
graph       feature       fusion       improves       the                                                Rp      as       much       as       possible                                     aff inity         from         the         hidden         feature         vector.
within                   capability.                 As                   is                   mentioned,                 symmetric                   opera-                                      The       ‘head’       part       contains       an       convolution       layer       followed
tors               are               more               suitable               for               nodes               with               a               huge               amount          by       an       encoder.      In       the       convolution       layer,      node       features       are
instead       of       concatenation.      Some       works       [38,39]       in       the       f ield                                                                                  convoluted         as         in         the         traditional         GCN         model,
of               3D               feature               learning               prefer               channel-wise               maximum.                                                                                                  Hl+1         =     σ(          ˆD−          1/2       ˆA     ˆD−          1/2 Hl Wl ),                                                                                                                                                                                     (10)
Experiments              show              that              channel-wise              maximum              f ilters
the                nodes                and                reserves                the                extreme                values                after                fea-
ture         embedding.        Feature         visualization         reveals         that         these                                                                                    in           which            Dˆ     and           Aˆ     denote           the           symmetrical           and           normal-
extremes       originate       more       from       the       inf   lection,     depressions                                                                                              ized             degree             and             adjacent             matrices             of             the             graph,                         Hl       the
and           contours           where           the           features           stand           out.          In           contrast,                                                     node          feature          matrix          of          the                       lth       layer,            W        the          layer-specif ic
the       features       that       contribute       more       to       the       complex       binding                                                                                   weight            matrix            and                      σ(     ·)    denotes            the            activation            function.
aff inity         exist         more         in         the         chemical         bond         force         of         binding                                                         Note              that              the              input              for  Hl       is              just              the              node              features              as
site            elements            than            in            the            3D            prof ile            of            the            protein.          The                      in          Eq.        (6).        The          encoder          part          consists          of          a          fully          connected
binding              aff inity              can              be              viewed              as              the              superposition              of                            layer,        dropout,        and         followed         by         the         activation         layer.
all           chemical           bond           force           in           the           binding           site,         so           using           the                                       The               ‘tail’               part               also               contains               two               parts,             i.e.             a               feature
maximum               operator               will               lose               most               of               the               information.                                      fusion       part       and       encoder-based       prediction       part.     In       the       fea-
This             explains             the             applicability             of             sum             operator.            On             the                                     ture          fusion          part,         all          node          feature          vectors          are          aggregated
other        hand,       based        on        the        premise        of        limited        number        and                                                                       into                a                single                feature                vector.               The                commonly                used                fuse

8        |     Li        et        al.
operations              include              concatenation,             sum,             average,             maxi-                                                                   EP       =               [C,             N,             O,             S,             H]               andEL=               [C,             N,             O,             S,             H,             F,             Cl,             Br,             I].
mum             and             minimum             of             all             node             features.            Here,            we             use                          For        generalized        expoential/Lorentz        kernel        and        Euclidean
summation              of              node              features              for              each              element-specif ic                                                   distance       based       atomic       interaction       functions,      we       consider
graph.      Then        we        concatenate        summation        vectors        from        all                                                                                  only           36               =                   4         ∗                 9types              element-specif ic              bipartite              graph
50         element-specif ic         graphs         into         a         long         vector.        In         the         sec-                                                    representation                and                omit                the                inf   luence                from                hydrogen
ond            part,          two            encoders            together            with            a            fully            connected                                          (H)        atoms.       For        electrostatic-based        interaction        function,       a
layer              are              used              to              predict              the              binding              aff inity              from              the         total          50       =         5  ∗         10types       of       bipartite       graphs       are       constructed.
concatenated         vector.                                                                                                                                                          Note                   that                   for                   these                   bipartite                   graphs,                 their                   sizes                   may
       Ensemble                  learning                  The                  ensemble                  learning                  method                                            vary        greatly        between        different        protein–ligand        complexes
is               to               use               multiple               learning               models               simultaneously               for                               and        between        different        element        combinations.       In        our        MP-
boosting             the             performance.            Each             learning             model,            which                                                            GNN,         node          importance          is          def ined          using          rigidity          index          as
is         known         as         a         base         learner,        can         give         a         prediction         individ-                                             in             Eq.            (3).            To             share             the             weights             (in             GNN             model)             among
ually.           Ensemble             learning             improves             the             prediction             accu-                                                          different            graphs,           we            choose            a            same            cropping            size,           i.e.           a
racy            of            each            base            learner            by            assembling            them            together                                         total           of           56           nodes,          for           all           bipartite           graphs.          Computation-
under                 a                 certain                 combination                 strategy.                The                 commonly                                     ally,              it                is                found                that                56                is                roughly                the                average                size                of
used           combination           approaches           including           bagging,          boost-                                                                                these              element-specif ic              bipartite              graphs.             For              large-sized
ing         and         stacking         [40].       In         our         MP-GNN,       we         focus         on         stack-                                                  graphs,                 we                  will                  remove                  the                  nodes                  that                  have                  a                  lower
ing        ensemble        model.       The        essential        idea        is        to        assign        a        cer-                                                       node              importance.             For              small-sized              graph,             pseudo-nodes
tain          weight,         which          is          to          be          learned,         to          each          base          learner                                     are         added         until         a         common         size         of         56         is         reached.
and       use       the       weighted       results       as       the       f inal       prediction.     More                                                                              In          our          MP-GNN          model,         node          features          are          only          related
specif ically,            we              can              denote              the              prediction              of                 n     number                               to            atomic            distances.          For            generalized            kernel            based            func-
of              base              learners              as   Y1 ,     Y2 ,     ...,     Yn,             the              f inal              prediction              as               tion          /Phi1as             in             Eqs.            (1)             and             (2),            we             setκto             be             2             and             four
Ystacking             and            the            ground            truth            value            of            training            set            as        Y.                 different         scale         parameters         are         considered,       that         is,                                                     η   =2,       5,
The            weight            for            each            based            learner            is            linearly            related            to                           10          and          20          Å.        For          Euclidean          distance          based          function                                    /Phi1as
their       prediction       accuracy       on       training       set.     For       instance,     we                                                                               in           Eq.          (8),          we           setκ       =−               1and           let/Phi1simply           equals           to           the
can        use          Rp (Yi ,     Y),      which        is        Pearson        correlation        coeff icient                                                                   atomic         distance.        The         function         domain         of         Eqs.        (1),        Eqs.        (2)
Rp       between               the               prediction                        Yi       and               true               value       Y,              as               the     and            Eqs.           (8)            is            set            to            be            [2            Å,           30Å]            with            each            interval            of
measurements              for              the              model              accuracy.             The              weight              for                                         length          1          Å,        and          the          node          vector          as          in          Eq.        (6)          is          of          size          29.
the         the        i-th         base         learner         is         then         def ined         as                                                                          For       electrostatic-based       function                                                    /Phi1as       in       Eq.      (9),      we       set       the
                                                                                                                                                                                      domain        to        [0,      1]        with        each        interval        of        length        0.04,      and        the
                                               Wstacking                       ∑n                                                                                                     node         vector         is         of         size         25.
                                                      i                                                                        =                                                      Rp (Yi ,     Y)j=1       Rp (Yj ,     Y)    ,MP-GNN       parameter       settings       The       encoders       in       MP-GNN
                                                                                                                                                                                      head            have            the            output            size            of            64            and            16            for            every            node.
and         the         f inal         prediction         results         are                                                                                                         After                 node                 feature                 fusion,                preliminary                 sub-graph                 fea-
                                                                                                                                                                                      tures          go          through          an          encoder          with          output          feature          length
                                                                               ∑n                                                                                                     16.         Then          the          sub-graph          feature          matrix          with          shape          (M*N,
                                                  Ystacking        =                     Wstackingi                                                                                Yi .16)               is               concatenated               into               one               feature               vector               describing
                                                                                i=1                                                                                                   the           binding           site,          which           passes           through           the           hidden           layer
                                                                                                                                                                                      with           256           and           64           neurons           for           f inal           regression.          Every           MP-
Other          than          using                   Rp      as          accuracy          measurement,        we          have                                                       GNN        sub-learner        is        trained        for        6400        epochs        to        obtain        the
also         considered         RMSE         in         our         MP-GNN         models.                                                                                            optimal              model              with              a              dropout              rate              of              0.5              and              ELUs              as
                                                                                                                                                                                      the         activation         unit.       The         learning         rate         starts         from         0.1         and
MP-GNN          for          Covid          drug          design                                                                                                                      decays            every            800            epochs,          and            the            decay            rate            is            0.5.          The
Graph                   representation                   for                   protein–ligand                   interactions                                                          decay         scheme         is         depicted         in                          Figure         3.
Ligands             usually             bind             to             proteins,            which             tend             to             have             a                            Performance                 of                 MP-GNN                 on                 PBDbind                 datasets                    Fo r
much          larger          size,         at          a          certain          special          region          called          bind-                                            PBDbind               datasets,              a               total               of               10               different               scale-specif ic
ing             site.            Computationally,            the             binding             site             is             chosen             as                                GNN         models         are         considered         based         on         10         atomic         interac-
the        protein        region        that        is        within        a        certain        cutoff        distance                                                            tion           functions,          including           four           different           exponential           ker-
of            the            ligand            atoms.           Here,           we            use            10            Å            in            our            MP-GNN           nel         functions,       four         different         Lorentz         kernels,       a         Euclidean
model.       The        protein–ligand        interaction        matrix                                                                 M       in        Eq.       (7)               distance       based       function       and       an       electrostatic-based       func-
is        def ined        only        on        the        protein        binding        sites        instead        of        the                                                    tion.     A      total      of      10      GNN      base      learners      can      be      obtained.     The
entire         protein         domain.       Three         types         of         atomic         interaction                                                                        stacking           models           are           chosen           based           on                              Rp       on           the           training
function                  /Phi1are      considered,     including      generalized      expoen-                                                                                       set            performance.          Due           to            the           high            computation            cost,          we
tial/Lorentz        kernel        function,      Euclidean        distance        function                                                                                            conducted        22        repeated        experiments        with        random        initial-
and                electrostatic                function.               Under                different                interaction                                                     ization.       The         detailed         results         can         be         found         in         Tables         S1         to
functions,     different      types      of      element-specif ic      graph      mod-                                                                                               S3.          The            best            results            in            every            sector            are            marked            in            bold.
els       are       constructed.      As       stated       above,      proteins       and       ligands                                                                              Note            that            stacking            with                  Rp       is            better            than            the            ones            with
in                general                have                f ive                and                nine                types                of                atoms,              that                is,RMSE.

                                                                                                                                                                                                                   Multiphysical        graph        neural        network        (MP-GNN)                                      |     9
      We       carry       out       the       ablation       analysis       for       our       model       based                                                        Code           and           Data           Availability
on            the            Tables            S1            to            S3.           First,           we            focus            on            the            effective-The                       code                       is                       available                       at                       https://github.com/Alibaba-
ness                 of                 one-scale                 stacking.                When                 looking                 at                 one                 rowDAMO- DrugAI/MGNN.                 Additional                   data                   or                   code                   would
of          each          scale          separately,        one-scale          stacking          signif icantly                                                           be         available         upon         reasonable         request.
improves               the               result               on               PBDbind-v2007,              PBDbind-v2013
and       PBDbind-v2016       by       around       6%,     8%       and       8%.     If       we       apply
the          multi-scale          stacking          without          the          one-scale          stacking,                                                            Author           contributions           statement
both              average              and              best       Rp      are              improved              by              5–10%              than                 K.X.         conceived          MP-GNN          model.         K.X.,         X-S.L.         and          Y.C.         con-
the               single               scale               results.              In               contrast,              multi-scale               stack-                 ceived          the          graph          neural          network          and          ensemble          learning
ing            after            one-scale            stacking            can            only            boost                Rp       by            0–3%.                 architecture.       X.L.       prepared         the         input         data.       X-S.L.       and        Y.C.
This          indicates          that          the          one-scale          stacking          can          avoid          the                                          wrote          up          all          algorithm          codes          and          accomplished          try-run.
information            blind            spot            caused            by            initialization.           Second,                                                 X-S.L.       conducted        the        experiments        in        large        scale        and        ana-
we       study       the       effectiveness       of       multi-scale       stacking.      Before                                                                       lyzed        the        results.       X-S.L.       ref ined        the        network        architecture.
one-scale       stacking,      the       multi-scale       stacking       average                                                                Rp      of               K.X.            validated             the             results             according             to             experience.            K.X.
22      random      trials      is      at      least      6%,     7%      and      6%      higher      than      the                                                     and        X-S.L.       wrote        up        the        paper,       all        other        authors        reviewed
single       scale       average                        Rps       on       all       three       datasets.      It       is       obvious                                 the         manuscript.
that         there         is         information         complement         between         different
scales.          Based           on           the           result           of           one-scale           stacking,          multi-                                   Supplementary           data
scale          still          increases          the               Rp      by          almost          3%.        In          comparison,
we                have                noticed                that                although                the                one-scale                stacking             Supplementary                       data                        are                       available                       online                       at                       https://
avoids                 randomness                 and                 collects                 the                 information                 in                 a       academic.oup.com/bib.
single           scale           as           much           as           possible,          it           can           not           overrun           the
Best         Rp      of         the         randomly         initialized         multi-scale         stacking.                                                            Acknowledgments
      Performance                 of                 MP-GNN                 on                 SARS-CoV                 BA                 dataset                        This       work       was       supported       by       Alibaba       Innovative       Research
In                our                MP-GNN               model                for                SARS-CoV                BA                dataset,              only    (AIR)       Program       and       Alibaba-NTU       Singapore       Joint       Research
multiscale       stacking       is       employed.     This       is       due       to       the       reason                                                            Institute                grant                AN-GC-2020-002,               Singapore                Ministry                of
that          the          training          set          has          incorporated          in          it          the          PBDbind-                                Education        Academic        Research        fund        Tier        1        RG109/19,       and
v2019          general          set,         which          has          17652          PBD          data.         Similar          to                                    Tier         2         MOE-T2EP20220-0010,        MOE-T2EP20120-0013.
PBDbind               datasets,              the               same               10               different               scale-specif ic
GNN        models        are        considered        in        our        MP-GNN.       Since        stack-
ing           with       Rp      gives           better           accuracy,          we           also           use               Rp      results                        References
on           training           set           as           the           weighting           scheme.         From           Table                     S4,
it          can          be          seen         that          the          multiscale         stacking          improves          the                                      1.     Zhang            L,           Tan            J,           Han            D,           et            al.           From            machine            learning            to            deep
Rp     by         15–24%         for         SARS-CoV         BA         dataset.                                                                                                   learning:                progress                in                machine               intelligence                for                rational                drug
                                                                                                                                                                                    discovery.      Drug       Discov       Today       2017;22(11):1680–5.
                                                                                                                                                                             2.     Lusci     A,    Pollastri     G,    Baldi     P.    Deep     architectures     and     deep     learning
                                                                                                                                                                                    in            chemoinformatics:            the            prediction            of            aqueous            solubility            for
     Key        Points                                                                                                                                                              drug-like       molecules.      J       Chem       Inf       Model       2013;53(7):1563–75.
     Our        main        contributions        in        this        paper        are        as        follows:                                                            3.     Pereira             JC,            Caffarena             ER,            Nogueira             C,            et             al.            Boosting             docking-
         •     We       propose       the       f irst       multiphysical       molecular       graph       rep-                                                                   based              virtual              screening              with              deep              learning.            J              Chem             Inf             Model
               resentation.     All      kinds      of      molecular      interactions,     between                                                                                2016;56(12):2495–506.
               different               atom               types               and               at               different               scales,              are               sys-4.Kearnes          S,         McCloskey          K,         Berndl          M,         et          al.         Molecular          graph          con-
               tematically        represented        by        a        series        of        scale-specif ic        and                                                          volutions:           moving           beyond           f ingerprints.         J           Comput           Aided           Mol           Des
               element-specif ic        graphs        with        distance-related        node        fea-                                                                          2016;30(8):595–608.
               tures.                                                                                                                                                        5.     Gomes               J,              Ramsundar               B,              Feinberg               EN,              et               al.              Atomic               convolu-
         •     We      develop      the      f irst      multiphysical      graph      neural      network                                                                          tional              networks              for              predicting              protein-ligand              binding              aff init-
               (MP-GNN)         model.        Our         MP-GNN         is         free         from         the         compli-                                                   yarXiv       preprint       arXiv:1703.10603.      2017.
               cated          feature          generation          process.         A          unique          GNN          archi-                                           6.     Feinberg             EN,            Sur             D,            Wu             ZQ  ,            et             al.            Potentialnet             for             molecular
               tecture        is        developed        in        our        MP-GNN        to        incorporate        both                                                       property       prediction.      ACS       central       science       2018;4(11):1520–30.
               scale-specif ic               and               element-specif ic               graph               information                                               7.     Tsubaki        M,       Tomii        K,       Sese        J.       Compound–protein        interaction        pre-
               into        a        consolidated        model.                                                                                                                      diction        with        end-to-end        learning        of        neural        networks        for        graphs
         •     Our          model          has          achieved          the          state-of-the-art          results          for                                               and       sequences.      Bioinformatics       2019;35(2):309–18.
               protein–ligand              binding              aff inity              prediction.             It              has              been                         8.     Li          X,         Yan          X,         Qiong          G,         et          al.         Deepchemstable:          chemical          stability
               found         that         our         model         can         outperform         all         existing         mod-                                                prediction        with        an        attention-based        graph        convolution        network.
               els,       as        far        as        we        know.                                                                                                            J       Chem       Inf       Model       2019;59(3):1044–9.
         •     Our                   model                   is                   highly                   accurate                   for                   the                   prediction                   of9.Wang        X,       Li        Z,       Jiang        M,       et        al.       Molecule        property        prediction        based
               complexes          of          inhibitors          for          SARS-CoV/SARS-CoV-2.         Our                                                                     on      spatial      graph      embedding.     J      Chem      Inf      Model      2019;59(9):3817–28.
               model        has        great        potential        for        COVID-19        drug        design.                                                       10.       Szocinski      T,     Nguyen      DD,     Wei      G-W.     AweGNN:      Auto-parametrized
                                                                                                                                                                                    weighted     element-specif ic     graph     neural     networks     for     molecules.
                                                                                                                                                                                    Comput       Biol       Med       2021;134:104460.

10          |      Li        et        al.
11.        Li     S,    Zhou     J,    Tong     X,    et     al.    (eds).    Structure-aware      interactive      graph                                                                       25.         Cang         ZX,        Mu         L,        Wei         GW.        Representability         of         algebraic         topology
           neural              networks              for              the              prediction              of              protein-ligand              binding                                          for       biomolecules       in       machine       learning       based       scoring       and       virtual
           aff inity.             In:              Proceedings              of              the              27th              ACM              SIGKDD              Conference              on              screening.      PLoS       Comput       Biol       2018;14(1):e1005929.
           Knowledge       Discovery       &       Data       Mining,      2021,      975–85.                                                                                                   26.         Nguyen       DD,      Cang       ZX,      Wu       KD,      et       al.      Mathematical       deep       learning
12.        Nguyen            T,           Le            H,           Quinn            TP,           et            al.           GraphDTA:            Predicting            drug–                            for              pose              and              binding             aff inity              prediction              and              ranking             in              D3R
           target     binding     aff inity     with     graph     neural     networks.    Bioinformatics                                                                                                   Grand       Challenges.      J       Comput       Aided       Mol       Des       2019;33(1):71–82.
           2021;37(8):1140–7.                                                                                                                                                                   27.         Nguyen       DD,      Wei       GW.      AGL-Score:       Algebraic       graph       learning       score
13.        Lin               X.              DeepGS:               Deep               representation               learning               of               graphs               and                         for      protein-ligand      binding      scoring,     ranking,     docking,     and      screen-
           sequences                      for                      drug-target                      binding                      aff inity                      predictionarXiv                             ing.      J       Chem       Inf       Model       2019;59(7):3291–304.
           preprint       arXiv:2003.13902.      2020.                                                                                                                                          28.         Nguyen      DD,     Gao      KF,     Wang      ML,    et      al.     MathDL:      Mathematical      deep
14.        Jiang                   M,                  Li                   Z,                  Zhang                   S,                  et                   al.                  Drug-target                   aff inity                   predic-learning             for             D3R             Grand             Challenge             4.            Journal             of             computer-aided
           tion            using            graph            neural            network            and            contact            maps.           RSC            Adv                                      molecular       design,      pages       2019;1–17      .
           2020;10(35):20701–12.                                                                                                                                                                29.         Nguyen       DD,      Cang       ZX,      Wei       GW.      A       review       of       mathematical       repre-
15.        Wang     X,    Liu     Y,    Fan     L,    et     al.    Dipeptide     frequency     of     word     frequency                                                                                   sentations       of       biomolecular       data.      Phys       Chem       Chem       Phys       2020.
           and               graph               convolutional               networks               for               DTA               prediction.               Front                         30.         Puzyn       T,      Leszczynski       J,      Cronin       MT.      Recent       advances       in       QSAR       stud-
           Bioeng       Biotechnol       2020;8:267.                                                                                                                                                        ies:         methods         and         applications,        Vol.       8.        Springer         Science         &         Business
16.        Stokes          JM,         Yang          K,         Swanson          K,         et          al.         Zohar          Bloom-Ackermann,                                                         Media,      2010.
           et               al.              A               deep               learning               approach               to               antibiotic               discovery.              Cell31.     Lo       YC,      Rensi       SE,      Torng       W,      et       al.      Machine       learning       in       chemoinfor-
           2020;180(4):688–702.                                                                                                                                                                             matics      and      drug      discovery.     Drug      Discov      Today      2018;23(8):1538–46.
17.        Liu       X,      Luo       Y,      Li       P,      et       al.      Deep       geometric       representations       for       mod-                                               32.         Bajorath          J.         Chemoinformatics:          concepts,         methods,         and          tools          for          drug
           eling            effects            of            mutations            on            protein-protein            binding            aff inity.                                                    discovery,      Vol.      275.      Springer       Science       &       Business       Media,      2004.
           PLoS       Comput       Biol       2021;17(8):e1009284.                                                                                                                              33.         Nguyen        DD,      Gao        K,      Chen       J,      et       al.      Unveiling        the        molecular        mech-
18.        Gaudelet            T,           Day            B,           Jamasb            AR,           et            al.           Utilising            graph            machine                           anism       of       SARS-CoV-2       main       protease       inhibition       from       137       crystal
           learning       within       drug       discovery       and       development.      Brief       Bioinform                                                                                         structures        using       algebraic        topology        and        deep        learning.      Chem       Sci
           05       2021;bbab159.                                                                                                                                                                           2020;11(44):12036–46.
19.        Wei           GW.          Persistent           homology           analysis           of           biomolecular           data.         J                                            34.         Nguyen         DD,        Wei         GW.        DG-GL:         Differential         geometry-based         geo-
           Comput       Phys       2017;305:276–99.                                                                                                                                                         metric             learning             of             molecular             datasets.           International             journal             for
20.        Wei              GW.             Mathematics              at              the              eve              of              a              historic              transition              in      numerical       methods       in       biomedical       engineering       2019;35(3):e3179.
           biology.      Computational       and       Mathematical       Biophysics       2017;5(1).                                                                                           35.         Liu                 ZH,                Li                 Y,                Han                 L,                et                 al.                PDB-wide                 collection                 of                 binding
21.        Cang         ZX,        Wei         GW.        TopologyNet:         Topology         based         deep         convolu-                                                                         data:                current                status                of                the                PDBbind                database.              Bioinformatics
           tional           and           multi-task           neural           networks           for           biomolecular           prop-                                                               2015;31(3):405–12.
           erty       predictions.      PLoS       Comput       Biol       2017;13(7):e1005690.                                                                                                 36.         Wee           JJ,          Xia           K.          Forman           persistent           ricci           curvature           (FPRC)           based
22.        Cang             ZX,            Wei             GW.            Integration             of             element             specif ic             persistent                                       machine               learning               models               for               protein-ligand               binding               aff inity
           homology                 and                 machine                 learning                 for                 protein-ligand                 binding                                         prediction.      Brief ings       in       Bioinformatics,      in       press       2021.
           aff inity          prediction.         In:          International          journal          for          numerical          methods                                                  37.         Qi         CR,        Hao         S,        Mo         K,        et         al.        Pointnet:         Deep         learning         on         point         sets
           in       biomedical       engineering,page       10.1002/cnm.2914,      2017.                                                                                                                    for       3d       classif ication       and       segmentation.      In:       Proceedings       of       the       IEEE
23.        Nguyen               DD,              Xiao               T,              Wang               ML,              et               al.              Rigidity               strengthening:             conference       on       computer       vision       and       pattern       recognition,      2017,      652–60.
           A                  mechanism                  for                  protein–ligand                  binding.                 J                  Chem                  Inf                  Model38.Qi               CR,              Yi               L,              Su               H,              et               al.               Pointnet++:               Deep               hierarchical               fea-
           2017;57(7):1715–21.                                                                                                                                                                              ture               learning               on               point               sets               in               a               metric               spacearXiv               preprint
24.        Cang             ZX,            Wei             GW.            Integration             of             element             specif ic             persistent                                       arXiv:1706.02413.      2017.
           homology                 and                 machine                 learning                 for                 protein-ligand                 binding                             39.         Sagi                           O,                         Rokach                           L.                         Ensemble                           learning:                           A                          survey.                             Wi l ey
           aff inity            prediction.           International            journal            for            numerical            methods            in                                                 Interdisciplinary                 Reviews:                 Data                 Mining                 and                 Knowledge                 Discovery
           biomedical       engineering       2018;34(2):e2914.                                                                                                                                             2018;8(4):e1249.

