                                                                                                                                                                                                   Bioinformatics,         38(9),         2022,         2571–2578
                                                                                                                                                                                  https://doi.org/10.1093/bioinformatics/btac155
                                                                                                                                                                        Advance        Access        Publication        Date:         11         March         2022
                                                                                                                                                                                                                                                      Original        Paper
Systems                     biology
BridgeDPI:                a               novel               Graph               Neural               Network               for               predicting
drug–protein               interactions
Yifan          Wu1,2,†,          Min          Gao1,†,          Min          Zeng2,          Jie          Zhang                                                                                    1,3,4,*          and          Min          Li                             2,*
1SenseTime               Research,               Shanghai               200233,               China,               2School               of               Computer               Science               and               Engineering,               Central               South              University,               Changsha
410083,           China,           3Qing           Yuan           Research           Institute,           Shanghai           Jiao           Tong           University,           Shanghai           200240,           China           and           4Merck           Advisory           Committee
for          AI-enabled          Health          Solution,          Shanghai          200126,          China
*To         whom         correspondence         should         be         addressed.
†The         authors         wish         it         to         be         known         that,         in         their         opinion,         the         ﬁrst         two         authors         should         be         regarded         as         Joint         First         Authors.
Associate         Editor:         Jinbo         Xu
Received        on       August        19,       2021;        revised        on        January        20,       2022;        editorial        decision        on        March        6,        2022;        accepted        on       March        10,        2022
Abstract
Motivation:         Exploring         drug–protein         interactions         (DPIs)         provides         a         rapid         and         precise         approach         to         assist         in         laboratory
experiments         for         discovering         new         drugs.         Network-based         methods         usually         utilize         a         drug–protein         association         network
and                 predict                 DPIs                 by                 the                 information                 of                 its                 associated                 proteins                 or                 drugs,                 called                 ‘guilt-by-association’                 principle.
However,          the          ‘guilt-by-association’          principle          is          not          always          true          because          sometimes          similar          proteins          cannot          interact
with        similar        drugs.        Recently,        learning-based        methods        learn        molecule        properties        underlying        DPIs        by        utilizing        existing
databases        of        characterized        interactions        but        neglect        the        network-level        information.
Results:           We           propose           a           novel           method,           namely           BridgeDPI.           We           devise           a           class           of           virtual           nodes           to           bridge           the           gap           be-
tween             drugs             and             proteins             and             construct             a             learnable             drug–protein             association             network.             The             network             is             optimized
based            on            the            supervised            signals            from            the            downstream            task—the            DPI            prediction.            Through            information            passing            on
this        drug–protein        association        network,        a        Graph        Neural        Network        can        capture        the        network-level        information        among        di-
verse           drugs           and           proteins.            By           combining           the            network-level           information           and           the           learning-based           method,           BridgeDPI
achieves          signiﬁcant          improvement          in          three          real-world          DPI          datasets.          Moreover,          the          case          study          further          veriﬁes          the          ef-
fectiveness        and        reliability        of        BridgeDPI.
Availability              and              implementation:                The                source                code                of                BridgeDPI               can               be                accessed               at                https://github.com/SenseTime-
Knowledge-Mining/BridgeDPI.                          The                          source                          data                          used                          in                          this                          study                          is                          available                          on                          the                          https://github.com/IBM/
InterpretableDTIP         (for         the         BindingDB         dataset),         https://github.com/masashitsubaki/CPI_prediction         (for         the         C.ELEGANS         and
HUMAN)         datasets,        http://dude.docking.org/        (for         the        DUD-E         dataset),        repectively.
Contact:        limin@mail.csu.edu.cn        or        zhangjie1@sensetime.com
1        Introduction                                                                                                                                                                The       interaction      mechanism       between      drugs      and      proteins      is      very
                                                                                                                                                                             complicated,         so         it         is         a        big         challenge         to         develop         an         efficient         com-
Developing          a          new          drug          conventionally          takes          tens          of          years          and          bil-                  putational           method           that           can           accurately           determine           DPIs.           Current
lions          of          dollars          (Avorn,          2015;          Paul          et          al.,          2010).          In          the          process          ofcomputational              methods             for              predicting              DPIs             are              classified              into
new        drug        development,        drug–protein        interaction        (DPI)        prediction        is                                                          three      categories:      docking-based      methods,       network-based      methods
a       crucial       step.       Traditional       experimental       assays       for       determining       DTIs                                                         and        learning-based        methods.        Docking-based        methods        usually        use
are          to          measure          the          value          of          half-maximal          inhibitory          concentration                                    molecular      dynamic      simulation      to      reconstruct      the      contact      relation-
(IC50)        or        inhibitory        constant        (Ki)        in        wet        experiments.        Although        the                                           ship        between        proteins        and        drugs        in         space.        These         methods        aim        to
experimental         assays         remain         the         most         reliable         approach         for         examin-                                            look       for       the       best       binding       position       inside       the       binding       pocket       of       the
ing             DPIs,             they             are             time-consuming             and             cost-intensive             due             to             the  proteins              for              drug              molecules              ( Gschwend              et             al.,1996;              Led              and
need          of          individual          experiments          for          all          of          the          possible          drug–protein                         Caflisch,      2018).      Nevertheless,      they      require      accurate      3D      structures
pairs.                  Therefore,                  exploring                  the                  interaction                  mechanism                  between          of      proteins,      which      are      hard      to      obtain      while       some      proteins      even      do
drugs              and              proteins              and              developing              some              efficient              computational                    not       have       the       3D       structure       ( Liu      and      Altman,      2015 ;       Mizianty       et       al.,
methods                  for                  predicting                  DPIs                  are                  very                  significant                  and                  urgently2014 ;       Zhang       et       al.,2012).       In       contrast      to      the      docking-based      meth-
demanded.                                                                                                                                                                    ods,                network-based                methods                bypass                the                direct                reconstructing
VC             The        Author(s)        2022.        Published        by        Oxford        University        Press.        All        rights        reserved.        For        permissions,        please        e-mail:        journals.permissions@oup.com                                                                                                                                                                                                                                                        2571

2572                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Y.Wu       et       al.
their        contact        relationship        by        using        the        ‘guilt-by-association’        prin-                                                                                              interactions              between              proteins              and              drugs              (i.e.              P2–D2).              In              this
ciple       to       predict       DPIs       based       on       some       priori       DPI       data       ( Ballester      and                                                                               way,                 the                 interaction                 between                 P1                 and                 D1                 can                 be                 implicitly
Mitchell,       2010;        Bleakley       and       Yamanishi,       2009  ;       Ding       et       al.,2014;                                                                                                 inferred         by         P1–P2,          P2–D2,         and         D2–D1.          Paths          like          this          type          are
Durrant            and             McCammon,             2011 ;             Luo            et            al.,2017,             2019 ).             The                                                             the       implicit       paths.       In       a       word,       due       to       the       bridge       nodes,       BridgeDPI
‘guilt-by-association’       principle       assumes       that       if       a       target       has       a       simi-                                                                                        not       only      learns       a      deep       interaction      mechanism      but      also      assists       DPI
lar       profile       with       another       target,       the       former       target       is       likely       to       inter-                                                                           prediction                        from                        a                        network-level                        perspective.                        This                        makes
act        with        a        drug,        which        can        directly        interact        with        the        latter        target                                                                   BridgeDPI         more         comprehensively         grasp         features         to         perform         DPI
(Luo        et        al.,        2021;        Wang        and        Lukasz,        2019 ).        Thus,        these        methods                                                                              prediction      and      the      prediction      results      are      also      more      reliable.
usually            need            to            construct             a            network            that            contains            the            existing                                                           We       provide       comprehensive       comparison       results       with       other       base-
drugs           and           proteins           and           calculate           similarity           scores           of           the           drug–                                                          lines            in            four            different            datasets.            Compared            to            the            state-of-the-art
drug             and              protein–protein              pairs.             However,              the              ‘guilt-by-associ-                                                                         (SOTA)             results,             BridgeDPI             achieves             the             area            under             the             receiver
ation’         principle         relies         on         the         quality         of         the         similarity         scores,         and                                                               operating           characteristic           curve           (AUROC)           scores           of           97.5%           (1.9%
cannot             be             easily             applied             to             low-frequency             or             unseen             proteins.                                                      higher)              in              the              BindingDB             dataset,             99.5%             (0.7%             higher)              in              the
Moreover,         the         ‘guilt-by-association’         principle         is         not         always         true,                                                                                         C.ELEGANS           dataset            and            99.0%           (1.1%           higher)           in           the           HUMAN
because        there        are        also        some        similar        proteins        that        cannot        interact                                                                                   dataset.                In                addition,                we                use                the                Directory                of                Useful                Decoys,
with            similar            drugs            ( Maggiora             et            al.,2014).            Recently,            with            the                                                            Enhanced          (DUD-E)          dataset          as          an          independent          test          set          to          evaluate
plentiful       accumulation      of      data,      learning-based      methods      have      been                                                                                                               the             generalization.             BridgeDPI             trained             on             the             BindingDB             dataset
successfully         applied          to          predict          DPIs          (  Li          et         al.,2020,          2022 ;          Wan                                                                  gets         the        AUROC        scores         of        77.2%        (3.2%        higher).         In        summary,         all
et         al.,2019;          Wang          et          al.,          2021;          Wang         and         Zeng,         2013;          Yuvaraj                                                                 results         indicate         that         BridgeDPI         is         effective         and         reliable         in         predict-
et                   al.,2021).                   Learning-based                   methods                   usually                   take                   protein                                              ing      DPIs.
sequences            and            drug            molecules            as            the            input            for            DPI            prediction.
Compared          to          the          network-based          methods,           learning-based          meth-
ods          mainly          focus          on         learning          the          interaction         mechanism         of          each                                                                       2        Materials        and        methods
single                 drug–protein                 pair                 by                 the                 prior                 data                 but                 ignore                 some         2.1        Datasets
network-level                information,                i.e.                 the                ‘guilt-by-association’                prin-
ciple,       which       is       quite       crucial       to       infer       DPIs.       Therefore,       it       is       very       im-                                                                     BindingDB:            The            BindingDB            dataset            has            collected            affinity            data            of
portant       and       necessary       to       design       a       model       that       brings        the       ‘guilt-by-                                                                                    2          286         319          drug–protein          pairs          from         corresponding          research         papers,
association’      principle      into      the      learning-based      methods.                                                                                                                                   where        8536        proteins        and        989        383        drugs        are        included        (Gilson        et        al.,
          In         this         paper,         we         develop         BridgeDPI,         a         deep         learning         frame-                                                                      2016;                the                raw                BindingDB                can                be                accessed                at                https://www.bind
work,                        which                        simultaneously                        combines                        the                        advantages                        of                    ingdb.org/bind/index.jsp,           and           the           Gao’s           version           of           BindingDB           can
network-based          methods          and         learning-based         methods,          to          predict                                                                                                   be                  downloaded                   from                  https://github.com/IBM/InterpretableDTIP).
DTIs.         Compared         to         previous         learning-based         methods,         we         intro-                                                                                               On          this          basis,          Gao          et          al.          (2018)          select          the          data          having          IC50          value
duce              protein             sequences             and              drug             molecules             into              a             supervised                                                     and                                  convert                                  the                                  IC50                                  values                                  into                                  1                                  for                                  interactions
drug–protein         association         network.         The         network         provides         neigh-                                                                                                      (IC50    <    100        nM),        0        for        no        interactions        (IC50    >    10        000        nM)        to        con-
borhood             information             of             proteins             and            drugs,             which             makes             the                                                          struct           a           binary           classification           dataset.           The           dataset           contains           39           747
model        not        only        learn        the        interaction        mechanism        of        drug–protein                                                                                             positive           samples           and           31           218           negative           samples           and           is           divided           into
pairs               but               also               provides               a               network-level               perspective               to               assist                                      training         (28          240          positive          and          21          915          negative          samples),          validation
learning.         In         the         process         of         constructing         the         supervised         network,                                                                                   (2831          positive          and          2776          negative          samples)          and          test          (2706          positive
we           introduce           a           class           of           nodes           called           bridge           nodes.           The           bridge                                                  and          2802          samples)          sets.          We          use          this          dataset          for          our          main          head-to-
nodes        are        devised        to        connect        all        the        proteins        or        drugs        and        meas-                                                                      head      comparisons.
ure         the         associations         among         proteins/drugs         from         a         network-level                                                                                                       C.ELEGANS                               and                               HUMAN                               datasets:                              C.ELEGANS                              and
perspective.         From        the        network-based        point        of        view,         through        the                                                                                           HUMAN       datasets       have       been       widely       used       in       DPI       prediction       (the       bal-
bridge        nodes,         we         can        get        two        types         of        paths        from        a         protein        to        a                                                     anced       versions       of       C.ELEGANS       and       HUMAN       datasets       can       be       down-
drug:           explicit           paths           and           implicit           paths.           As           shown           in               Figure           1,                                             loaded                               from                               https://github.com/masashitsubaki/CPI_prediction).
considering        the        protein–drug        pair        P1–D1,        we        can        go        from        P1        to                                                                                Both         of         them         are         constructed         by         combining         a         set         of         highly         credible
bridge          nodes          to          D1,          where          the          bridge          nodes          explicitly          measure                                                                     negative              drug–protein              samples              via              an              in              silico              screening              method
the               interaction              between              P1              and              D1              and              thus              the              interaction                                   with               the               known              positive               samples               (Liu               et               al.,               2015).               We               follow
mechanism        is        learned        to        decide        whether        the        pair        interacts.        Paths                                                                                    Tsubaki         et         al.         (2019)         and         use         the         balanced         versions         to         do         research.
likeP1tobridgenodestoD1aredefinedastheexplicitpaths.                                                                                                                                                               The             C.ELEGANS             dataset             has             7786             drug–protein             pairs,             including
From          another          point          of          view,          we          can          also          go          from          P1          to          bridge                                           1876              proteins              and              1767              drugs.              The              HUMAN              dataset              has              6728
nodes        to        P2        to        bridge        nodes        to        D2        to        bridge        nodes        to        D1,        where                                                          drug–protein           pairs,           including          2001          proteins           and           2726          drugs.          Both
the        bridge        nodes       not        only       measure        the        associations        between        pro-                                                                                       the            C.ELEGANS            dataset            and             the            HUMAN            dataset            are            randomly
teins                  (i.e.                  P1–P2)/drugs                  (i.e.                  D2–D1)                  but                  also                  measure                  the                 divided      for       5-fold      cross-validation.
                                                                                                                                                                                                                             DUD-E           dataset:           The           DUD-E           is           a           widely           used           dataset           covering
                                                                                                                                                                                                                   102          proteins           and           22           886          clustered           ligands          (Mysinger          et           al.,           2012;
                                                                                                                                                                                                                   the              DUD-E              dataset               can              be               accessed              at               http://dude.docking.org/).
                                                                                                                                                                                                                   There        are        50        decoys        for        each       active        with        similar       physical       and        chem-
                                                                                                                                                                                                                   ical            properties            but            dissimilar           2D            topology.            It            contains            1            429            790
                                                                                                                                                                                                                   protein-ligand        samples        in        total        (22        645        positive        samples,        1        407        145
                                                                                                                                                                                                                   negative             samples).             The             samples             are             demonstrated             by             wet             experi-
                                                                                                                                                                                                                   ments       or        computational        methods.       DUD-E       is        used        as        an        independent
                                                                                                                                                                                                                   test       set       to       evaluate       how       our       model       performs       in       reality.       In       this       paper,
                                                                                                                                                                                                                   we             train             BridgeDPI             on             the             BindingDB             dataset             and             test             it             on             the
                                                                                                                                                                                                                   DUD-E      dataset.
                                                                                                                                                                                                                   2.2        Framework        of        BridgeDPI
Fig.        1.       A        toy       example       of       our       constructed       graph.       The       graph       includes       three       types        of                                           We         propose         a         novel         end-to-end         deep         learning         framework,         namely
nodes:       protein       nodes,       drug       nodes,       bridge       nodes;       three       types       of       edges:       bridge       node-                                                         BridgeDPI,          for         DPI         prediction         task.         The         overall         learning         architec-
protein         node,         bridge         node-drug         node,         bridge         node-bridge         node;         and         two         types         of                                             ture          is          illustrated           in           Figure          2.          BridgeDPI          takes          protein          sequences
paths:      the      explicit      path      (e.g.      P1->bridge      nodes->D1)      and      the      implicit      path      (e.g.      P1-                                                                   and          drug          SMILES          (Weininger,          1988)          as          inputs          and         predicts          their
>bridge     nodes->P2->bridge     nodes->D2->bridge     nodes->D1)                                                                                                                                                 interactions.        It       consists        of        four        parts:        drug        feature        extraction        part,

BridgeDPI                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              2573
Fig.       2.       The       framework       of       BridgeDPI:       on       the       left       is       the       protein       feature       extraction       part,       on       the       right       is       the       drug       feature       extraction       part,       the       middle       is       the       drug–protein       bridge
graph      construction      part,      and      the      bottom      is      the      classiﬁcation      part.      BridgeDPI      uses      CNN      and      FFN      to      extract      proteins’/drugs’      local      and      global      features.      After      that,      some      bridge
nodes      are      introduced      to      construct      the      bridges      between      proteins      and      drugs      and      thus      we      can      use      a      GNN      to      obtain      the      graph      embeddings      of      proteins      and      drugs.      Finally,      we      feed
the     graph     embedding     into     a     linear     layer     with     sigmoid     activation     to     predict     DPIs
protein             feature             extraction             part,             drug–protein             bridge             graph             con-                                                                                                                                                                                     Pi         ¼       Plocali                                          þ      Pglobali                                             ;                                                                                                                                                                                                                                                                       (1)
struction       part       and       classification       part.       For       the       drug       and       protein       fea-
ture          extraction          parts,          the          Convolutional          Neural          Network          (CNN)                                                                                                                          where          Plocali                                           2      Rdh             and          Pglobali                                                       2      Rdh             are          protein          i’s          local          and          global
layers         and         some         Feed-Forward         Network         (FFN)         layers         are         applied                                                                                                                         features,       dh         is        the        dimension       of        them.        The        Plocali                                             and        Pglobali                                                         are        com-
to              extract              the              features              from              the              drug              SMILES              and              the              protein                                                        puted         by         Equation         (2),         pi         2      Rlp /C2  20           is         the         one-hot         embedding         of         the
sequences.           For           the           drug–protein           bridge           graph           construction           part,                                                                                                                 protein                 i,                 ai         2      R8420                   is                 the                 concatenated                 vectors                 of                 the                 1-mer,
some       bridge       nodes       are       introduced       to       construct       the       bridges       between                                                                                                                               2-mer,      3-mer      statistic      vectors.
proteins        and        drugs,        and        thus        we        can        use        a        Graph        Neural        Network
(GNN)         to          capture         the         network-level         information          to          predict         DPIs.                                                                                                                                                                                  Plocali                                                                           ¼       FFNlocalp                              ðCNNp ðpi ÞÞ
For       the       classification       part,       we       get       element-wise       product       of       the       pro-                                                                                                                                                                                    Pglobal
teins’          and          drugs’          graph          embeddings          after          GNN,          and          then          feed          it                                                                                                                                                                 i                                                                           ¼       FFNglobalp                                          ðai Þ                                                                                         ;                                                                                                                                                                                         (2)
into          a         linear         layer         with          sigmoid          activation         to         predict          the         interac-                                                                                               where              FFNlocal
tions.          In          this          section,          we          will          describe          the          details          of          each          compo-                                                                                                                        p                              ð/C1Þ                      is              a              one-layer              FFN,              FFNglobalp                                          ð/C1Þ                      is              a              two-layer
nent       in      BridgeDPI.                                                                                                                                                                                                                         FFN,       CNNp ð/C1Þ              is       the       1D       CNN       with       max-pooling.       And       the       output       of
                                                                                                                                                                                                                                                      them      is       all       activated      by      the       Rectified      Linear      Unit       (ReLU)      function.
2.2.1       Feature       extraction       of       proteins                                                                                                                                                                                          2.2.2       Feature       extraction       of       drugs
Before       feeding       proteins       into       BridgeDPI,       we       need       to       describe       them                                                                                                                                After         representing         proteins,         drug         molecules         also         need         to         be         vectorized.
as          numeric          vectors.          In          order          to          better          describe          the          properties          of                                                                                           Analogously,       we       also       extract       the       local       view’s       and       global       view’s       features
protein      sequences,      we      vectorize      them      from      both      a      local      and      global                                                                                                                                   from        drug        molecules.        Different        from        the        protein        sequences,        drug        mole-
perspective.         For         the         local         view,         we         employ         some         CNN         filters         to                                                                                                        cules          are          graphs          with          atoms          as          nodes          and          chemical          bonds          as          edges.          It
capture           the           key           local           patterns           in           the           protein           sequences.           Firstly,                                                                                           means         that         the         statistics         of         k-mer-like         information         are         not         appropriate
the             one-hot             encoding             is             used             to             encode             the             protein’s              primary                                                                             anymore.            Therefore,            we            choose            another            global            view’s            representation
amino             acid             sequences.             Then,             a             1D             CNN             with             max-pooling             is                                                                                  technique:          molecular          fingerprint.          The          molecular          fingerprint          encodes          a
applied           to           extract           the           local           features.           Finally,           we           transform           the                                                                                            drug         molecular         into         a         series         of         binary         digits,          where         some         substructure
CNN’s         output         by         a         two-layer         FFN         to         get         the         final         local         features                                                                                               and         topological         information         are         implied         (Rogers         and         Hahn,         2010).        As
of         the        protein.        For         the         global        view,        we        select         the        protein’s           k-mer                                                                                                for         the         local         view’s         representation,         similar         to         proteins,         we         also         employ
statistics              as              its              global              features              because              the                k-mer              information                                                                             some           CNNs.           Firstly,           we           encode           each           atom           of           the           drug           into           a           75-di-
                                                                                                                                                                                                                                                      mension             vector,             which             contains            physical–chemical             features             of             atoms
reveals              the              distribution              of              global              characteristics              and              measures                                                                                            and         bonds         (Ramsundar       et         al.,         2019).         Next,         a         CNN         with         max-pooling
biological            similarity            for            discrimination            ( Leslie             et            al.,2004).            In                                                                                                      is               used               to               extract               features               from               the               75-dimension               vectors               and               the
our            study,            we            set             k     ¼    1,            2,            3,            which            generate            20-            ( k     ¼    1),            400-                                              extracted          features          are          fed          into          a          3-layer          FFN          to          obtain          the          final          local
(20     /C2       20,            k     ¼    2)           and           8000-           (20     /C2       20     /C2       20,            k     ¼    3)            dimension           vec-                                                            features       of        the       drug       molecules.       Specifically,        for       drug    j       with      ld         atoms,       its
tors,          respectively.          The          vectors          are          normalized          and          concatenated                                                                                                                        final       representation     Dj         2      Rdh           is       defined       as:
to       represent       the       protein       from       a       global       view.       Here,       we       give       up          k       /C21
4             because              it              generates              too              many              (  /C21          160       000)              dimensions               and                                                                                                                                               Dj         ¼       Dlocalj                                          þ      Dglobalj                                             ;                                                                                                                                                                                                                                                             (3)
makes         the         method         too         complex,         easily         leading         to         overfitting         and
time-consuming       training.       Specifically,       for       protein          i       with       length         lp ,                                                                                                                            where            Dlocalj                                           2      Rdh               and            Dglobalj                                                       2      Rdh                are            drug            j’s            local            and            global
its      final      representation       Pi         2      Rdh         is      vectorized      as:                                                                                                                                                    features,      dh        is      the      dimension      of      them      which      is      equal      to      the      dimension

2574                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Y.Wu       et       al.
                                                                                                                                                                                                                                                                                                                                                                                                                 0           Pi         1
of           proteins’           final           representations.           The           Dlocalj                                                and           Dglobalj                                                            are           calcu-                                                                                                                                                          B          Di          C
lated        by        Equation        (4),        dj         2      Rld /C2  75          is        the        75-dimension        per-atom        fea-                                                                                                                                                                                                                                                          B                      C
                                                                                                                                                                                                                                                                                                                                                                                  Z0                             ¼B        B1           C
                                                                                                                                                                                                                                                                                                                                                                                                                 B                      C
tures,       bj         2      R1024        is       the       molecular      fingerprint      of       the       drug      j.                                                                                                                                                                                                                                                                                   @         ...          A
                                                                                                                                                                                                                                                                                                                                                                                                                          Bm                                                                                        ;                                                                                                                                                                                    (8)
                                                                         Dlocalj                                                                           ¼       FFNlocald                              ðCNNd ðdj ÞÞ                                                                                                                                                            Z1                             ¼       ReLUðLði/C0               jÞ Z0 W 1       þ      b1 Þ
                                                                         Dglobalj                                                                           ¼       FFNglobald                                          ðbj Þ                                                                                        ;                                                                                                                                                                                      (4)Z2                             ¼       ReLUðLði/C0               jÞ Z1 W 2       þ      b2 Þ
                                                                                                                                                                                                                                                                                                                                                                                  Z3                             ¼       ReLUðLði/C0               jÞ Z2 W 3       þ      b3 Þ
where            FFNlocald                              ð/C1Þ                    is             a             one-layer            FFN,            FFNglobald                                          ð/C1Þ                    is             a             three-layer
FFN,       CNNd ð/C1Þ              is       the       1D       CNN       with       max-pooling.       And       the       output       of
them      is       all       activated      by      the       ReLU      function.                                                                                                                                                                                                                        where            Z0         2      Rðmþ2Þ/C2  dh               is            the            initial            node            embedding            of            the            graph,
                                                                                                                                                                                                                                                                                                         Z1 ;    Z2 ;    Z3         2      Rðmþ2Þ/C2  dh                are            the             node            embeddings            after             aggregating
                                                                                                                                                                                                                                                                                                         the                neighbor                information,                W 1 ;    W 2 ;    W 3         2      Rdh /C2  dh                    and                b1 ;    b2 ;    b3         2
2.2.3       Bridge       graph’s       construction                                                                                                                                                                                                                                                      R1/C2  dh           are        the        hidden       parameters       of       the        GNN,       ReLUð/C1Þ               is        the        activa-
After         obtaining         the         final         representations         of         proteins         and         drugs,                                                                                                                                                                         tion      function.
the       next       thing       is       to       introduce       the       network-level       information.       To                                                                                                                                                                                                 Finally,          we           select           the           first           two           rows           of           Z0 ;    Z1 ;    Z2 ;    Z3             and           sum
do        this,        we        construct        a       supervised        drug–protein        association        net-                                                                                                                                                                                  them         up         to         get         the         final         vectors         of         protein         i         and         drug         j         which         have
work,         called         bridge         graph.         Specifically,         we         introduce         a         class         of                                                                                                                                                                 aggregated      the       network-level       information.
nodes         called         bridge         nodes         to         the         constructed         network         to         super-
visedly          measure          the          associations          among          proteins/drugs          and          the                                                                                                                                                                                                                                                                                             ^                                3
interactions          between          drug–protein          pairs.          The          bridge          nodes          are                                                                                                                                                                                                                                                                                           Pi                                    ¼    XZk1
actually           some             dh -dimension           vectors           in           the           space           of              Pi           and             Dj,                                                                                                                                                                                                                                                                             k¼0                      ;                                                                                                                                                                                                                                                                                                  (9)
                                                                                                                                                                                                                                                                                                                                                                                                                                                          3
and         their          associations         are          defined         as          the         cosine         similarities          be-                                                                                                                                                                                                                                                                          Dj                              ¼    X^     Zk2
tween      them:                                                                                                                                                                                                                                                                                                                                                                                                                                      k¼0
                                                                                           aPi ;Bk                                   ¼         Pi       /C1              Bk                                                                                                                              where       Zk1         2      R1/C2  dh            is        the        first        row        of       Zk,       Zk2         2      R1/C2  dh            is        the        second        row
                                                                                                                                    jjPi jj2 jjBk jj2                                   ;                                                                                                                                                                                                                                            (5)of      Zk,            ^Pi   ;              ^Dj           2      R1/C2  dh           are       the       final       vectors       of       protein    i       and       drug      j.
                                                                                           aDj ;Bk                              ¼              Dj       /C1             Bk
                                                                                                                                    jjDj jj2 jjBk jj2
where          Pi         2      Rdh            and          Dj         2      Rdh            are         the         final         representations         of         pro-                                                                                                                              2.2.4       Classification
tein          i          and         drug          j,          Bk         2      Rdh            represents         a         bridge         node,           /C1                is         to         com-                                                                                                After         aggregating         the         network-level          information         to         the         protein         and
pute          the          inner          product          of          vectors,             jj      /C1      jj       2            is          to          compute          the          two-                                                                                                            drug       representations,       the       last       thing       is       to       infer       whether       the       drug–pro-
norm      value      of      a      vector.      The      vectors      of      the      bridge      nodes      are      initial-                                                                                                                                                                         tein         pair         interacts.         We         use         element-wise         product         of         drugs’         and         pro-
ized        randomly        from        a        normal        distribution                   Nð0;    1Þ.        In        our        model,                                                                                                                                                             teins’         final         vectors          to         model          the          interaction         mechanism         and          after          a
we       employ         m        bridge       nodes        (i.e.          B1 ;    B2 ;   ...    ;    Bm )        to        jointly        measure                                                                                                                                                        two-layer       FFN      the      interaction      probabilities       are       predicted:
the           associations           and           the           interactions:           for           the           protein–drug           pair                                                                                                                                                                                                                                         y ði/C0               jÞ                             ¼       FFNoutput ð     ^^ Pi         /C12                   ^Dj   Þ    ;                                                                                                                                                                                       (10)
ði      /C0                            jÞ,         their         interaction         can         be         measured         by              aPi ;Bk             and            aDj ;Bk ;for
another           protein–drug           pair             ðu      /C0                            vÞ,           their           interaction           can           be           also                                                                                                                     where            /C12                is            to            compute            the            element-wise            product            of            two            vectors,
measured        in        the        same        way;        moreover,        if        we        think        about        the        both                                                                                                                                                              FFNoutput          is        a        two-layer        FFN        with        ReLU        activation        for        its        first        layer
pairs         ði      /C0                            jÞ        and         ðu      /C0                            vÞ,        with        the        bridge        nodes        as        its        medium,        the                                                                                   and            Sigmoid            activation            for             its            second            layer,               ^y ði/C0               jÞ         2      R             is            the            pre-
association           between           protein              i            and           protein             u    canbeinferredby                                                                                                                                                                         dicted        interaction        probability.         In         order         to         make         the         predicted        inter-
aPi ;Bk           and          aPu ;Bk   ,        and        the        association        between        drug j        and        drugv         can                                                                                                                                                     action            probabilities            close            to            the            true            interaction            values,            we            use            a
be       inferred       in       the       same       way.       In       other       words,       the       associations’       and                                                                                                                                                                     binary        cross        entropy        loss        function        as        our        training        objective,        and        the
the      interactions’      information,      or      the      network-level      information,      is                                                                                                                                                                                                   L2      regularization      is       added      to       improve      the       model’s      robustness.
embedded          in          the          constructed          network          with             m           bridge          nodes           and                                                                                                                                                                                              1              X
protein/drug          nodes,          and          thus          we          can          use          a          GNN          to          capture          the                                                                                                                                                  L¼                        jP j                                 ½yði/C0               jÞ      logð ^y ði/C0               jÞ  Þþð1      /C0                            yði/C0               jÞ Þlogð1      /C0                                ^y ði/C0               jÞÞ Þ/C138
information.                                                                                                                                                                                                                                                                                                                                             ði/C0               jÞ2P                                                                                                                                                                                   ;                   (11)
              In               detail,              for               the              protein–drug              pair              ði     /C0                            jÞ               and              the               m               bridge                                                                                        þk  Ph2H      khk2
nodes,      we       first      compute      the      cosine      similarity      between      them      to       obtain                                                                                                                                                                                 where             P               is              the              set              of              all              protein–drug              pairs              in              the              training              set,
a      weighted      adjacency       matrix      of       the       graph:                                                                                                                                                                                                                               yði/C0               jÞ         2      R          is          the          true          interaction          value          for          protein–drug          pair         ði      /C0                            jÞ,
                                                                        0                                                                                                                                                         1
                                                                        B           aPi ;Pi                                              aPi ;Dj                                             aPi ;B1                        ...                                    aPi ;BmC
                                                                        B          aDj ;Pi                                         aDj ;Dj                                         aDi ;B1                       ...                                  aDi ;BmC                                           Table      1.      The       setting      of       hyper-parameters      in       BridgeDPI
                                                                        B          aB1 ;Pi                                        aB1 ;Dj                                        aB1 ;B1                      ...                                  aB1 ;BmC
                                      Aði/C0               jÞ         ¼ B                                                                                                                                                         C;                                                                                  (6)
                                                                        B               ...                                       ...                                        ...                            ...                             ...C                                                         Hyper-parameter                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Value
                                                                        @                                                                                                                                                         A
                                                                                  aBm ;Pi                                    aBm ;Dj                                   aBm ;B1                     ...                               aBm ;Bm
                                                                                                                                                                                                                                                                                                         Protein                                                                                                                                                                                Filter      size      of      CNNp                                                                                                                                                                                                                                                         25
where         Aði/C0               jÞ         2      Rðmþ2Þ/C2ð  mþ2Þ           is         the         weighted         adjacency         matrix,          a/C1     ;/C1                is                                                                                                                                                                                                      Filter      num      of      CNNp                                                                                                                                                                                                                                               64
the        cosine        similarity        between        nodes        defined        as        Equation       (5).        For        the                                                                                                                                                                                                                                                       Neuron      num      of      FFNlocalp                                                                                                                                                                                                                        128
stability       of       convergence,       we       filter       out       the       negative       edges       and       do       nor-                                                                                                                                                                                                                                                        Neuron      num      of      FFNglobalp                                                                                                                                                                                   1024,      128
malization       for       Aði/C0               jÞ :                                                                                                                                                                                                                                                     Drug                                                                                                                                                                                                        Filter      size      of      CNNd                                                                                                                                                                                                                                                                 7
                                                                                                                                                                                                                                                                                                                                                                                                Filter      num      of      CNNd                                                                                                                                                                                                                                               64
                                                                                                                            1                                                                   1
                                                                              Lði/C0               jÞ         ¼       D/C0  2 ReLUðAði/C0               jÞ ÞD/C0                                2 ;                                                                                                                                                                                                        (7)Neuron      num      of      FFNlocald                                                                                                                                                                                                                        128
                                        1                                                                                                                                                                                                                                                                                                                                                       Neuron      num      of      FFNglobald                                                                                                                                                      1024,      256,      128
where         D/C0                      2         2      Rðmþ2Þ/C2ð  mþ2Þ           is         the         degree         matrix         of         Aði/C0               jÞ ;         ReLUð/C1Þ                 is                                                                                       Bridge      graph                                                                                                                Num      of      bridge      nodes                                                                                                                                                                                                                                    64
to       filter      out      the       negative      edges.                                                                                                                                                                                                                                                                                                                                    Neuron      num      of      GNN                                                                                                                                                           128,      128,      128
              Then,         to         capture         the         network-level         information,        a         3-layer         GNN                                                                                                                                                               Classiﬁcation                                                                                                         Neuron      num      of      FFNoutput                                                                                                                                                              128,      1
is       implemented       as:

BridgeDPI                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              2575
Table      2.      Performances       of       BridgeDPI       and      baselines       on      BindingDB       dataset
Methods                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Overall      test      set                                                                                                                                                                                                                                                                            Seen      protein      set                                                                                                                                                                                                                                               Unseen      protein      set
                                                                                                                                                                                   ACC                                                 AUROC                                                                AUPR                                               ACC                                                 AUROC                                                             AUPR                                             ACC                                               AUROC                                                              AUPR
                                                                                                                                                                                     (%)                                                          (%)                                                             (%)                                             (%)                                                        (%)                                                           (%)                                          (%)                                                        (%)                                                          (%)
Tiresias      (Fokoue      et      al.      (2016)                                                                                                                                                  —                                                                                                                                    —                                                                                                                                           —                                                                                      91.5                                                                                                        93.9                                                                                                                        —                                                                                           —                                                                                                                 68.0                                                                                                                        —
E2E/GO      (Gao      et      al.      (2018)                                                                                                                                                                 78.3                                                                                                                       —                                                                                                                                           —                                                                                      80.8                                                                                                        92.2                                                                                                                        —                                                                               75.3                                                                                                     89.4                                                                                                                        —
E2E      (Gao      et      al.      (2018)                                                                                                                                                                                                                   84.8                                                                                                                       —                                                                                                                              91.0                                                                         85.0                                                                                                        91.6                                                                                                                        —                                                                               84.6                                                                                                     90.5                                                                                                                        —
CPI-GNN      (Tsubaki      et      al.      (2019)                                                                                                  83.2                                                                                                                       —                                                                                                                                           —                                                                                      93.0                                                                                                        97.0                                                                                                                        —                                                                               71.5                                                                                                     73.8                                                                                                                        —
DrugVQA      (Zheng      et      al.      (2020)                                                                                                                  88.7                                                                                                           93.6                                                                                                                              —                                                                                      91.0                                                                                                        96.0                                                                                                                        —                                                                               86.0                                                                                                     92.2                                                                                                                        —
GraphDTA      (Nguyen      et      al.      (2021)                                                                                      85.5                                                                                                           93.6                                                                                                                  93.4                                                                         94.7                                                                                                        98.2                                                                                                            97.8                                                                  74.4                                                                                                     82.7                                                                                                            82.7
TransformerCPI      (Chen      et      al.      (2020)                                                            89.3                                                                                                           95.7                                                                                                                  95.8                                                                         94.9                                                                                                        98.6                                                                                                            98.5                                                                  82.6                                                                                                     90.7                                                                                                            91.1
BridgeDPI                                                                                                                                                                                                                                                                                                                                                               93.0                                                                                                           97.5                                                                                                                  97.3                                                                         96.1                                                                                                        98.9                                                                                                            98.6                                                                  89.3                                                                                                     95.8                                                                                                            95.5
H      is      the      set      of      all      parameters      in      our      model,       k      is      an      adjustable      regular-                                                                                                                                                                                                        Table                      3.                       Performances                        of                       BridgeDPI                        and                       baselines                        on                       the
ization      coefficient       that      balances      the       terms.                                                                                                                                                                                                                                                                                C.ELEGANS       and      the       HUMAN      datasets
2.3        Implementation        details                                                                                                                                                                                                                                                                                                               Methods                                                                                                                                                                                                 C.ELEGANS                                                                                                                                                                                               HUMAN
We        use        Pytorch       1.6.0        (Paszke        et        al.,        2019)        to        implement        BridgeDPI.                                                                                                                                                                                                                                                                                                          AUROC      (%)                                             F1      (%)                                             AUROC      (%)                                             F1      (%)
The              Adam              (Kingma              and              Ba,              2019)              optimizer              with              learning              rate
0.001         is         used         in         our         experiments.         And         the         L2         regularization         coeffi-                                                                                                                                                                                                    k-NN                                                                                                                                                                                                                      85.8                                                                                                             81.4                                                                                                             86.0                                                                                                             85.8
cient       k      is      set      to       0.001.      For       each      epoch,      the      data      of       protein–drug      pairs                                                                                                                                                                                                           RF                                                                                                                                                                                                                                                        90.2                                                                                                             83.2                                                                                                             94.0                                                                                                             87.9
are           randomly           shuffled           and           the           batch           size           is           set           to           512.           BridgeDPI                                                                                                                                                                        LR                                                                                                                                                                                                                                                       89.2                                                                                                             88.3                                                                                                             91.1                                                                                                             90.2
will        be        trained       for        100       epochs        and        the        model        with        the        best        AUROC                                                                                                                                                                                                     SVM                                                                                                                                                                                                                               89.4                                                                                                             80.1                                                                                                             91.0                                                                                                             95.8
on       the        validation       set       will        be        retained.       For       the        setting       of        other       hyper-                                                                                                                                                                                                   E2E/GOa                                                                                                                                                                              98.6                                                                                                             95.0                                                                                                             97.0                                                                                                             90.3
parameters,           such          as           the          number          of           layers,          the          number          of          neurons,                                                                                                                                                                                          CPI-GNN                                                                                                                                                                     97.8                                                                                                             93.3                                                                                                             97.0                                                                                                             92.0
the              ratio              of              dropout              and              others,              experiments              are              carried              out              to                                                                                                                                                      DrugVQA                                                                                                                                                                               —                                                                                                                                      —                                                                                                                          97.9                                                                                                             95.7
choose         their         values         according         to         the         performance         on         the         validation                                                                                                                                                                                                             GraphDTA                                                                                                                                                       97.4                                                                                                             91.9                                                                                                             96.0                                                                                                             89.7
set.          All          of          the          experimental          processes          are          running          on          one          NVIDIA                                                                                                                                                                                             TransformerCPI                                                                                               98.8                                                                                                             95.2                                                                                                             97.3                                                                                                             92.0
GeForce      RTX       1080       Ti       GPU.       And       the       final       hyper-parameters’       setting                                                                                                                                                                                                                                  BridgeDPI                                                                                                                                                                    99.5                                                                                                             97.0                                                                                                             99.0                                                                                                             94.9
of       BridgeDPI       is       shown      in      Table       1.
                For        proteins,       we        use        a       1D        CNN,       which       has       64       filters        with       width                                                                                                                                                                                                       aGO       feature        required       by       E2E       needs       to       be       obtained        from       the       UniProt       data-
25       to       extract       its       local       features.       After       that,       a       one-layer       FFN       is       used       to                                                                                                                                                                                                 base,          and          there           are          a          large          number          of          GO          missing           for          these           two          datasets.
transform       the        local        features        to        a        128-dimension        space.       For        protein’s                                                                                                                                                                                                                      Therefore,       we        only        reproduced        the        results        of        the        E2E/GO        [without        (the)        GO
global         features,        we         introduce        a         two-layer         FFN         for         non-linear        trans-                                                                                                                                                                                                               feature]      model.
formation,                 containing                1024                and                 128                 neurons,                respectively.                 For
drugs,        we         employ        a         1D         CNN        with        64         filters        of         width        25         to         extract
its          local          features.          Then,          a          one-layer          FFN          is          also          used          to          transform                                                                                                                                                                                 Table        2        shows        the        results        of        BridgeDPI        and        other        baselines        on        these
the        local       features        to        the        128-dimension       space.       For        drug’s       global        fea-                                                                                                                                                                                                                two       test       sets.       The       symbol       ‘—’       indicates       the       absence       of       results,       which
tures,         we         introduce        a         three-layer         network        for         non-linear        transform-                                                                                                                                                                                                                       means              that              the              experimental              results              are              not              present              in              the              cited
ation,            containing            1024,            256            and            128            neurons,            respectively.            Their                                                                                                                                                                                               paper.           Overall,           BridgeDPI           achieves           SOTA           performance           in           the           test
outputs         are         all         128-dimension         vectors         that         serve         as         the         embeddings                                                                                                                                                                                                             set.       The       accuracy       (ACC),       AUROC       and        the       area       under       the       precision-
of            protein/drug            nodes            on            the            bridge            graph.            For            bridge            graph,            we                                                                                                                                                                          recall           curve           (AUPR)           of           BridgeDPI           have           reached           to           93.0%,           97.5%
introduce       64       bridge       nodes       with       the       same       128       dimensions       as       protein/                                                                                                                                                                                                                         and               97.3%,              respectively,               which              are               significantly               superior               to               the
drug       nodes       in       the       graph.       Then,       we       use       a       three-layer       GNN       to       capture                                                                                                                                                                                                             previous      method.
the        network-level        information,        which        means        that        each        node        can        ag-                                                                                                                                                                                                                                        For       the       seen       protein       test       set,       we       find       that       the       results       in       all       models
gregate         at         most         three-depth         neighbor          information         outwards.         In          the                                                                                                                                                                                                                    are                 decent,                 with                 AUROC                 and                 ACC                 generally                 exceeding                 90%.
end,          the           scores           of           the          interactions          between           proteins          and           drugs           are                                                                                                                                                                                     Tiresias,        which        infers        DPIs        by        constructing        a        large-scale        association
obtained        through        a        two-layer        FFN         with        128,         1         neurons.        Moreover,                                                                                                                                                                                                                      network,        has        a        relatively        low        AUROC        of        93.9%.        The        potential        rea-
we           also           use           the          dropout          technique          (Srivastava          et           al.,          2014)           to           im-                                                                                                                                                                            son          is          that           the          DPIs          do          not          necessarily          exist          between          proteins          and
prove      BridgeDPI’s       generalization       and      the      dropout      rate       is       set      to       0.5.                                                                                                                                                                                                                            drugs      with       high      similarity      information.       This       also      proves       that      it       is       not
                                                                                                                                                                                                                                                                                                                                                       enough         to         rely         solely         on         the         ‘guilt-by-association’         principle         to         infer
                                                                                                                                                                                                                                                                                                                                                       DPIs.          Other          learning-based          approaches         adequately         learn          the          mech-
3        Results                                                                                                                                                                                                                                                                                                                                       anism       of       DPIs       by      the       priori      DPI       information,       increasing       the       AUROC
                                                                                                                                                                                                                                                                                                                                                       to           around           97%,           suggesting           that           learning           to           model           the           interaction
3.1        Performance        on        the        BindingDB        dataset                                                                                                                                                                                                                                                                            mechanism           is           significant           and           necessary           for           DPI           prediction           (except
To           evaluate           the           performance           of           BridgeDPI           in           predicting           DPIs,           we                                                                                                                                                                                              for        E2E,        which       we        speculate        the        reason        is        that        the        E2E        model        is        too
compare         BridgeDPI          with         some         methods         including         Tiresias         (Fokoue                                                                                                                                                                                                                                partial           to           the           performance           on           the           unseen           proteins           but           ignores           the
et             al.,             2016),             E2E             (Gao             et             al.,             2018),             CPI-GNN             (Tsubaki             et             al.,                                                                                                                                                    seen          proteins).          Among          them,          BridgeDPI          achieves          the          best          AUROC
2019),          DrugVQA          (Zheng         et          al.,          2020),          GraphDTA          (Nguyen         et          al.,                                                                                                                                                                                                           and           ACC,          reaching          98.9%           and           96.1%,          respectively.           Compared          to
2021)           and           TransformerCPI           (Chen           et           al.,           2020)           on           the           BindingDB                                                                                                                                                                                                other       baselines,       BridgeDPI       not       only       pays       attention       to       learn       the       inter-
dataset.      In       addition,      we       believe      that       a       practical      model      should      be       able                                                                                                                                                                                                                     action        mechanism        but         also        combines         a        network-level         perspective         to
to       handle       unseen       proteins       because       of       a       mass       of       unknown       proteins       in                                                                                                                                                                                                                   assist          learning.          Consequently,          our          model          can          obtain          more          compre-
nature.           Thus,           we           follow           Zheng           et           al.           (2020)           to           divide           the           test           set                                                                                                                                                             hensive        feature        expressions        of        proteins        and        drugs        that        aggregate        the
into       a       seen       protein       test       set       and       an       unseen       protein       test       set,       and       we       in-                                                                                                                                                                                            network-level           information.          And           the           expressions           will           be           more           con-
vestigate                 the                 performance                 of                 BridgeDPI                 in                 these                 two                 test                 sets.                                                                                                                                         ductive      to       the       learning      of       the       DPI       mechanism.

2576                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Y.Wu       et       al.
               For          the          unseen          protein          test          set,          it          is          clear          that          the          difference          in                                                                                                                                learning        models         have         strong         feature         extraction         abilities         to         learn         the
the         performance         of         these         methods         varies         greatly.         Tiresias         yields         the                                                                                                                                                                                  interaction           rules.           On           this          basis,           BridgeDPI           integrates          the           network-
worst       AUROC       of       68.0%,       which       means       this       method       cannot       be       easily                                                                                                                                                                                                    level             information             and             the             learned             interaction             mechanism,             further
applied          to          unseen          proteins.          This          is          easy          to          understand          because          the                                                                                                                                                                  improving       the       results.
unseen        proteins        are         some        isolated        nodes         in        the         drug–protein         associ-
ation           network,           lacking           enough           neighbor           information           to           infer           how
they       interact.       Other       learning-based       approaches      make       some       improve-                                                                                                                                                                                                                    3.3        Performance        on        an        independent        test        set
ments        by        learning        the        interaction        mechanism        of        drugs        and        proteins.                                                                                                                                                                                             Although        we        have        achieved        excellent        results        on        these        benchmark
Among           them,           BridgeDPI           outperforms           other           methods           and           achieves                                                                                                                                                                                            datasets,         such         datasets        have         serious         data         bias,        which         will         lead         to
SOTA           performance,           with           AUROC           and           ACC           reaching           95.8%           and                                                                                                                                                                                       the       inflated       performance       (Chen       et      al.,2019;       Yang       et      al.,2020).      In
89.3%       in       unseen       proteins,       3.9%       and        3.8%       higher       than       the        previous                                                                                                                                                                                                order           to           investigate           the           realistic           performance           of           our           model,           we
optimal       method       (DrugVQA).       It       indicates       that       introducing       the       bridge                                                                                                                                                                                                            conduct                        the                        following                        experiments:                        train                        models                        on                        the
nodes       indeed       improves       the       performance       on       the       unseen       protein       set       by                                                                                                                                                                                                BindingDB                  dataset                  and                  test                  models                  on                  the                  DUD-E                  dataset.
incorporating          the          network-level          information          and          the          learned          inter-                                                                                                                                                                                             Additionally,          we          propose          an          evaluation          indicators          called          pP@k,
action              mechanism.               Moreover,              the               constructed               bridge               graph              also                                                                                                                                                                  which      is      defined      as      the       average      protein-level       precision      at      the      top      k
enables            BridgeDPI            to            learn            some            deeper            interaction            rules            because                                                                                                                                                                      predictions         (as         shown         in         Formula         12).         pP  @k         indicates        the         accur-
each              information              aggregation              in              the              GNN              is              like              an              interaction                                                                                                                                           acy      of      the      model      for      k      recalled      drugs      from      the      protein      level,      which
among       proteins       or       drugs,       or       between       proteins       and       drugs.       And       this       is                                                                                                                                                                                         can        reasonably        evaluate        the        reliability        of        DPI        prediction        methods
why       BridgeDPI       is       more       accurate       in       predicting       DPIs       on       both       the       seen                                                                                                                                                                                          in      drug      screening.
protein      test      set      and      the       unseen       protein      test      set.                                                                                                                                                                                                                                                                                                                                                                                          P
                                                                                                                                                                                                                                                                                                                                                                                                                                                            1        Xm                         j2^s k
3.2        Performance        on        the        C.ELEGANS        and        HUMAN                                                                                                                                                                                                                                                                                                                                 pP@k       ¼                                                                         r   ðiÞ      yði/C0               jÞ
                                                                                                                                                                                                                                                                                                                                                                                                                                                           m           i¼1                                   k                                                                    ;                                                                                                                                                                                                 (12)
datasets
Furthermore,               we               also               conduct               experiments               on               the               C.ELEGANS                                                                                                                                                                   where       m       is       the       number       of       proteins,       ^s kr    ðiÞ       is       the       set       of       k       most       possible
dataset           and           HUMAN           dataset           (Liu           et           al.,           2015),           which           are           widely                                                                                                                                                            drugs          recalled          by          the          model          for          protein          i,          yði/C0               jÞ          2f0;    1g          is          the          true
used                in                many                studies.                We                choose                k-Nearest                Neighbor                (k-NN),                                                                                                                                            interaction      value       for       the       pair      of      protein      i       and      drug      j.
Random                Forest                (RF),                Logistic                Regression                (LR),                Support                Vector                                                                                                                                                                         We        set       k       ¼       10;    20;    40;    80;    160       in       this       experiment.       The       results       are
Machine           (SVM),           E2E/GO,           CPI-GNN,           DrugVQA,           GraphDTA           and                                                                                                                                                                                                             shown       in        Table        4.        Not       surprisingly,        the        performances        of        these        mod-
TransformerCPI         as         the         baselines.         The         results         are         shown         in         Table         3.                                                                                                                                                                            els        are        all        greatly        reduced,        with        AUROC        of        the        SVM        even        less        than
Since         Gao         et          al.         (2018)         do         not         provide         the         code         of         E2E,         we          repro-                                                                                                                                                   50%.         Compared         with         other         models,         the         performance         of         BridgeDPI
duce            their            model            and            obtain            the            experimental            results            on            the            two                                                                                                                                                 is              the              best.              For              AUROC,              BridgeDPI              is              9.41%,              8.58%,              29.14%,
datasets.               We               used               the               codes               from               GraphDTA’s               (Nguyen               et               al.,                                                                                                                                     32.03%,            46.79%            higher            than            E2E/GO,            KNN,            RF,            LR,            SVM,            re-
2021)         and         TransformerCPI’s         (Chen         et         al.,         2020)         github         to         generate                                                                                                                                                                                     spectively.            Moreover,            if            the            whole            BindingDB            dataset            is            used            for
their       results       on       the       BindingDB       dataset.       The       results       of       other       baselines                                                                                                                                                                                            training,               the                AUROC               of                BridgeDPI               and                E2E/GO                will                reach                to
are        from        their        original        papers.        As        can        be        seen        from        Table        3,        for        the                                                                                                                                                               77.2%       and       74.8%,       respectively.       For       pP@k,       BridgeDPI       can       accurate-
randomly           divided           C.ELEGANS           and           HUMAN           datasets,           almost           all                                                                                                                                                                                               ly          recall          more          than          60%         of          drug          candidates         at          k            /C20               80,          which          sig-
proteins        in       the        test       set       are        seen       proteins,       which       means       that       the        mod-                                                                                                                                                                             nificantly           outperforms           other          compared           methods.          And          if          the          whole
els            can            learn            almost            all            of            the            protein            information            well            from            the                                                                                                                                    BindingDB       dataset       is       used       for       training,       BridgeDPI       can       accurately       re-
training         dataset,         resulting         in         very         good         results.         In         this         case,         the         un-                                                                                                                                                               call       more       than       80%       of       drug       candidates       at       k            /C20               40.       The       results       show
supervised        k-NN        is        slightly        worse        than        other        models,        with        AUROC                                                                                                                                                                                                the       reliability       of       BridgeDPI       and       that       BridgeDPI       performs       better       even
85.8%        and        F1        81.4%        on        the        C.ELEGANS        dataset,        AUROC        86.0%                                                                                                                                                                                                       under      a       more      realistic      condition.
and      F1       85.8%      on       the       HUMAN      dataset,       respectively.       In       contrast,       the
traditional            supervised           machine           learning           methods           (i.e.            RF,           LR           and
SVM)         are         slightly         better,         with         AUROC         of         the         C.ELEGANS         dataset
reaching       around      90.0%,       AUROC      of       the       HUMAN       dataset      exceeding
91.0%.       The       deep       learning       methods       E2E/GO,       CPI-GNN,       DrugVQA,
GraphDTA,        TransformerCPI        and        BridgeDPI        all        reach        excellent        per-
formance,             with             AUROC             over             96.0%             and             F1             over             89%.             Among
them,          BridgeDPI          achieves          the          best          performance,          with          AUROC,          F1
of             99.5%,             97.0%             on             the             C.ELEGANS              dataset,             respectively,             and
99.0%,            94.9%            on            the            HUMAN            dataset,            respectively.            The            results
are        in        line        with        our        expectations.        Because        the        models        such        as        KNN,
RF,             LR             and              SVM,             without             high-quality             features,             are             difficult             to
learn         complex         non-linear        relationships         among        DPIs,         while         the         deep                                                                                                                                                                                               Fig.     3.     Performance     of     BridgeDPI     with     different     number     of     bridge     nodes
Table      4.      Performances       of       different       methods       on      an      independent       test       set
Methods                                                                                                                                                         AUROC      (%)                                                                                                                                  pP@10      (%)                                                                                                                                  pP@20      (%)                                                                                                                                  pP@40      (%)                                                                                                                                  pP@80      (%)                                                                                                                                  pP@160      (%)
Training      on      the      customized      BindingDB      dataset      (Gao      et      al.,      2018).
         k-NN                                                                                                                                                                                                                 65.3                                                                                                                                                                                                                           40.3                                                                                                                                                                                                                  41.3                                                                                                                                                                                                                  40.9                                                                                                                                                                                                                  40.3                                                                                                                                                                                                                         37.8
         RF                                                                                                                                                                                                                                                   54.9                                                                                                                                                                                                                           42.0                                                                                                                                                                                                                  41.2                                                                                                                                                                                                                  40.5                                                                                                                                                                                                                  40.0                                                                                                                                                                                                                         37.6
         LR                                                                                                                                                                                                                                                  53.7                                                                                                                                                                                                                           47.9                                                                                                                                                                                                                  48.4                                                                                                                                                                                                                  48.6                                                                                                                                                                                                                  47.1                                                                                                                                                                                                                         42.2
         SVM                                                                                                                                                                                                                          48.3                                                                                                                                                                                                                           40.3                                                                                                                                                                                                                  39.7                                                                                                                                                                                                                  38.7                                                                                                                                                                                                                  37.8                                                                                                                                                                                                                         35.0
         E2E/GO                                                                                                                                                                                   64.8                                                                                                                                                                                                                           53.1                                                                                                                                                                                                                  54.8                                                                                                                                                                                                                  54.8                                                                                                                                                                                                                  52.7                                                                                                                                                                                                                         46.4
BridgeDPI                                                                                                                                                                                         70.9                                                                                                                                                                                                                           68.2                                                                                                                                                                                                                  67.5                                                                                                                                                                                                                  66.1                                                                                                                                                                                                                  61.6                                                                                                                                                                                                                         53.1
Training      on      the      full      BindingDB      dataset      (Gilson      et      al.,      2016).
         E2E/GO                                                                                                                                                                                   74.8                                                                                                                                                                                                                           70.1                                                                                                                                                                                                                  70.1                                                                                                                                                                                                                  67.5                                                                                                                                                                                                                  62.5                                                                                                                                                                                                                         53.3
BridgeDPI                                                                                                                                                                                         77.2                                                                                                                                                                                                                           85.2                                                                                                                                                                                                                  84.8                                                                                                                                                                                                                  81.5                                                                                                                                                                                                                  73.3                                                                                                                                                                                                                         60.4

BridgeDPI                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              2577
Table              5.               Prediction               results               of               possible               antiviral               drugs               and              viral                                                                predicting           new           drugs,           indicating           that           BridgeDPI           plays           a           guide           role           in
targets                                                                                                                                                                                                                                                      actual       research       and       drug       discovery.
Molecules                                                                                                                  Predicted      probability      (%)                                                                                                                                                        Source
3CLpro                                                                                                                                                                                                                                                       5        Conclusion
       Baricitinib                                                                                                                                                                                                            99.5                                                                                                                                                                                                  Kalil      et      al.      (2021)In       this       study,       we       propose       an       end-to-end       deep       learning       framework       to
       Remdesivir                                                                                                                                                                                                    97.0                                                                                                                                                                                                  Elﬁky      (2020)predict               DPIs               by               introducing               the               network-level               information               to               a
       Lopinavir                                                                                                                                                                                                                    82.3                                                                                                                                                                                                  Stower      (2020)learning-based         framework.         We         construct         a         supervised         drug–protein
       Ritonavir                                                                                                                                                                                                                       55.5                                                                                                                                                                                                  Stower      (2020)network             and             introduce             a             class             of             bridge             nodes             to             it.             The             bridge
       Aspirin                                                                                                                                                                                                                                                               3.7nodes      bridge      the       gap      among      drugs      and      proteins      by      information      pass-
RdRp                                                                                                                                                                                                                                                         ing        among        diverse        drugs        and         proteins,         and         thus        we         can        use         a        GNN
       Ivermectin                                                                                                                                                                                                           99.6                                                                                                                                                                                                  Caly      et      al.      (2020)to           capture          the           network-level           information          and           rely          on           a           supervised
       Remdesivir                                                                                                                                                                                                    98.3                                                                                                                                                                                                  Elﬁky      (2020)‘guilt-by-association’          to          perform          predictions.          Therefore,          our          model
       Sofosbuvir                                                                                                                                                                                                          97.8                                                                                                                                                                                                  Sadeghi      et      al.      (2020)integrates       more        comprehensive        features        by       taking        into       account       both
       Daclatasvir                                                                                                                                                                                                  94.0                                                                                                                                                                                                  Sadeghi      et      al.      (2020)the       advantages       of       network-based       methods       and       learning-based       meth-
       Lopinavir                                                                                                                                                                                                                    87.5                                                                                                                                                                                                  Stower      (2020)ods.            The            experiments            show            that            our            approach            outperforms            other
       Ritonavir                                                                                                                                                                                                                       60.6                                                                                                                                                                                                  Stower      (2020)competing          methods          on          BindingDB,          C.ELEGANS,           Human,          DUD-E
       Aspirin                                                                                                                                                                                                                                                               3.3datasets                 and                 achieves                SOTA                 performances.                 Moreover,                 the                 case
                                                                                                                                                                                                                                                             study           with           concrete           examples           also            reaffirms           the           usefulness           of           our
                                                                                                                                                                                                                                                             model.
3.4        Effect        of        the        number        of        bridge        nodes
By      introducing      the       bridge       nodes,      BridgeDPI      builds      multiple      bridges                                                                                                                                                 Funding
between          all          proteins          and          drugs.          Thus,          we          carry          out          further          study
about         the         influence         of         the         number         of         bridge         nodes.         We         apply         dif-                                                                                                     This      work      was      supported      in      part      by      the      National      Natural      Science      Foundation
ferent            numbers            (i.e.            1,            2,            4,            8,           16,            32,           64,            128,            256)            of            bridge                                                of       China       under       Grant       No.       61832019,       and       the       Human       Provincial       Science       and
nodes                to                observe                the                performance                on                the                BindingDB                dataset.                                                                           Technology      Program      [2019CB1007      and      2021RC4008].
Figure         3          shows         the         results          of         the         overall         test         set,         the         seen         protein                                                                                       Conflict      of      Interest:      none      declared.
set              and              the              unseen              protein              set              for              different              numbers              of              bridge
nodes.           As           we           can           see,            the           performances           of           different           numbers           of
bridge          nodes          are          stable          on          seen           proteins          but          fluctuate          greatly          on                                                                                                 References
unseen              proteins.              As              the              number              of              bridge              nodes              increases,              the
AUROC             and             AUPR             of             the             unseen             protein             set             are             continuously                                                                                        Avorn,J.      (2015)      The      $2.6      billion      pill–methodologic      and      policy      considerations.
improving,           and           thus           it           leads           a           better           performance           on           the           overall                                                                                                N.     Engl.     J.     Med.,     372,     1877–1879.
test              set.              This              is              in              line              with              our              respective              because              the              bridge                                              Ballester,P.J.         and         Mitchell,J.B.         (2010)         A         machine         learning         approach         to         pre-
nodes’        connection        to        all        proteins        and        drugs        means        the        unseen        pro-                                                                                                                             dicting      protein–ligand      binding      afﬁnity      with      applications      to      molecular      dock-
teins          are          no          longer          an          isolated          node,          which          will          make          predicting                                                                                                          ing.     Bioinformatics,     26,     1169–1175.
DPIs         easier.         When         the         number         of         bridge         nodes         is         64,         the         perform-                                                                                                     Bleakley,K.           and           Yamanishi,Y.           (2009)           Supervised           prediction           of           drug–target
ance        is        the        best.        More        bridge        nodes        can        explore        more        potential        re-                                                                                                                     interactions     using     bipartite     local     models.     Bioinformatics,     25,     2397–2403.
lationship              between               proteins               and               drugs               together.               However,               too                                                                                                Caly,L.       et       al.       (2020)       The       FDA-approved       drug       ivermectin        inhibits       the       replica-
many          bridge          nodes          will          lead          some         nodes          to          play          a          similar          role         in                                                                                          tion     of     SARS-CoV-2     in     vitro.     Antiviral     Res.,     178,     104787.
                                                                                                                                                                                                                                                             Chen,L.      et      al.      (2019)      Hidden      bias      in      the      dud-e      dataset      leads      to      misleading      per-
voting      and      bring       the      risk      of      overfitting      and      excessive      costs.                                                                                                                                                         formance       of        deep        learning        in        structure-based        virtual        screening.        PLoS        One,
                                                                                                                                                                                                                                                                    14,     e0220113.
                                                                                                                                                                                                                                                             Chen,L.          et          al.          (2020)          TransformerCPI:          improving          compound–protein          inter-
4        Case        study                                                                                                                                                                                                                                          action      prediction      by      sequence-based      deep      learning      with      self-attention      mech-
In         order         to         show         the         performance         of         BridgeDPI         in         practical         virtual                                                                                                                  anism              and              label              reversal              experiments.              Bioinformatics,               36,              4406–4414.
screening,            we            select            two            important            viral            targets,            3C-like            protease                                                                                                          [CrossRef][10.1093/bioinformatics/btaa524]
(3CLpro)         and         RNA-dependent         RNA         polymerase         (RdRp),         as         the         re-                                                                                                                                 Ding,H.        et       al.       (2014)       Similarity-based       machine       learning        methods        for       predict-
search       objects.      The      two       targets      play       a      major       role       in      the       protein      repli-                                                                                                                           ing     drug–target     interactions:     a     brief     review.     Brief.     Bioinform.,     15,     734–747.
                                                                                                                                                                                                                                                             Durrant,J.D.           and           McCammon,J.A.           (2011)           NNScore           2.0:           a           neural-network
cation/transcription       and       host       cell       recognition,       and       therefore       are       vital                                                                                                                                             receptor–ligand     scoring     function.     J.     Chem.     Inf.     Model.,     51,     2897–2903.
for          the          viral          reproduction          and          spread          of          infection         (Murugan         et          al.,                                                                                                  Elﬁky,A.A.      (2020)     Ribavirin,      remdesivir,     sofosbuvir,     galidesivir,     and     tenofovir
2020).      As      same      as      (Kim      et      al.,      2021),      we      also      select      some      candidates,                                                                                                                                   against       SARS-CoV-2       RNA       dependent       RNA       polymerase       (RdRp):       a       molecu-
such       as       Baricitinib       and       Ivermectin,       to       test       the       predicted       interactions.                                                                                                                                       lar     docking     study.     Life     Sci.,     253,     117592.
First,       we        obtain       the        amino       acid        sequences       of        the        two        targets        [the        se-                                                                                                        Favalli,E.G.           et           al.           (2020)           Baricitinib           for           COVID-19:           a           suitable           treatment?
quence           of           3CLpro           is           from           the           Protein           Data           Bank           (PDB)           database                                                                                                   Lancet     Infect.     Dis.,     20,     1012–1013.
(Sussman              et              al.,              1998)              with              PDB              ID              6WQF,              RdRp              is              from              the                                                     Fokoue,A.       et      al.       (2016).      Predicting       drug-drug      interactions       through       large-scale
National        Center        for        Biotechnology        Information        (NCBI;       Pruitt        et        al.,                                                                                                                                          similarity-based          link          prediction.          In:          European          Semantic          Web          Conference,
2007)           with           NCBI          YP_009725307.1].           Then,          the           sequences          and           the                                                                                                                           Crete,     Greece,     Springer,     pp.     774–789.
candidate        molecules         are         fed         into        BridgeDPI.        Finally,        the         interaction                                                                                                                             Gao,K.Y.       et       al.        (2018)       Interpretable        drug       target       prediction        using       deep       neural
probabilities       of      them      are       predicted,      as       shown      in      Table       5.                                                                                                                                                          representation.         In:         Proceedings         of         the         Twenty-Seventh          International          Joint
            Table                         5                        shows                        that                        Baricitinib,                        Remdesivir,                        Lopinavir                        and                             Conference                    on                    Artiﬁcial                    Intelligence                    (IJCAI-18),                    International                    Joint
Ritonavir                  are                  all                  very                  potential                  drugs                  that                  can                  interact                  with                  the                         Conferences     on     Artiﬁcial     Intelligence,     Stockholm,     pp.     3371–3377.
3CLpro;            Ivermectin,            Redmdesivir,            Sofosbuvir,            Daclatasvir,            Lopinavir                                                                                                                                   Gilson,M.K.       et       al.       (2016)       Bindingdb       in       2015:      a       public       database       for       medicinal
and          Ritonavir          are          all          effective          drugs          that          can          bind          with          the          RdRp.          In                                                                                   chemistry,            computational            chemistry            and           systems           pharmacology.            Nucleic
                                                                                                                                                                                                                                                                    Acids     Res.,     44,     D1045–D1053.
fact,              many              studies              and              clinical              trials              also              validate              the              results              (Caly                                                     Gschwend,D.A.        et       al.        (1996)       Molecular       docking       towards       drug       discovery.
et                al.,                2020;               Elfiky,                2020;               Favalli               et                al.,                2020;               Kalil               et                al.,                2021;                J.    Mol.    Recogn.,      9,     175–186.
Sadeghi        et          al.,          2020;         Stower,          2020).          In          contrast,          unrelated          drugs          such                                                                                                Kalil,A.C.            et            al.;            ACTT-2            Study            Group            Members.            (2021)            Baricitinib            plus
as         Aspirin         have         little         interaction         potential         with         the          viral         targets.         These                                                                                                         remdesivir          for          hospitalized          adults          with          covid-19.          N.          Engl.          J.          Med.,           384,
experimental           results           verify          the           validity           and           reliability           of           BridgeDPI           in                                                                                                   795–807.

2578                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Y.Wu       et       al.
Kim,Q.            et           al.            (2021)           Bayesian           neural           network           with           pretrained           protein                                                        Paul,S.M.       et       al.       (2010)       How       to       improve       R&D       productivity:       the       pharmaceut-
     embedding            enhances            prediction            accuracy            of           drug-protein           interaction.                                                                                     ical     industry’s     grand     challenge.     Nat.     Rev.     Drug     Discov.,     9,     203–214.
     Bioinformatics  ,     37,     3428–3435.                                                                                                                                                                           Pruitt,K.D.               et               al.               (2007)               NCBI               reference               sequences               (RefSeq):               a               curated
Kingma,D.P.      and       Ba,J.A.       (2019).      A      method      for       stochastic      optimization.       arxiv                                                                                                 non-redundant            sequence            database            of            genomes,            transcripts            and            proteins.
     2014.                 arXiv                 preprint                 arXiv:1412.6980,                 434.                 https://doi.org/10.48550/                                                                    Nucleic     Acids     Res.,     35,     D61–D65.
     arXiv.1412.6980.                                                                                                                                                                                                   Ramsundar,B.          et          al.          (2019).          Deep           Learning          for          the          Life          Sciences.          O’Reilly
Led,P.      and      Caﬂisch,A.      (2018)      Protein      structure-based      drug      design:      from      dock-                                                                                                    Media,     Inc.,     Sebastopol.
     ing     to     molecular     dynamics.     Curr.     Opin.     Struct.     Biol.,     48,     93–102.                                                                                                              Rogers,D.        and        Hahn,M.       (2010)       Extended-connectivity       ﬁngerprints.        J.        Chem.
Leslie,C.S.          et          al.          (2004)          Mismatch          string          kernels           for          discriminative          protein                                                               Inf.     Model.,     50,     742–754.
     classiﬁcation.     Bioinformatics,     20,     467–476.                                                                                                                                                            Sadeghi,A.     et     al.     (2020)     Sofosbuvir     and     daclatasvir     compared     with     standard      of
Li,M.               et               al.                (2022)                BACPI:                 a                 bi-directional                 attention                 neural                 network                forcare       in       the       treatment       of       patients       admitted       to       hospital       with       moderate       or       se-
     compound-protein       interaction       and       binding       afﬁnity       prediction.Bioinformatics,                                                                                                               vere            coronavirus            infection            (Covid-19):            a            randomized            controlled            trial.            J.
     38,     1995–2002.                                                                                                                                                                                                      Antimicrob.     Chemother.,     75,     3379–3385.
Li,S.          et          al.          (2020)          MONN:          a          multi-objective          neural          network          for          predicting                                                     Srivastava,N.       et      al.      (2014)      Dropout:      a      simple       way      to       prevent      neural      networks
     compound-protein     interactions     and     afﬁnities.     Cell     Syst.,     10,     308–322.                                                                                                                       from     overﬁtting.     J.     Mach.     Learn.     Res.,     15,     1929–1958.
Liu,H.         et         al.         (2015)         Improving         compound–protein         interaction         prediction                                                                                          Stower,H.     (2020)     Lopinavir–ritonavir     in     severe     Covid-19.     Nat.     Med.,     26,     465.
     by           building           up           highly            credible           negative           samples.               Bioinformatics  ,             31,                                                      Sussman,J.L.                      et                      al.                      (1998)                      Protein                      Data                      Bank                      (PDB):                      database                      of
     i221–i229.                                                                                                                                                                                                              three-dimensional               structural              information              of               biological               macromolecules.
Liu,T.       and       Altman,R.B.      (2015)      Relating      essential       proteins       to       drug      side-effects                                                                                             Acta     Crystallogr.     D     Biol.     Crystallogr.,     54,     1078–1084.
     using        canonical        component        analysis:        a       structure-based        approach.        J.        Chem.                                                                                    Tsubaki,M.               et              al.              (2019)              Compound–protein              interaction              prediction              with
     Inf.     Model.,     55,     1483–1494.                                                                                                                                                                                 end-to-end                    learning                   of                   neural                   networks                    for                   graphs                    and                   sequences.
Luo,H.      et      al.      (2019)      A      novel      drug      repositioning      approach      based      on      collabora-                                                                                          Bioinformatics,     35,     309–318.
     tive             metric             learning.             IEEE/ACM             Trans.             Comput.             Biol.             Bioinform.,             18,                                                Wan,F.           et            al.           (2019)           NeoDTI:           neural           integration            of           neighbor           information
     463–471.                                                                                                                                                                                                                from     a     heterogeneous     network     for     discovering     new     drug–target     interactions.
Luo,H.     et     al.     (2021)     Biomedical     data     and     computational     models     for     drug     repo-                                                                                                     Bioinformatics,     35,     104–111.
     sitioning:     a     comprehensive     review.     Brief.     Bioinform.,     22,     1604–1619.                                                                                                                   Wang,C.               and               Lukasz,K.               (2019)               Review               and               comparative               assessment                of
Luo,Y.          et          al.          (2017)          A          network          integration          approach          for          drug-target          inter-                                                         similarity-based        methods        for        prediction        of        drug-protein        interactions        in        the
     action     prediction     and     computational     drug     repositioning     from     heterogeneous                                                                                                                   druggable                            human                            proteome.                            Brief.                            Bioinform.,                             20,                            2066–2087.
     information.     Nat.     Commun.,     8,     1–13.                                                                                                                                                                     [CrossRef][10.1093/bib/bby069]
Maggiora,G.M.        et       al.       (2014)       Molecular       similarity       in       medicinal       chemistry.                                                                                               Wang,K.     et     al.     (2021)     DeepDTAF:     a     deep     learning     method     to     predict     protein–-
     J.     Med.     Chem.,      57,     3186–3204.                                                                                                                                                                          ligand     binding     afﬁnity.     Brief.     Bioinform.     22,     bbab072.
Mizianty,M.J.         et         al.         (2014)         Covering         complete         proteomes         with         X-ray         struc-                                                                       Wang,Y.                   and                   Zeng,J.                   (2013)                   Predicting                   drug-target                   interactions                   using
     tures:               a               current               snapshot.               Acta               Crystallogr.               D               Biol.               Crystallogr.,                70,                   restricted     Boltzmann     machines.     Bioinformatics,     29,     i126–i134.
                                                                                                                                                                                                                        Weininger,D.       (1988)       Smiles,       a       chemical        language       and       information       system.       1.
     2781–2793.                                                                                                                                                                                                              Introduction          to          methodology          and          encoding          rules.          J.          Chem.          Inf.          Comput.
Murugan,N.A.        et        al.        (2020)        Searching        for        target-speciﬁc        and        multi-targeting                                                                                          Sci.,     28,     31–36.
     organics       for       Covid-19       in       the       Drugbank       database       with       a       double       scoring       ap-                                                                         Yang,J.               et               al.               (2020)               Predicting               or               pretending:               artiﬁcial               intelligence               for
     proach.     Sci.     Rep.,     10,     1–16.                                                                                                                                                                            protein-ligand        interactions        lack        of        sufﬁciently        large        and        unbiased        datasets.
Mysinger,M.M.         et          al.          (2012)          Directory          of          useful          decoys,          enhanced          (dud-e):                                                                    Front.     Pharmacol.,     11,     69.
     better            ligands           and           decoys           for           better           benchmarking.           J.           Med.           Chem.,            55,                                        Yuvaraj,N.                      et                      al.                      (2021)                      Analysis                      of                      protein-ligand                      interactions                      of
     6582–6594.                                                                                                                                                                                                              SARS-CoV-2         against         selective        drug        using         deep         neural        networks.         Big        Data
Nguyen,T.          et         al.          (2021)         GraphDTA:          predicting          drug–target         binding          afﬁnity                                                                                Min.     Anal.,     4,     76–83.
     with     graph     neural     networks.     Bioinformatics,     37,     1140–1147.                                                                                                                                 Zhang,Q.C.       et       al.       (2012)       Structure-based       prediction       of       protein–protein       inter-
Paszke,A.        et        al.        (2019)        PyTorch:        an        imperative        style,        high-performance        deep                                                                                   actions     on     a     genome-wide     scale.     Nature,     490,     556–560.
     learning         library.         In:         Advances          in         Neural         Information         Processing         Systems,                                                                          Zheng,S.         et         al.        (2020)         Predicting         drug–protein         interaction         using         quasi-visual
     pp.     8026–8037.                                                                                                                                                                                                      question     answering     system.     Nat.     Mach.     Intell.,     2,     134–140.

